{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "3388eb60-271b-492f-b015-05d1b081691f",
   "metadata": {
    "kernel": "SoS",
    "tags": []
   },
   "source": [
    "# PCSK9 vs. APOC3\n",
    "\n",
    "This notebook records [BRV](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4459641/) analysis result for PCSK9 v.s LDL and APOC3 v.s. TG"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b486533c-8a01-4884-b060-3830c23c6116",
   "metadata": {
    "kernel": "SoS",
    "tags": []
   },
   "source": [
    "## Extract PCSK9 + APOC3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "76dea2bc-5912-4425-8fec-80915b7a7a5e",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(dplyr)\n",
    "library(data.table)\n",
    "setwd(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c0c78a1b-dbde-4be9-8cac-2a2f9ca1dc04",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### Exome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "e8ac29ed-6226-4947-bcce-c9cf66c0a3b3",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "for(chr in c(1,11)){\n",
    "    gene <- ifelse(chr==1, \"PCSK9\", \"APOC3\")\n",
    "    \n",
    "    for(maf in c(0.01, 0.005, 0.001)){\n",
    "        maf_c <- gsub(\"\\\\.\", \"\", as.character(maf))\n",
    "        \n",
    "        annot <- fread(sprintf(\"./exome/ukb23156_c%i.merged.filtered.hg38.hg38_multianno_formatted_sel_col_maf%s_LOF_missense.csv.gz\", chr, maf_c)) %>% \n",
    "            filter(Gene.refGene == gene)\n",
    "        annot %>% fwrite(sprintf(\"./%s/%s_exome_maf%s_annot.csv.gz\", tolower(gene), gene, maf_c))\n",
    "        annot %>% select(ID_hg38) %>% fwrite(sprintf(\"./%s/%s_exome_maf%s_extractlist_snplist\", tolower(gene), gene, maf_c), col.names = FALSE)\n",
    "        \n",
    "        annot_cadd <- fread(sprintf(\"./exome/ukb23156_c%i.merged.filtered.hg38.hg38_multianno_formatted_sel_col_maf%s_LOF_missense_cadd.csv.gz\", chr, maf_c)) %>% \n",
    "            filter(Gene.refGene == gene)\n",
    "        annot_cadd %>% fwrite(sprintf(\"./%s/%s_exome_maf%s_cadd_annot.csv.gz\", tolower(gene), gene, maf_c))\n",
    "        annot_cadd %>% select(ID_hg38) %>% fwrite(sprintf(\"./%s/%s_exome_maf%s_cadd_extractlist_snplist\", tolower(gene), gene, maf_c), col.names = FALSE)\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "136a6c79-3924-4bc0-be27-e15c446bac07",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### HRC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "8af5537b-0624-4548-8272-297b75b0b900",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "for(chr in c(1,11)){\n",
    "    gene <- ifelse(chr==1, \"PCSK9\", \"APOC3\")\n",
    "    \n",
    "    for(rsq in c(3, 8)){\n",
    "        for(maf in c(0.01, 0.005, 0.001)){\n",
    "            maf_c <- gsub(\"\\\\.\", \"\", as.character(maf))\n",
    "\n",
    "            annot <- fread(sprintf(\"./hrc/hrc_chr%i_rsq0%i_hg19_hg38_maf%s_LOF_missense_annot.csv.gz\", chr, rsq, maf_c)) %>% \n",
    "                filter(Gene.refGene == gene)\n",
    "            annot %>% fwrite(sprintf(\"./%s/%s_hrc_rsq0%i_maf%s_annot.csv.gz\", tolower(gene), gene, rsq, maf_c))\n",
    "            annot %>% select(ID_hg19) %>% fwrite(sprintf(\"./%s/%s_hrc_rsq0%i_maf%s_extractlist_snplist\", tolower(gene), gene, rsq, maf_c), col.names = FALSE)\n",
    "        }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "08d7d176-1f71-4412-8468-3a00a6d8e0d7",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "for(chr in c(1, 11)){\n",
    "    gene <- ifelse(chr==1, \"PCSK9\", \"APOC3\")\n",
    "    \n",
    "    for(rsq in c(3, 8)){\n",
    "        for(maf in c(0.01, 0.005, 0.001)){\n",
    "            maf_c <- gsub(\"\\\\.\", \"\", as.character(maf))\n",
    "\n",
    "            annot_cadd <- fread(sprintf(\"./hrc/hrc_chr%i_rsq0%i_hg19_hg38_maf%s_LOF_missense_cadd_annot.csv.gz\", chr, rsq, maf_c)) %>% \n",
    "                filter(Gene.refGene == gene)\n",
    "            annot_cadd %>% fwrite(sprintf(\"./%s/%s_hrc_rsq0%i_maf%s_cadd_annot.csv.gz\", tolower(gene), gene, rsq, maf_c))\n",
    "            annot_cadd %>% select(ID_hg19) %>% fwrite(sprintf(\"./%s/%s_hrc_rsq0%i_maf%s_cadd_extractlist_snplist\", tolower(gene), gene, rsq, maf_c), col.names = FALSE)\n",
    "        }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "36daeda2-8c72-4161-8569-6f5b35e3b234",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### TOPMed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "f6c94c59-734f-47ee-89e0-d451136cb573",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "for(chr in c(1,11)){\n",
    "    gene <- ifelse(chr==1, \"PCSK9\", \"APOC3\")\n",
    "    \n",
    "    for(rsq in c(3, 8)){\n",
    "        for(maf in c(0.01, 0.005, 0.001)){\n",
    "            maf_c <- gsub(\"\\\\.\", \"\", as.character(maf))\n",
    "\n",
    "            annot <- fread(sprintf(\"./topmed/topmed_chr%i_rsq0%i_hg19_hg38_maf%s_LOF_missense_annot.csv.gz\", chr, rsq, maf_c)) %>% \n",
    "                filter(Gene.refGene == gene)\n",
    "            annot %>% fwrite(sprintf(\"./%s/%s_topmed_rsq0%i_maf%s_annot.csv.gz\", tolower(gene), gene, rsq, maf_c))\n",
    "            annot %>% select(ID_hg38) %>% fwrite(sprintf(\"./%s/%s_topmed_rsq0%i_maf%s_extractlist_snplist\", tolower(gene), gene, rsq, maf_c), col.names = FALSE)\n",
    "        }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "42536aeb-453f-4600-8935-0ad8479a53c9",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "for(chr in c(1, 11)){\n",
    "    gene <- ifelse(chr==1, \"PCSK9\", \"APOC3\")\n",
    "    \n",
    "    for(rsq in c(3, 8)){\n",
    "        for(maf in c(0.01, 0.005, 0.001)){\n",
    "            maf_c <- gsub(\"\\\\.\", \"\", as.character(maf))\n",
    "\n",
    "            annot_cadd <- fread(sprintf(\"./topmed/topmed_chr%i_rsq0%i_hg19_hg38_maf%s_LOF_missense_cadd_annot.csv.gz\", chr, rsq, maf_c)) %>% \n",
    "                filter(Gene.refGene == gene)\n",
    "            annot_cadd %>% fwrite(sprintf(\"./%s/%s_topmed_rsq0%i_maf%s_cadd_annot.csv.gz\", tolower(gene), gene, rsq, maf_c))\n",
    "            annot_cadd %>% select(ID_hg38) %>% fwrite(sprintf(\"./%s/%s_topmed_rsq0%i_maf%s_cadd_extractlist_snplist\", tolower(gene), gene, rsq, maf_c), col.names = FALSE)\n",
    "        }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f1f3d44f-3483-465c-9cc3-e9ac54d7dd88",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "### Extract"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "14718091-0b4f-4460-b2ed-323515e32bdc",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_exome_maf001_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c1_maf001_LOF_missense_extracted\n",
      "  --extract PCSK9_exome_maf001_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out PCSK9_exome_maf001_extracted\n",
      "\n",
      "Start time: Tue Mar  5 11:50:54 2024\n",
      "257483 MiB RAM detected, ~148510 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c1_maf001_LOF_missense_extracted.fam.\n",
      "478357 variants loaded from\n",
      "../exome/ukb23156_c1_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 394 variants remaining.\n",
      "394 variants remaining after main filters.\n",
      "Writing PCSK9_exome_maf001_extracted.fam ... done.\n",
      "Writing PCSK9_exome_maf001_extracted.bim ... done.\n",
      "Writing PCSK9_exome_maf001_extracted.bed ... done.\n",
      "End time: Tue Mar  5 11:50:54 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq03_maf001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq03_maf001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_hrc_rsq03_maf001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq03_maf001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:50:54 2024\n",
      "257483 MiB RAM detected, ~147824 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq03_maf001_LOF_missense_extracted.fam.\n",
      "20551 variants loaded from\n",
      "../hrc/hrc_chr1_rsq03_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 16 variants remaining.\n",
      "16 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq03_maf001_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf001_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf001_extracted.pgen ... done.\n",
      "--export Av to PCSK9_hrc_rsq03_maf001_extracted.traw ... 1218253137435056626875818793done.\n",
      "End time: Tue Mar  5 11:50:54 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_rsq03_maf001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr1_rsq03_maf001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_topmed_rsq03_maf001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_rsq03_maf001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:50:54 2024\n",
      "257483 MiB RAM detected, ~147418 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr1_rsq03_maf001_LOF_missense_extracted.fam.\n",
      "123704 variants loaded from\n",
      "../topmed/topmed_chr1_rsq03_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 109 variants remaining.\n",
      "109 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_rsq03_maf001_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_rsq03_maf001_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_rsq03_maf001_extracted.pgen ... done.\n",
      "--export Av to PCSK9_topmed_rsq03_maf001_extracted.traw ... 1011111213141516171819202122222324252627282930313233333435363738394041424344444546474849505152535455555657585960616263646566666768697071727374757677777879808182838485868788888990919293949596979899done.\n",
      "End time: Tue Mar  5 11:50:54 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq08_maf001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq08_maf001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_hrc_rsq08_maf001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq08_maf001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:50:54 2024\n",
      "257483 MiB RAM detected, ~146672 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq08_maf001_LOF_missense_extracted.fam.\n",
      "8396 variants loaded from\n",
      "../hrc/hrc_chr1_rsq08_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 6 variants remaining.\n",
      "6 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq08_maf001_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf001_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf001_extracted.pgen ... done.\n",
      "--export Av to PCSK9_hrc_rsq08_maf001_extracted.traw ... 1633506683done.\n",
      "End time: Tue Mar  5 11:50:54 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_rsq08_maf001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr1_rsq08_maf001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_topmed_rsq08_maf001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_rsq08_maf001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:50:54 2024\n",
      "257483 MiB RAM detected, ~146383 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr1_rsq08_maf001_LOF_missense_extracted.fam.\n",
      "44948 variants loaded from\n",
      "../topmed/topmed_chr1_rsq08_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 47 variants remaining.\n",
      "47 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_rsq08_maf001_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_rsq08_maf001_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_rsq08_maf001_extracted.pgen ... done.\n",
      "--export Av to PCSK9_topmed_rsq08_maf001_extracted.traw ... 101214171921232527293134363840424446485153555759616365687072747678808285878991939597done.\n",
      "End time: Tue Mar  5 11:50:54 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_exome_maf0005_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c1_maf0005_LOF_missense_extracted\n",
      "  --extract PCSK9_exome_maf0005_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out PCSK9_exome_maf0005_extracted\n",
      "\n",
      "Start time: Tue Mar  5 11:50:54 2024\n",
      "257483 MiB RAM detected, ~145914 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c1_maf0005_LOF_missense_extracted.fam.\n",
      "477452 variants loaded from\n",
      "../exome/ukb23156_c1_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 394 variants remaining.\n",
      "394 variants remaining after main filters.\n",
      "Writing PCSK9_exome_maf0005_extracted.fam ... done.\n",
      "Writing PCSK9_exome_maf0005_extracted.bim ... done.\n",
      "Writing PCSK9_exome_maf0005_extracted.bed ... done.\n",
      "End time: Tue Mar  5 11:50:55 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq03_maf0005_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq03_maf0005_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_hrc_rsq03_maf0005_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq03_maf0005_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:50:55 2024\n",
      "257483 MiB RAM detected, ~145268 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq03_maf0005_LOF_missense_extracted.fam.\n",
      "19743 variants loaded from\n",
      "../hrc/hrc_chr1_rsq03_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 16 variants remaining.\n",
      "16 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq03_maf0005_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf0005_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf0005_extracted.pgen ... done.\n",
      "--export Av to PCSK9_hrc_rsq03_maf0005_extracted.traw ... 1218253137435056626875818793done.\n",
      "End time: Tue Mar  5 11:50:55 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_rsq03_maf0005_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr1_rsq03_maf0005_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_topmed_rsq03_maf0005_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_rsq03_maf0005_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:50:55 2024\n",
      "257483 MiB RAM detected, ~144896 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr1_rsq03_maf0005_LOF_missense_extracted.fam.\n",
      "122865 variants loaded from\n",
      "../topmed/topmed_chr1_rsq03_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 109 variants remaining.\n",
      "109 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_rsq03_maf0005_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_rsq03_maf0005_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_rsq03_maf0005_extracted.pgen ... done.\n",
      "--export Av to PCSK9_topmed_rsq03_maf0005_extracted.traw ... 1011111213141516171819202122222324252627282930313233333435363738394041424344444546474849505152535455555657585960616263646566666768697071727374757677777879808182838485868788888990919293949596979899done.\n",
      "End time: Tue Mar  5 11:50:55 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq08_maf0005_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq08_maf0005_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_hrc_rsq08_maf0005_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq08_maf0005_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:50:55 2024\n",
      "257483 MiB RAM detected, ~144182 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq08_maf0005_LOF_missense_extracted.fam.\n",
      "7666 variants loaded from\n",
      "../hrc/hrc_chr1_rsq08_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 6 variants remaining.\n",
      "6 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq08_maf0005_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf0005_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf0005_extracted.pgen ... done.\n",
      "--export Av to PCSK9_hrc_rsq08_maf0005_extracted.traw ... 1633506683done.\n",
      "End time: Tue Mar  5 11:50:55 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_rsq08_maf0005_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr1_rsq08_maf0005_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_topmed_rsq08_maf0005_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_rsq08_maf0005_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:50:55 2024\n",
      "257483 MiB RAM detected, ~143909 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr1_rsq08_maf0005_LOF_missense_extracted.fam.\n",
      "44134 variants loaded from\n",
      "../topmed/topmed_chr1_rsq08_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 47 variants remaining.\n",
      "47 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_rsq08_maf0005_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_rsq08_maf0005_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_rsq08_maf0005_extracted.pgen ... done.\n",
      "--export Av to PCSK9_topmed_rsq08_maf0005_extracted.traw ... 101214171921232527293134363840424446485153555759616365687072747678808285878991939597done.\n",
      "End time: Tue Mar  5 11:50:55 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_exome_maf0001_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c1_maf0001_LOF_missense_extracted\n",
      "  --extract PCSK9_exome_maf0001_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out PCSK9_exome_maf0001_extracted\n",
      "\n",
      "Start time: Tue Mar  5 11:50:55 2024\n",
      "257483 MiB RAM detected, ~143462 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c1_maf0001_LOF_missense_extracted.fam.\n",
      "473986 variants loaded from\n",
      "../exome/ukb23156_c1_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 393 variants remaining.\n",
      "393 variants remaining after main filters.\n",
      "Writing PCSK9_exome_maf0001_extracted.fam ... done.\n",
      "Writing PCSK9_exome_maf0001_extracted.bim ... done.\n",
      "Writing PCSK9_exome_maf0001_extracted.bed ... done.\n",
      "End time: Tue Mar  5 11:50:56 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq03_maf0001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq03_maf0001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_hrc_rsq03_maf0001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq03_maf0001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:50:56 2024\n",
      "257483 MiB RAM detected, ~142726 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq03_maf0001_LOF_missense_extracted.fam.\n",
      "16591 variants loaded from\n",
      "../hrc/hrc_chr1_rsq03_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 15 variants remaining.\n",
      "15 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq03_maf0001_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf0001_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf0001_extracted.pgen ... done.\n",
      "--export Av to PCSK9_hrc_rsq03_maf0001_extracted.traw ... 13202633404653606673808693done.\n",
      "End time: Tue Mar  5 11:50:56 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_rsq03_maf0001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr1_rsq03_maf0001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_topmed_rsq03_maf0001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_rsq03_maf0001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:50:56 2024\n",
      "257483 MiB RAM detected, ~142379 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr1_rsq03_maf0001_LOF_missense_extracted.fam.\n",
      "119617 variants loaded from\n",
      "../topmed/topmed_chr1_rsq03_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 108 variants remaining.\n",
      "108 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_rsq03_maf0001_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_rsq03_maf0001_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_rsq03_maf0001_extracted.pgen ... done.\n",
      "--export Av to PCSK9_topmed_rsq03_maf0001_extracted.traw ... 10111212131415161718192021222324252526272829303132333435363737383940414243444546474849505051525354555657585960616262636465666768697071727374757576777879808182838485868787888990919293949596979899done.\n",
      "End time: Tue Mar  5 11:50:56 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq08_maf0001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq08_maf0001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_hrc_rsq08_maf0001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq08_maf0001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:50:56 2024\n",
      "257483 MiB RAM detected, ~141607 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq08_maf0001_LOF_missense_extracted.fam.\n",
      "5243 variants loaded from\n",
      "../hrc/hrc_chr1_rsq08_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 6 variants remaining.\n",
      "6 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq08_maf0001_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf0001_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf0001_extracted.pgen ... done.\n",
      "--export Av to PCSK9_hrc_rsq08_maf0001_extracted.traw ... 1633506683done.\n",
      "End time: Tue Mar  5 11:50:56 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_rsq08_maf0001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr1_rsq08_maf0001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_topmed_rsq08_maf0001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_rsq08_maf0001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:50:56 2024\n",
      "257483 MiB RAM detected, ~141207 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr1_rsq08_maf0001_LOF_missense_extracted.fam.\n",
      "41265 variants loaded from\n",
      "../topmed/topmed_chr1_rsq08_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 46 variants remaining.\n",
      "46 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_rsq08_maf0001_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_rsq08_maf0001_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_rsq08_maf0001_extracted.pgen ... done.\n",
      "--export Av to PCSK9_topmed_rsq08_maf0001_extracted.traw ... 1013151719212326283032343639414345475052545658606365676971737678808284868991939597done.\n",
      "End time: Tue Mar  5 11:50:56 2024\n"
     ]
    }
   ],
   "source": [
    "cd ~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/pcsk9\n",
    "\n",
    "module load Plink/2.00a\n",
    "\n",
    "for i in 1 05 01; do\n",
    "    plink2 \\\n",
    "        --bfile ../exome/ukb23156_c1_maf00${i}_LOF_missense_extracted \\\n",
    "        --extract PCSK9_exome_maf00${i}_extractlist_snplist \\\n",
    "        --make-bed \\\n",
    "        --out PCSK9_exome_maf00${i}_extracted\n",
    "\n",
    "    for j in 3 8; do\n",
    "        plink2 \\\n",
    "            --bpfile ../hrc/hrc_chr1_rsq0${j}_maf00${i}_LOF_missense_extracted \\\n",
    "            --extract PCSK9_hrc_rsq0${j}_maf00${i}_extractlist_snplist \\\n",
    "            --make-bpgen --sort-vars \\\n",
    "            --export A-transpose \\\n",
    "            --out PCSK9_hrc_rsq0${j}_maf00${i}_extracted\n",
    "\n",
    "        plink2 \\\n",
    "            --bpfile ../topmed/topmed_chr1_rsq0${j}_maf00${i}_LOF_missense_extracted \\\n",
    "            --extract PCSK9_topmed_rsq0${j}_maf00${i}_extractlist_snplist \\\n",
    "            --make-bpgen --sort-vars \\\n",
    "            --export A-transpose \\\n",
    "            --out PCSK9_topmed_rsq0${j}_maf00${i}_extracted\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "706717d5-8ee4-4657-9bac-77bc3bfc791a",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_exome_maf001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c1_maf001_LOF_missense_extracted\n",
      "  --extract PCSK9_exome_maf001_cadd_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out PCSK9_exome_maf001_cadd_extracted\n",
      "\n",
      "Start time: Tue Mar  5 11:51:22 2024\n",
      "257483 MiB RAM detected, ~169645 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c1_maf001_LOF_missense_extracted.fam.\n",
      "478357 variants loaded from\n",
      "../exome/ukb23156_c1_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 200 variants remaining.\n",
      "200 variants remaining after main filters.\n",
      "Writing PCSK9_exome_maf001_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_exome_maf001_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_exome_maf001_cadd_extracted.bed ... done.\n",
      "End time: Tue Mar  5 11:51:22 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq03_maf001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq03_maf001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_hrc_rsq03_maf001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq03_maf001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:22 2024\n",
      "257483 MiB RAM detected, ~169221 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq03_maf001_LOF_missense_extracted.fam.\n",
      "20551 variants loaded from\n",
      "../hrc/hrc_chr1_rsq03_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 6 variants remaining.\n",
      "6 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq03_maf001_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf001_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf001_cadd_extracted.pgen ... done.\n",
      "--export Av to PCSK9_hrc_rsq03_maf001_cadd_extracted.traw ... 1633506683done.\n",
      "End time: Tue Mar  5 11:51:22 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_rsq03_maf001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr1_rsq03_maf001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_topmed_rsq03_maf001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_rsq03_maf001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:22 2024\n",
      "257483 MiB RAM detected, ~168928 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr1_rsq03_maf001_LOF_missense_extracted.fam.\n",
      "123704 variants loaded from\n",
      "../topmed/topmed_chr1_rsq03_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 61 variants remaining.\n",
      "61 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_rsq03_maf001_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_rsq03_maf001_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_rsq03_maf001_cadd_extracted.pgen ... done.\n",
      "--export Av to PCSK9_topmed_rsq03_maf001_cadd_extracted.traw ... 111314161819212224262729313234363739404244454749505254555759606263656768707273757778808183858688909193959698done.\n",
      "End time: Tue Mar  5 11:51:22 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq08_maf001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq08_maf001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_hrc_rsq08_maf001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq08_maf001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:22 2024\n",
      "257483 MiB RAM detected, ~168434 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq08_maf001_LOF_missense_extracted.fam.\n",
      "8396 variants loaded from\n",
      "../hrc/hrc_chr1_rsq08_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 2 variants remaining.\n",
      "2 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq08_maf001_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf001_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf001_cadd_extracted.pgen ... done.\n",
      "--export Av to PCSK9_hrc_rsq08_maf001_cadd_extracted.traw ... 50done.\n",
      "End time: Tue Mar  5 11:51:22 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_rsq08_maf001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr1_rsq08_maf001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_topmed_rsq08_maf001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_rsq08_maf001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:22 2024\n",
      "257483 MiB RAM detected, ~168208 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr1_rsq08_maf001_LOF_missense_extracted.fam.\n",
      "44948 variants loaded from\n",
      "../topmed/topmed_chr1_rsq08_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 25 variants remaining.\n",
      "25 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_rsq08_maf001_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_rsq08_maf001_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_rsq08_maf001_cadd_extracted.pgen ... done.\n",
      "--export Av to PCSK9_topmed_rsq08_maf001_cadd_extracted.traw ... 12162024283236404448525660646872768084889296done.\n",
      "End time: Tue Mar  5 11:51:22 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_exome_maf0005_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c1_maf0005_LOF_missense_extracted\n",
      "  --extract PCSK9_exome_maf0005_cadd_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out PCSK9_exome_maf0005_cadd_extracted\n",
      "\n",
      "Start time: Tue Mar  5 11:51:22 2024\n",
      "257483 MiB RAM detected, ~167848 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c1_maf0005_LOF_missense_extracted.fam.\n",
      "477452 variants loaded from\n",
      "../exome/ukb23156_c1_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 200 variants remaining.\n",
      "200 variants remaining after main filters.\n",
      "Writing PCSK9_exome_maf0005_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_exome_maf0005_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_exome_maf0005_cadd_extracted.bed ... done.\n",
      "End time: Tue Mar  5 11:51:22 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq03_maf0005_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq03_maf0005_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_hrc_rsq03_maf0005_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq03_maf0005_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:22 2024\n",
      "257483 MiB RAM detected, ~167412 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq03_maf0005_LOF_missense_extracted.fam.\n",
      "19743 variants loaded from\n",
      "../hrc/hrc_chr1_rsq03_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 6 variants remaining.\n",
      "6 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq03_maf0005_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf0005_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf0005_cadd_extracted.pgen ... done.\n",
      "--export Av to PCSK9_hrc_rsq03_maf0005_cadd_extracted.traw ... 1633506683done.\n",
      "End time: Tue Mar  5 11:51:23 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_rsq03_maf0005_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr1_rsq03_maf0005_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_topmed_rsq03_maf0005_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_rsq03_maf0005_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:23 2024\n",
      "257483 MiB RAM detected, ~167120 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr1_rsq03_maf0005_LOF_missense_extracted.fam.\n",
      "122865 variants loaded from\n",
      "../topmed/topmed_chr1_rsq03_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 61 variants remaining.\n",
      "61 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_rsq03_maf0005_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_rsq03_maf0005_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_rsq03_maf0005_cadd_extracted.pgen ... done.\n",
      "--export Av to PCSK9_topmed_rsq03_maf0005_cadd_extracted.traw ... 111314161819212224262729313234363739404244454749505254555759606263656768707273757778808183858688909193959698done.\n",
      "End time: Tue Mar  5 11:51:23 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq08_maf0005_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq08_maf0005_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_hrc_rsq08_maf0005_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq08_maf0005_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:23 2024\n",
      "257483 MiB RAM detected, ~166541 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq08_maf0005_LOF_missense_extracted.fam.\n",
      "7666 variants loaded from\n",
      "../hrc/hrc_chr1_rsq08_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 2 variants remaining.\n",
      "2 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq08_maf0005_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf0005_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf0005_cadd_extracted.pgen ... done.\n",
      "--export Av to PCSK9_hrc_rsq08_maf0005_cadd_extracted.traw ... 50done.\n",
      "End time: Tue Mar  5 11:51:23 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_rsq08_maf0005_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr1_rsq08_maf0005_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_topmed_rsq08_maf0005_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_rsq08_maf0005_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:23 2024\n",
      "257483 MiB RAM detected, ~166280 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr1_rsq08_maf0005_LOF_missense_extracted.fam.\n",
      "44134 variants loaded from\n",
      "../topmed/topmed_chr1_rsq08_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 25 variants remaining.\n",
      "25 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_rsq08_maf0005_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_rsq08_maf0005_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_rsq08_maf0005_cadd_extracted.pgen ... done.\n",
      "--export Av to PCSK9_topmed_rsq08_maf0005_cadd_extracted.traw ... 12162024283236404448525660646872768084889296done.\n",
      "End time: Tue Mar  5 11:51:23 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_exome_maf0001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c1_maf0001_LOF_missense_extracted\n",
      "  --extract PCSK9_exome_maf0001_cadd_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out PCSK9_exome_maf0001_cadd_extracted\n",
      "\n",
      "Start time: Tue Mar  5 11:51:23 2024\n",
      "257483 MiB RAM detected, ~166021 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c1_maf0001_LOF_missense_extracted.fam.\n",
      "473986 variants loaded from\n",
      "../exome/ukb23156_c1_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 199 variants remaining.\n",
      "199 variants remaining after main filters.\n",
      "Writing PCSK9_exome_maf0001_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_exome_maf0001_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_exome_maf0001_cadd_extracted.bed ... done.\n",
      "End time: Tue Mar  5 11:51:23 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq03_maf0001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq03_maf0001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_hrc_rsq03_maf0001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq03_maf0001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:23 2024\n",
      "257483 MiB RAM detected, ~165588 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq03_maf0001_LOF_missense_extracted.fam.\n",
      "16591 variants loaded from\n",
      "../hrc/hrc_chr1_rsq03_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 5 variants remaining.\n",
      "5 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq03_maf0001_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf0001_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf0001_cadd_extracted.pgen ... done.\n",
      "--export Av to PCSK9_hrc_rsq03_maf0001_cadd_extracted.traw ... 20406080done.\n",
      "End time: Tue Mar  5 11:51:23 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_rsq03_maf0001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr1_rsq03_maf0001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_topmed_rsq03_maf0001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_rsq03_maf0001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:23 2024\n",
      "257483 MiB RAM detected, ~165316 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr1_rsq03_maf0001_LOF_missense_extracted.fam.\n",
      "119617 variants loaded from\n",
      "../topmed/topmed_chr1_rsq03_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 60 variants remaining.\n",
      "60 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_rsq03_maf0001_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_rsq03_maf0001_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_rsq03_maf0001_cadd_extracted.pgen ... done.\n",
      "--export Av to PCSK9_topmed_rsq03_maf0001_cadd_extracted.traw ... 101113151618202123252628303133353638404143454648505153555658606163656668707173757678808183858688909193959698done.\n",
      "End time: Tue Mar  5 11:51:23 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq08_maf0001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq08_maf0001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_hrc_rsq08_maf0001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq08_maf0001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:24 2024\n",
      "257483 MiB RAM detected, ~164801 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq08_maf0001_LOF_missense_extracted.fam.\n",
      "5243 variants loaded from\n",
      "../hrc/hrc_chr1_rsq08_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 2 variants remaining.\n",
      "2 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq08_maf0001_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf0001_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf0001_cadd_extracted.pgen ... done.\n",
      "--export Av to PCSK9_hrc_rsq08_maf0001_cadd_extracted.traw ... 50done.\n",
      "End time: Tue Mar  5 11:51:24 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_rsq08_maf0001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr1_rsq08_maf0001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract PCSK9_topmed_rsq08_maf0001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_rsq08_maf0001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:24 2024\n",
      "257483 MiB RAM detected, ~164541 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr1_rsq08_maf0001_LOF_missense_extracted.fam.\n",
      "41265 variants loaded from\n",
      "../topmed/topmed_chr1_rsq08_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 24 variants remaining.\n",
      "24 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_rsq08_maf0001_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_rsq08_maf0001_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_rsq08_maf0001_cadd_extracted.pgen ... done.\n",
      "--export Av to PCSK9_topmed_rsq08_maf0001_cadd_extracted.traw ... 121620252933374145505458626670757983879195done.\n",
      "End time: Tue Mar  5 11:51:24 2024\n"
     ]
    }
   ],
   "source": [
    "for i in 1 05 01; do\n",
    "    plink2 \\\n",
    "        --bfile ../exome/ukb23156_c1_maf00${i}_LOF_missense_extracted \\\n",
    "        --extract PCSK9_exome_maf00${i}_cadd_extractlist_snplist \\\n",
    "        --make-bed \\\n",
    "        --out PCSK9_exome_maf00${i}_cadd_extracted\n",
    "\n",
    "    for j in 3 8; do\n",
    "        plink2 \\\n",
    "            --bpfile ../hrc/hrc_chr1_rsq0${j}_maf00${i}_LOF_missense_extracted \\\n",
    "            --extract PCSK9_hrc_rsq0${j}_maf00${i}_cadd_extractlist_snplist \\\n",
    "            --make-bpgen --sort-vars \\\n",
    "            --export A-transpose \\\n",
    "            --out PCSK9_hrc_rsq0${j}_maf00${i}_cadd_extracted\n",
    "\n",
    "        plink2 \\\n",
    "            --bpfile ../topmed/topmed_chr1_rsq0${j}_maf00${i}_LOF_missense_extracted \\\n",
    "            --extract PCSK9_topmed_rsq0${j}_maf00${i}_cadd_extractlist_snplist \\\n",
    "            --make-bpgen --sort-vars \\\n",
    "            --export A-transpose \\\n",
    "            --out PCSK9_topmed_rsq0${j}_maf00${i}_cadd_extracted\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1cc93470-3364-43c2-8915-0152ba22e9c9",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_exome_maf001_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c11_maf001_LOF_missense_extracted\n",
      "  --extract APOC3_exome_maf001_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out APOC3_exome_maf001_extracted\n",
      "\n",
      "Start time: Tue Mar  5 11:51:29 2024\n",
      "257483 MiB RAM detected, ~148366 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c11_maf001_LOF_missense_extracted.fam.\n",
      "293784 variants loaded from\n",
      "../exome/ukb23156_c11_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 60 variants remaining.\n",
      "60 variants remaining after main filters.\n",
      "Writing APOC3_exome_maf001_extracted.fam ... done.\n",
      "Writing APOC3_exome_maf001_extracted.bim ... done.\n",
      "Writing APOC3_exome_maf001_extracted.bed ... done.\n",
      "End time: Tue Mar  5 11:51:29 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq03_maf001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq03_maf001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_hrc_rsq03_maf001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq03_maf001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:29 2024\n",
      "257483 MiB RAM detected, ~147850 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq03_maf001_LOF_missense_extracted.fam.\n",
      "13392 variants loaded from\n",
      "../hrc/hrc_chr11_rsq03_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 3 variants remaining.\n",
      "3 variants remaining after main filters.\n",
      "Writing APOC3_hrc_rsq03_maf001_extracted.bim ... done.\n",
      "Writing APOC3_hrc_rsq03_maf001_extracted.fam ... done.\n",
      "Writing APOC3_hrc_rsq03_maf001_extracted.pgen ... done.\n",
      "--export Av to APOC3_hrc_rsq03_maf001_extracted.traw ... 3366done.\n",
      "End time: Tue Mar  5 11:51:30 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_rsq03_maf001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr11_rsq03_maf001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_topmed_rsq03_maf001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_rsq03_maf001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:30 2024\n",
      "257483 MiB RAM detected, ~147526 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr11_rsq03_maf001_LOF_missense_extracted.fam.\n",
      "77513 variants loaded from\n",
      "../topmed/topmed_chr11_rsq03_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 15 variants remaining.\n",
      "15 variants remaining after main filters.\n",
      "Writing APOC3_topmed_rsq03_maf001_extracted.bim ... done.\n",
      "Writing APOC3_topmed_rsq03_maf001_extracted.fam ... done.\n",
      "Writing APOC3_topmed_rsq03_maf001_extracted.pgen ... done.\n",
      "--export Av to APOC3_topmed_rsq03_maf001_extracted.traw ... 13202633404653606673808693done.\n",
      "End time: Tue Mar  5 11:51:30 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq08_maf001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq08_maf001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_hrc_rsq08_maf001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq08_maf001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:30 2024\n",
      "257483 MiB RAM detected, ~147085 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq08_maf001_LOF_missense_extracted.fam.\n",
      "5811 variants loaded from\n",
      "../hrc/hrc_chr11_rsq08_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 2 variants remaining.\n",
      "2 variants remaining after main filters.\n",
      "Writing APOC3_hrc_rsq08_maf001_extracted.bim ... done.\n",
      "Writing APOC3_hrc_rsq08_maf001_extracted.fam ... done.\n",
      "Writing APOC3_hrc_rsq08_maf001_extracted.pgen ... done.\n",
      "--export Av to APOC3_hrc_rsq08_maf001_extracted.traw ... 50done.\n",
      "End time: Tue Mar  5 11:51:30 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_rsq08_maf001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr11_rsq08_maf001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_topmed_rsq08_maf001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_rsq08_maf001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:30 2024\n",
      "257483 MiB RAM detected, ~146703 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr11_rsq08_maf001_LOF_missense_extracted.fam.\n",
      "28631 variants loaded from\n",
      "../topmed/topmed_chr11_rsq08_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 5 variants remaining.\n",
      "5 variants remaining after main filters.\n",
      "Writing APOC3_topmed_rsq08_maf001_extracted.bim ... done.\n",
      "Writing APOC3_topmed_rsq08_maf001_extracted.fam ... done.\n",
      "Writing APOC3_topmed_rsq08_maf001_extracted.pgen ... done.\n",
      "--export Av to APOC3_topmed_rsq08_maf001_extracted.traw ... 20406080done.\n",
      "End time: Tue Mar  5 11:51:30 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_exome_maf0005_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c11_maf0005_LOF_missense_extracted\n",
      "  --extract APOC3_exome_maf0005_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out APOC3_exome_maf0005_extracted\n",
      "\n",
      "Start time: Tue Mar  5 11:51:30 2024\n",
      "257483 MiB RAM detected, ~146347 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c11_maf0005_LOF_missense_extracted.fam.\n",
      "293221 variants loaded from\n",
      "../exome/ukb23156_c11_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 60 variants remaining.\n",
      "60 variants remaining after main filters.\n",
      "Writing APOC3_exome_maf0005_extracted.fam ... done.\n",
      "Writing APOC3_exome_maf0005_extracted.bim ... done.\n",
      "Writing APOC3_exome_maf0005_extracted.bed ... done.\n",
      "End time: Tue Mar  5 11:51:30 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq03_maf0005_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq03_maf0005_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_hrc_rsq03_maf0005_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq03_maf0005_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:30 2024\n",
      "257483 MiB RAM detected, ~145863 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq03_maf0005_LOF_missense_extracted.fam.\n",
      "12867 variants loaded from\n",
      "../hrc/hrc_chr11_rsq03_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 3 variants remaining.\n",
      "3 variants remaining after main filters.\n",
      "Writing APOC3_hrc_rsq03_maf0005_extracted.bim ... done.\n",
      "Writing APOC3_hrc_rsq03_maf0005_extracted.fam ... done.\n",
      "Writing APOC3_hrc_rsq03_maf0005_extracted.pgen ... done.\n",
      "--export Av to APOC3_hrc_rsq03_maf0005_extracted.traw ... 3366done.\n",
      "End time: Tue Mar  5 11:51:30 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_rsq03_maf0005_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr11_rsq03_maf0005_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_topmed_rsq03_maf0005_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_rsq03_maf0005_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:30 2024\n",
      "257483 MiB RAM detected, ~145401 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr11_rsq03_maf0005_LOF_missense_extracted.fam.\n",
      "76984 variants loaded from\n",
      "../topmed/topmed_chr11_rsq03_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 15 variants remaining.\n",
      "15 variants remaining after main filters.\n",
      "Writing APOC3_topmed_rsq03_maf0005_extracted.bim ... done.\n",
      "Writing APOC3_topmed_rsq03_maf0005_extracted.fam ... done.\n",
      "Writing APOC3_topmed_rsq03_maf0005_extracted.pgen ... done.\n",
      "--export Av to APOC3_topmed_rsq03_maf0005_extracted.traw ... 13202633404653606673808693done.\n",
      "End time: Tue Mar  5 11:51:30 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq08_maf0005_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq08_maf0005_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_hrc_rsq08_maf0005_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq08_maf0005_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:30 2024\n",
      "257483 MiB RAM detected, ~144851 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq08_maf0005_LOF_missense_extracted.fam.\n",
      "5315 variants loaded from\n",
      "../hrc/hrc_chr11_rsq08_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 2 variants remaining.\n",
      "2 variants remaining after main filters.\n",
      "Writing APOC3_hrc_rsq08_maf0005_extracted.bim ... done.\n",
      "Writing APOC3_hrc_rsq08_maf0005_extracted.fam ... done.\n",
      "Writing APOC3_hrc_rsq08_maf0005_extracted.pgen ... done.\n",
      "--export Av to APOC3_hrc_rsq08_maf0005_extracted.traw ... 50done.\n",
      "End time: Tue Mar  5 11:51:31 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_rsq08_maf0005_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr11_rsq08_maf0005_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_topmed_rsq08_maf0005_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_rsq08_maf0005_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:31 2024\n",
      "257483 MiB RAM detected, ~144503 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr11_rsq08_maf0005_LOF_missense_extracted.fam.\n",
      "28110 variants loaded from\n",
      "../topmed/topmed_chr11_rsq08_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 5 variants remaining.\n",
      "5 variants remaining after main filters.\n",
      "Writing APOC3_topmed_rsq08_maf0005_extracted.bim ... done.\n",
      "Writing APOC3_topmed_rsq08_maf0005_extracted.fam ... done.\n",
      "Writing APOC3_topmed_rsq08_maf0005_extracted.pgen ... done.\n",
      "--export Av to APOC3_topmed_rsq08_maf0005_extracted.traw ... 20406080done.\n",
      "End time: Tue Mar  5 11:51:31 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_exome_maf0001_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c11_maf0001_LOF_missense_extracted\n",
      "  --extract APOC3_exome_maf0001_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out APOC3_exome_maf0001_extracted\n",
      "\n",
      "Start time: Tue Mar  5 11:51:31 2024\n",
      "257483 MiB RAM detected, ~144020 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c11_maf0001_LOF_missense_extracted.fam.\n",
      "291093 variants loaded from\n",
      "../exome/ukb23156_c11_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 59 variants remaining.\n",
      "59 variants remaining after main filters.\n",
      "Writing APOC3_exome_maf0001_extracted.fam ... done.\n",
      "Writing APOC3_exome_maf0001_extracted.bim ... done.\n",
      "Writing APOC3_exome_maf0001_extracted.bed ... done.\n",
      "End time: Tue Mar  5 11:51:31 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq03_maf0001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq03_maf0001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_hrc_rsq03_maf0001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq03_maf0001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:31 2024\n",
      "257483 MiB RAM detected, ~143465 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq03_maf0001_LOF_missense_extracted.fam.\n",
      "10872 variants loaded from\n",
      "../hrc/hrc_chr11_rsq03_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 2 variants remaining.\n",
      "2 variants remaining after main filters.\n",
      "Writing APOC3_hrc_rsq03_maf0001_extracted.bim ... done.\n",
      "Writing APOC3_hrc_rsq03_maf0001_extracted.fam ... done.\n",
      "Writing APOC3_hrc_rsq03_maf0001_extracted.pgen ... done.\n",
      "--export Av to APOC3_hrc_rsq03_maf0001_extracted.traw ... 50done.\n",
      "End time: Tue Mar  5 11:51:31 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_rsq03_maf0001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr11_rsq03_maf0001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_topmed_rsq03_maf0001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_rsq03_maf0001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:31 2024\n",
      "257483 MiB RAM detected, ~142991 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr11_rsq03_maf0001_LOF_missense_extracted.fam.\n",
      "74968 variants loaded from\n",
      "../topmed/topmed_chr11_rsq03_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 14 variants remaining.\n",
      "14 variants remaining after main filters.\n",
      "Writing APOC3_topmed_rsq03_maf0001_extracted.bim ... done.\n",
      "Writing APOC3_topmed_rsq03_maf0001_extracted.fam ... done.\n",
      "Writing APOC3_topmed_rsq03_maf0001_extracted.pgen ... done.\n",
      "--export Av to APOC3_topmed_rsq03_maf0001_extracted.traw ... 142128354250576471788592done.\n",
      "End time: Tue Mar  5 11:51:31 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq08_maf0001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq08_maf0001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_hrc_rsq08_maf0001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq08_maf0001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:31 2024\n",
      "257483 MiB RAM detected, ~142476 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq08_maf0001_LOF_missense_extracted.fam.\n",
      "3788 variants loaded from\n",
      "../hrc/hrc_chr11_rsq08_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 0 variants remaining.\n",
      "Error: No variants remaining after main filters.\n",
      "End time: Tue Mar  5 11:51:31 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_rsq08_maf0001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr11_rsq08_maf0001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_topmed_rsq08_maf0001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_rsq08_maf0001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:31 2024\n",
      "257483 MiB RAM detected, ~142260 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr11_rsq08_maf0001_LOF_missense_extracted.fam.\n",
      "26335 variants loaded from\n",
      "../topmed/topmed_chr11_rsq08_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 4 variants remaining.\n",
      "4 variants remaining after main filters.\n",
      "Writing APOC3_topmed_rsq08_maf0001_extracted.bim ... done.\n",
      "Writing APOC3_topmed_rsq08_maf0001_extracted.fam ... done.\n",
      "Writing APOC3_topmed_rsq08_maf0001_extracted.pgen ... done.\n",
      "--export Av to APOC3_topmed_rsq08_maf0001_extracted.traw ... 255075done.\n",
      "End time: Tue Mar  5 11:51:32 2024\n"
     ]
    }
   ],
   "source": [
    "cd ~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/apoc3\n",
    "\n",
    "for i in 1 05 01; do\n",
    "    plink2 \\\n",
    "        --bfile ../exome/ukb23156_c11_maf00${i}_LOF_missense_extracted \\\n",
    "        --extract APOC3_exome_maf00${i}_extractlist_snplist \\\n",
    "        --make-bed \\\n",
    "        --out APOC3_exome_maf00${i}_extracted\n",
    "\n",
    "    for j in 3 8; do\n",
    "        plink2 \\\n",
    "            --bpfile ../hrc/hrc_chr11_rsq0${j}_maf00${i}_LOF_missense_extracted \\\n",
    "            --extract APOC3_hrc_rsq0${j}_maf00${i}_extractlist_snplist \\\n",
    "            --make-bpgen --sort-vars \\\n",
    "            --export A-transpose \\\n",
    "            --out APOC3_hrc_rsq0${j}_maf00${i}_extracted\n",
    "\n",
    "        plink2 \\\n",
    "            --bpfile ../topmed/topmed_chr11_rsq0${j}_maf00${i}_LOF_missense_extracted \\\n",
    "            --extract APOC3_topmed_rsq0${j}_maf00${i}_extractlist_snplist \\\n",
    "            --make-bpgen --sort-vars \\\n",
    "            --export A-transpose \\\n",
    "            --out APOC3_topmed_rsq0${j}_maf00${i}_extracted\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "db049040-a07b-457a-be12-0053daa19015",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_exome_maf001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c11_maf001_LOF_missense_extracted\n",
      "  --extract APOC3_exome_maf001_cadd_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out APOC3_exome_maf001_cadd_extracted\n",
      "\n",
      "Start time: Tue Mar  5 11:51:33 2024\n",
      "257483 MiB RAM detected, ~138087 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c11_maf001_LOF_missense_extracted.fam.\n",
      "293784 variants loaded from\n",
      "../exome/ukb23156_c11_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 26 variants remaining.\n",
      "26 variants remaining after main filters.\n",
      "Writing APOC3_exome_maf001_cadd_extracted.fam ... done.\n",
      "Writing APOC3_exome_maf001_cadd_extracted.bim ... done.\n",
      "Writing APOC3_exome_maf001_cadd_extracted.bed ... done.\n",
      "End time: Tue Mar  5 11:51:33 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq03_maf001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq03_maf001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_hrc_rsq03_maf001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq03_maf001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:33 2024\n",
      "257483 MiB RAM detected, ~137589 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq03_maf001_LOF_missense_extracted.fam.\n",
      "13392 variants loaded from\n",
      "../hrc/hrc_chr11_rsq03_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 3 variants remaining.\n",
      "3 variants remaining after main filters.\n",
      "Writing APOC3_hrc_rsq03_maf001_cadd_extracted.bim ... done.\n",
      "Writing APOC3_hrc_rsq03_maf001_cadd_extracted.fam ... done.\n",
      "Writing APOC3_hrc_rsq03_maf001_cadd_extracted.pgen ... done.\n",
      "--export Av to APOC3_hrc_rsq03_maf001_cadd_extracted.traw ... 3366done.\n",
      "End time: Tue Mar  5 11:51:33 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_rsq03_maf001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr11_rsq03_maf001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_topmed_rsq03_maf001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_rsq03_maf001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:33 2024\n",
      "257483 MiB RAM detected, ~137044 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr11_rsq03_maf001_LOF_missense_extracted.fam.\n",
      "77513 variants loaded from\n",
      "../topmed/topmed_chr11_rsq03_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 9 variants remaining.\n",
      "9 variants remaining after main filters.\n",
      "Writing APOC3_topmed_rsq03_maf001_cadd_extracted.bim ... done.\n",
      "Writing APOC3_topmed_rsq03_maf001_cadd_extracted.fam ... done.\n",
      "Writing APOC3_topmed_rsq03_maf001_cadd_extracted.pgen ... done.\n",
      "--export Av to APOC3_topmed_rsq03_maf001_cadd_extracted.traw ... 1122334455667788done.\n",
      "End time: Tue Mar  5 11:51:33 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq08_maf001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq08_maf001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_hrc_rsq08_maf001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq08_maf001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:33 2024\n",
      "257483 MiB RAM detected, ~136657 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq08_maf001_LOF_missense_extracted.fam.\n",
      "5811 variants loaded from\n",
      "../hrc/hrc_chr11_rsq08_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 2 variants remaining.\n",
      "2 variants remaining after main filters.\n",
      "Writing APOC3_hrc_rsq08_maf001_cadd_extracted.bim ... done.\n",
      "Writing APOC3_hrc_rsq08_maf001_cadd_extracted.fam ... done.\n",
      "Writing APOC3_hrc_rsq08_maf001_cadd_extracted.pgen ... done.\n",
      "--export Av to APOC3_hrc_rsq08_maf001_cadd_extracted.traw ... 50done.\n",
      "End time: Tue Mar  5 11:51:33 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_rsq08_maf001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr11_rsq08_maf001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_topmed_rsq08_maf001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_rsq08_maf001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:33 2024\n",
      "257483 MiB RAM detected, ~136418 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr11_rsq08_maf001_LOF_missense_extracted.fam.\n",
      "28631 variants loaded from\n",
      "../topmed/topmed_chr11_rsq08_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 4 variants remaining.\n",
      "4 variants remaining after main filters.\n",
      "Writing APOC3_topmed_rsq08_maf001_cadd_extracted.bim ... done.\n",
      "Writing APOC3_topmed_rsq08_maf001_cadd_extracted.fam ... done.\n",
      "Writing APOC3_topmed_rsq08_maf001_cadd_extracted.pgen ... done.\n",
      "--export Av to APOC3_topmed_rsq08_maf001_cadd_extracted.traw ... 255075done.\n",
      "End time: Tue Mar  5 11:51:33 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_exome_maf0005_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c11_maf0005_LOF_missense_extracted\n",
      "  --extract APOC3_exome_maf0005_cadd_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out APOC3_exome_maf0005_cadd_extracted\n",
      "\n",
      "Start time: Tue Mar  5 11:51:33 2024\n",
      "257483 MiB RAM detected, ~136083 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c11_maf0005_LOF_missense_extracted.fam.\n",
      "293221 variants loaded from\n",
      "../exome/ukb23156_c11_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 26 variants remaining.\n",
      "26 variants remaining after main filters.\n",
      "Writing APOC3_exome_maf0005_cadd_extracted.fam ... done.\n",
      "Writing APOC3_exome_maf0005_cadd_extracted.bim ... done.\n",
      "Writing APOC3_exome_maf0005_cadd_extracted.bed ... done.\n",
      "End time: Tue Mar  5 11:51:34 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq03_maf0005_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq03_maf0005_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_hrc_rsq03_maf0005_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq03_maf0005_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:34 2024\n",
      "257483 MiB RAM detected, ~135742 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq03_maf0005_LOF_missense_extracted.fam.\n",
      "12867 variants loaded from\n",
      "../hrc/hrc_chr11_rsq03_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 3 variants remaining.\n",
      "3 variants remaining after main filters.\n",
      "Writing APOC3_hrc_rsq03_maf0005_cadd_extracted.bim ... done.\n",
      "Writing APOC3_hrc_rsq03_maf0005_cadd_extracted.fam ... done.\n",
      "Writing APOC3_hrc_rsq03_maf0005_cadd_extracted.pgen ... done.\n",
      "--export Av to APOC3_hrc_rsq03_maf0005_cadd_extracted.traw ... 3366done.\n",
      "End time: Tue Mar  5 11:51:34 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_rsq03_maf0005_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr11_rsq03_maf0005_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_topmed_rsq03_maf0005_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_rsq03_maf0005_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:34 2024\n",
      "257483 MiB RAM detected, ~135464 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr11_rsq03_maf0005_LOF_missense_extracted.fam.\n",
      "76984 variants loaded from\n",
      "../topmed/topmed_chr11_rsq03_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 9 variants remaining.\n",
      "9 variants remaining after main filters.\n",
      "Writing APOC3_topmed_rsq03_maf0005_cadd_extracted.bim ... done.\n",
      "Writing APOC3_topmed_rsq03_maf0005_cadd_extracted.fam ... done.\n",
      "Writing APOC3_topmed_rsq03_maf0005_cadd_extracted.pgen ... done.\n",
      "--export Av to APOC3_topmed_rsq03_maf0005_cadd_extracted.traw ... 1122334455667788done.\n",
      "End time: Tue Mar  5 11:51:34 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq08_maf0005_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq08_maf0005_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_hrc_rsq08_maf0005_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq08_maf0005_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:34 2024\n",
      "257483 MiB RAM detected, ~135111 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq08_maf0005_LOF_missense_extracted.fam.\n",
      "5315 variants loaded from\n",
      "../hrc/hrc_chr11_rsq08_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 2 variants remaining.\n",
      "2 variants remaining after main filters.\n",
      "Writing APOC3_hrc_rsq08_maf0005_cadd_extracted.bim ... done.\n",
      "Writing APOC3_hrc_rsq08_maf0005_cadd_extracted.fam ... done.\n",
      "Writing APOC3_hrc_rsq08_maf0005_cadd_extracted.pgen ... done.\n",
      "--export Av to APOC3_hrc_rsq08_maf0005_cadd_extracted.traw ... 50done.\n",
      "End time: Tue Mar  5 11:51:34 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_rsq08_maf0005_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr11_rsq08_maf0005_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_topmed_rsq08_maf0005_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_rsq08_maf0005_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:34 2024\n",
      "257483 MiB RAM detected, ~134728 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr11_rsq08_maf0005_LOF_missense_extracted.fam.\n",
      "28110 variants loaded from\n",
      "../topmed/topmed_chr11_rsq08_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 4 variants remaining.\n",
      "4 variants remaining after main filters.\n",
      "Writing APOC3_topmed_rsq08_maf0005_cadd_extracted.bim ... done.\n",
      "Writing APOC3_topmed_rsq08_maf0005_cadd_extracted.fam ... done.\n",
      "Writing APOC3_topmed_rsq08_maf0005_cadd_extracted.pgen ... done.\n",
      "--export Av to APOC3_topmed_rsq08_maf0005_cadd_extracted.traw ... 255075done.\n",
      "End time: Tue Mar  5 11:51:34 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_exome_maf0001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c11_maf0001_LOF_missense_extracted\n",
      "  --extract APOC3_exome_maf0001_cadd_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out APOC3_exome_maf0001_cadd_extracted\n",
      "\n",
      "Start time: Tue Mar  5 11:51:34 2024\n",
      "257483 MiB RAM detected, ~134209 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c11_maf0001_LOF_missense_extracted.fam.\n",
      "291093 variants loaded from\n",
      "../exome/ukb23156_c11_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 25 variants remaining.\n",
      "25 variants remaining after main filters.\n",
      "Writing APOC3_exome_maf0001_cadd_extracted.fam ... done.\n",
      "Writing APOC3_exome_maf0001_cadd_extracted.bim ... done.\n",
      "Writing APOC3_exome_maf0001_cadd_extracted.bed ... done.\n",
      "End time: Tue Mar  5 11:51:34 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq03_maf0001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq03_maf0001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_hrc_rsq03_maf0001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq03_maf0001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:34 2024\n",
      "257483 MiB RAM detected, ~133839 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq03_maf0001_LOF_missense_extracted.fam.\n",
      "10872 variants loaded from\n",
      "../hrc/hrc_chr11_rsq03_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 2 variants remaining.\n",
      "2 variants remaining after main filters.\n",
      "Writing APOC3_hrc_rsq03_maf0001_cadd_extracted.bim ... done.\n",
      "Writing APOC3_hrc_rsq03_maf0001_cadd_extracted.fam ... done.\n",
      "Writing APOC3_hrc_rsq03_maf0001_cadd_extracted.pgen ... done.\n",
      "--export Av to APOC3_hrc_rsq03_maf0001_cadd_extracted.traw ... 50done.\n",
      "End time: Tue Mar  5 11:51:34 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_rsq03_maf0001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr11_rsq03_maf0001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_topmed_rsq03_maf0001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_rsq03_maf0001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:34 2024\n",
      "257483 MiB RAM detected, ~133629 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr11_rsq03_maf0001_LOF_missense_extracted.fam.\n",
      "74968 variants loaded from\n",
      "../topmed/topmed_chr11_rsq03_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 8 variants remaining.\n",
      "8 variants remaining after main filters.\n",
      "Writing APOC3_topmed_rsq03_maf0001_cadd_extracted.bim ... done.\n",
      "Writing APOC3_topmed_rsq03_maf0001_cadd_extracted.fam ... done.\n",
      "Writing APOC3_topmed_rsq03_maf0001_cadd_extracted.pgen ... done.\n",
      "--export Av to APOC3_topmed_rsq03_maf0001_cadd_extracted.traw ... 12253750627587done.\n",
      "End time: Tue Mar  5 11:51:34 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq08_maf0001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq08_maf0001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_hrc_rsq08_maf0001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq08_maf0001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:34 2024\n",
      "257483 MiB RAM detected, ~133301 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq08_maf0001_LOF_missense_extracted.fam.\n",
      "3788 variants loaded from\n",
      "../hrc/hrc_chr11_rsq08_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 0 variants remaining.\n",
      "Error: No variants remaining after main filters.\n",
      "End time: Tue Mar  5 11:51:34 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_rsq08_maf0001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed/topmed_chr11_rsq08_maf0001_LOF_missense_extracted\n",
      "  --export A-transpose\n",
      "  --extract APOC3_topmed_rsq08_maf0001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_rsq08_maf0001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Tue Mar  5 11:51:34 2024\n",
      "257483 MiB RAM detected, ~133184 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed/topmed_chr11_rsq08_maf0001_LOF_missense_extracted.fam.\n",
      "26335 variants loaded from\n",
      "../topmed/topmed_chr11_rsq08_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 3 variants remaining.\n",
      "3 variants remaining after main filters.\n",
      "Writing APOC3_topmed_rsq08_maf0001_cadd_extracted.bim ... done.\n",
      "Writing APOC3_topmed_rsq08_maf0001_cadd_extracted.fam ... done.\n",
      "Writing APOC3_topmed_rsq08_maf0001_cadd_extracted.pgen ... done.\n",
      "--export Av to APOC3_topmed_rsq08_maf0001_cadd_extracted.traw ... 3366done.\n",
      "End time: Tue Mar  5 11:51:35 2024\n"
     ]
    }
   ],
   "source": [
    "for i in 1 05 01; do\n",
    "    plink2 \\\n",
    "        --bfile ../exome/ukb23156_c11_maf00${i}_LOF_missense_extracted \\\n",
    "        --extract APOC3_exome_maf00${i}_cadd_extractlist_snplist \\\n",
    "        --make-bed \\\n",
    "        --out APOC3_exome_maf00${i}_cadd_extracted\n",
    "\n",
    "    for j in 3 8; do\n",
    "        plink2 \\\n",
    "            --bpfile ../hrc/hrc_chr11_rsq0${j}_maf00${i}_LOF_missense_extracted \\\n",
    "            --extract APOC3_hrc_rsq0${j}_maf00${i}_cadd_extractlist_snplist \\\n",
    "            --make-bpgen --sort-vars \\\n",
    "            --export A-transpose \\\n",
    "            --out APOC3_hrc_rsq0${j}_maf00${i}_cadd_extracted\n",
    "\n",
    "        plink2 \\\n",
    "            --bpfile ../topmed/topmed_chr11_rsq0${j}_maf00${i}_LOF_missense_extracted \\\n",
    "            --extract APOC3_topmed_rsq0${j}_maf00${i}_cadd_extractlist_snplist \\\n",
    "            --make-bpgen --sort-vars \\\n",
    "            --export A-transpose \\\n",
    "            --out APOC3_topmed_rsq0${j}_maf00${i}_cadd_extracted\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cac917a4-e325-48cb-8bd8-1e1a732199cc",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "## Make merged dataset"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "774d0c62-eaec-4012-b3f4-0ba48f98d2a4",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### PCSK9 - exome + topmed + hrc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "e5dbef7c-37d1-4335-ab60-d551ca60bb72",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "── \u001b[1mAttaching core tidyverse packages\u001b[22m ──────────────────────── tidyverse 2.0.0 ──\n",
      "\u001b[32m✔\u001b[39m \u001b[34mforcats  \u001b[39m 1.0.0     \u001b[32m✔\u001b[39m \u001b[34mreadr    \u001b[39m 2.1.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2  \u001b[39m 3.4.3     \u001b[32m✔\u001b[39m \u001b[34mstringr  \u001b[39m 1.5.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mlubridate\u001b[39m 1.9.2     \u001b[32m✔\u001b[39m \u001b[34mtibble   \u001b[39m 3.2.1\n",
      "\u001b[32m✔\u001b[39m \u001b[34mpurrr    \u001b[39m 1.0.1     \u001b[32m✔\u001b[39m \u001b[34mtidyr    \u001b[39m 1.3.0\n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdata.table\u001b[39m::\u001b[32mbetween()\u001b[39m masks \u001b[34mdplyr\u001b[39m::between()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m       masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdata.table\u001b[39m::\u001b[32mfirst()\u001b[39m   masks \u001b[34mdplyr\u001b[39m::first()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32mhour()\u001b[39m     masks \u001b[34mdata.table\u001b[39m::hour()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32misoweek()\u001b[39m  masks \u001b[34mdata.table\u001b[39m::isoweek()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m          masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdata.table\u001b[39m::\u001b[32mlast()\u001b[39m    masks \u001b[34mdplyr\u001b[39m::last()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32mmday()\u001b[39m     masks \u001b[34mdata.table\u001b[39m::mday()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32mminute()\u001b[39m   masks \u001b[34mdata.table\u001b[39m::minute()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32mmonth()\u001b[39m    masks \u001b[34mdata.table\u001b[39m::month()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32mquarter()\u001b[39m  masks \u001b[34mdata.table\u001b[39m::quarter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32msecond()\u001b[39m   masks \u001b[34mdata.table\u001b[39m::second()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mtranspose()\u001b[39m    masks \u001b[34mdata.table\u001b[39m::transpose()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32mwday()\u001b[39m     masks \u001b[34mdata.table\u001b[39m::wday()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32mweek()\u001b[39m     masks \u001b[34mdata.table\u001b[39m::week()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32myday()\u001b[39m     masks \u001b[34mdata.table\u001b[39m::yday()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32myear()\u001b[39m     masks \u001b[34mdata.table\u001b[39m::year()\n",
      "\u001b[36mℹ\u001b[39m Use the conflicted package (\u001b[3m\u001b[34m<http://conflicted.r-lib.org/>\u001b[39m\u001b[23m) to force all conflicts to become errors\n"
     ]
    }
   ],
   "source": [
    "setwd(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/pcsk9\")\n",
    "\n",
    "library(dplyr)\n",
    "library(genio)\n",
    "library(data.table)\n",
    "library(tidyverse)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "168cf009-8648-4ec3-be10-b6fe2502229f",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"==========RSQ 0.3; MAF 001; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: PCSK9_exome_maf001_extracted.bim\n",
      "\n",
      "Reading: PCSK9_exome_maf001_extracted.fam\n",
      "\n",
      "Reading: PCSK9_exome_maf001_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 2; TOPMed_snplist: 7; Exome: 394; Annot_gene: 403; Bed: 403\"\n",
      "[1] \"==========RSQ 0.3; MAF 001; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: PCSK9_exome_maf001_cadd_extracted.bim\n",
      "\n",
      "Reading: PCSK9_exome_maf001_cadd_extracted.fam\n",
      "\n",
      "Reading: PCSK9_exome_maf001_cadd_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 1; TOPMed_snplist: 1; Exome: 200; Annot_gene: 202; Bed: 202\"\n",
      "[1] \"==========RSQ 0.3; MAF 0005; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: PCSK9_exome_maf0005_extracted.bim\n",
      "\n",
      "Reading: PCSK9_exome_maf0005_extracted.fam\n",
      "\n",
      "Reading: PCSK9_exome_maf0005_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 2; TOPMed_snplist: 7; Exome: 394; Annot_gene: 403; Bed: 403\"\n",
      "[1] \"==========RSQ 0.3; MAF 0005; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: PCSK9_exome_maf0005_cadd_extracted.bim\n",
      "\n",
      "Reading: PCSK9_exome_maf0005_cadd_extracted.fam\n",
      "\n",
      "Reading: PCSK9_exome_maf0005_cadd_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 1; TOPMed_snplist: 1; Exome: 200; Annot_gene: 202; Bed: 202\"\n",
      "[1] \"==========RSQ 0.3; MAF 0001; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: PCSK9_exome_maf0001_extracted.bim\n",
      "\n",
      "Reading: PCSK9_exome_maf0001_extracted.fam\n",
      "\n",
      "Reading: PCSK9_exome_maf0001_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 2; TOPMed_snplist: 7; Exome: 393; Annot_gene: 402; Bed: 402\"\n",
      "[1] \"==========RSQ 0.3; MAF 0001; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: PCSK9_exome_maf0001_cadd_extracted.bim\n",
      "\n",
      "Reading: PCSK9_exome_maf0001_cadd_extracted.fam\n",
      "\n",
      "Reading: PCSK9_exome_maf0001_cadd_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 1; TOPMed_snplist: 1; Exome: 199; Annot_gene: 201; Bed: 201\"\n",
      "[1] \"==========RSQ 0.8; MAF 001; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: PCSK9_exome_maf001_extracted.bim\n",
      "\n",
      "Reading: PCSK9_exome_maf001_extracted.fam\n",
      "\n",
      "Reading: PCSK9_exome_maf001_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_snplist: 2; Exome: 394; Annot_gene: 396; Bed: 396\"\n",
      "[1] \"==========RSQ 0.8; MAF 001; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: PCSK9_exome_maf001_cadd_extracted.bim\n",
      "\n",
      "Reading: PCSK9_exome_maf001_cadd_extracted.fam\n",
      "\n",
      "Reading: PCSK9_exome_maf001_cadd_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_snplist: 1; Exome: 200; Annot_gene: 201; Bed: 201\"\n",
      "[1] \"==========RSQ 0.8; MAF 0005; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: PCSK9_exome_maf0005_extracted.bim\n",
      "\n",
      "Reading: PCSK9_exome_maf0005_extracted.fam\n",
      "\n",
      "Reading: PCSK9_exome_maf0005_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_snplist: 2; Exome: 394; Annot_gene: 396; Bed: 396\"\n",
      "[1] \"==========RSQ 0.8; MAF 0005; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: PCSK9_exome_maf0005_cadd_extracted.bim\n",
      "\n",
      "Reading: PCSK9_exome_maf0005_cadd_extracted.fam\n",
      "\n",
      "Reading: PCSK9_exome_maf0005_cadd_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_snplist: 1; Exome: 200; Annot_gene: 201; Bed: 201\"\n",
      "[1] \"==========RSQ 0.8; MAF 0001; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: PCSK9_exome_maf0001_extracted.bim\n",
      "\n",
      "Reading: PCSK9_exome_maf0001_extracted.fam\n",
      "\n",
      "Reading: PCSK9_exome_maf0001_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_snplist: 2; Exome: 393; Annot_gene: 395; Bed: 395\"\n",
      "[1] \"==========RSQ 0.8; MAF 0001; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: PCSK9_exome_maf0001_cadd_extracted.bim\n",
      "\n",
      "Reading: PCSK9_exome_maf0001_cadd_extracted.fam\n",
      "\n",
      "Reading: PCSK9_exome_maf0001_cadd_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_snplist: 1; Exome: 199; Annot_gene: 200; Bed: 200\"\n"
     ]
    }
   ],
   "source": [
    "for(rsq in c(3, 8)){\n",
    "    for(maf in c(\"1\", \"05\", \"01\")){\n",
    "        for(cadd in c(\"\", \"_cadd\")){\n",
    "            print(sprintf(\"==========RSQ 0.%d; MAF 00%s; CADD filtering:%s\", rsq, maf, cadd))\n",
    "            \n",
    "            ## exome matrix\n",
    "            fname = sprintf(\"PCSK9_exome_maf00%s%s_extracted\", maf, cadd)\n",
    "            bim_fname <- paste0(fname, '.bim')\n",
    "            fam_fname <- paste0(fname, '.fam')\n",
    "            bed_fname <- paste0(fname, '.bed')\n",
    "\n",
    "            bim <- read_bim(bim_fname)\n",
    "            fam <- read_fam(fam_fname)\n",
    "            bed <- read_bed(bed_fname, bim$id, fam$id)\n",
    "            bed <- bed[, order(colnames(bed))]\n",
    "            \n",
    "            annot_fname <- sprintf(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/hrc_topmed_exome/hrc_topmed_exome_168206ids_rsq0%d_maf00%s%s_annot.csv.gz\", \n",
    "                       rsq, maf, cadd)\n",
    "            annot_gene <- fread(annot_fname) %>% filter(Gene.refGene == \"PCSK9\")\n",
    "\n",
    "            hrc_snplist <- annot_gene %>% filter(source == \"hrc\") %>% pull(ID_hg19)\n",
    "            topmed_snplist <- annot_gene %>% filter(source == \"topmed\") %>% pull(ID)\n",
    "\n",
    "            hrc_fname <- sprintf(\"PCSK9_hrc_rsq0%d_maf00%s%s_extracted.traw\", rsq, maf, cadd)\n",
    "            if(file.exists(hrc_fname)){\n",
    "                hrc <- fread(hrc_fname)\n",
    "                hrc_names <- colnames(hrc)[7: dim(hrc)[2]] %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                colnames(hrc)[7:dim(hrc)[2]] <- hrc_names[,2]\n",
    "\n",
    "                hrc_gene <- hrc %>% filter(SNP %in% hrc_snplist)\n",
    "                hrc_gene <- hrc_gene %>% remove_rownames() %>% select(-c(\"CHR\", \"(C)M\", \"POS\", \"COUNTED\", \"ALT\", \"SNP\"))\n",
    "                hrc_gene <- 2 - hrc_gene %>% select(sort(colnames(hrc_gene)))\n",
    "            }\n",
    "            \n",
    "            topmed_fname <- sprintf(\"PCSK9_topmed_rsq0%d_maf00%s%s_extracted.traw\", rsq, maf, cadd)\n",
    "            if(file.exists(topmed_fname)){\n",
    "                topmed <- fread(topmed_fname)\n",
    "                topmed_names <- colnames(topmed)[7: dim(topmed)[2]] %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                colnames(topmed)[7:dim(topmed)[2]] <- topmed_names[,3]\n",
    "\n",
    "                topmed_gene <- topmed %>% filter(SNP %in% topmed_snplist)\n",
    "                topmed_gene <- topmed_gene %>% remove_rownames() %>% select(-c(\"CHR\", \"(C)M\", \"POS\", \"COUNTED\", \"ALT\", \"SNP\"))\n",
    "                topmed_gene <- 2 - topmed_gene %>% select(sort(colnames(topmed_gene)))\n",
    "            }\n",
    "            \n",
    "            if(exists(\"hrc_gene\") & exists(\"topmed_gene\")){\n",
    "                bed <- rbind(bed, hrc_gene, topmed_gene)\n",
    "                rm(hrc_gene)\n",
    "                rm(topmed_gene)\n",
    "            } else if (exists(\"hrc_gene\")) {\n",
    "                bed <- rbind(bed, hrc_gene)\n",
    "                rm(hrc_gene)\n",
    "            } else if (exists(\"topmed_gene\")){\n",
    "                bed <- rbind(bed, topmed_gene)\n",
    "                rm(topmed_gene)\n",
    "            }\n",
    "            fwrite(bed, sprintf(\"PCSK9_hrc_topmed_exome_rsq0%d_maf00%s%s_extracted.traw\", rsq, maf, cadd))\n",
    "            \n",
    "            print(sprintf(\"HRC_snplist: %d; TOPMed_snplist: %d; Exome: %d; Annot_gene: %d; Bed: %d\", \n",
    "                          length(hrc_snplist), length(topmed_snplist), nrow(bim), nrow(annot_gene), nrow(bed)))\n",
    "            rm(bed)\n",
    "        }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a06bc6c3-7631-42dd-9bca-7c7fdcfe841c",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### PCSK9 - topmed + hrc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "0b79f400-9b91-41ff-896f-165af12cb0ce",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"==========RSQ 0.3; MAF 001; CADD filtering:\"\n",
      "[1] \"HRC_snplist: 6; TOPMed_snplist: 104; Annot_gene: 110; Bed: 110\"\n",
      "[1] \"==========RSQ 0.3; MAF 001; CADD filtering:_cadd\"\n",
      "[1] \"HRC_snplist: 2; TOPMed_snplist: 60; Annot_gene: 62; Bed: 62\"\n",
      "[1] \"==========RSQ 0.3; MAF 0005; CADD filtering:\"\n",
      "[1] \"HRC_snplist: 6; TOPMed_snplist: 104; Annot_gene: 110; Bed: 110\"\n",
      "[1] \"==========RSQ 0.3; MAF 0005; CADD filtering:_cadd\"\n",
      "[1] \"HRC_snplist: 2; TOPMed_snplist: 60; Annot_gene: 62; Bed: 62\"\n",
      "[1] \"==========RSQ 0.3; MAF 0001; CADD filtering:\"\n",
      "[1] \"HRC_snplist: 6; TOPMed_snplist: 103; Annot_gene: 109; Bed: 109\"\n",
      "[1] \"==========RSQ 0.3; MAF 0001; CADD filtering:_cadd\"\n",
      "[1] \"HRC_snplist: 2; TOPMed_snplist: 59; Annot_gene: 61; Bed: 61\"\n",
      "[1] \"==========RSQ 0.8; MAF 001; CADD filtering:\"\n",
      "[1] \"HRC_snplist: 3; TOPMed_snplist: 45; Annot_gene: 48; Bed: 48\"\n",
      "[1] \"==========RSQ 0.8; MAF 001; CADD filtering:_cadd\"\n",
      "[1] \"HRC_snplist: 1; TOPMed_snplist: 25; Annot_gene: 26; Bed: 26\"\n",
      "[1] \"==========RSQ 0.8; MAF 0005; CADD filtering:\"\n",
      "[1] \"HRC_snplist: 3; TOPMed_snplist: 45; Annot_gene: 48; Bed: 48\"\n",
      "[1] \"==========RSQ 0.8; MAF 0005; CADD filtering:_cadd\"\n",
      "[1] \"HRC_snplist: 1; TOPMed_snplist: 25; Annot_gene: 26; Bed: 26\"\n",
      "[1] \"==========RSQ 0.8; MAF 0001; CADD filtering:\"\n",
      "[1] \"HRC_snplist: 3; TOPMed_snplist: 44; Annot_gene: 47; Bed: 47\"\n",
      "[1] \"==========RSQ 0.8; MAF 0001; CADD filtering:_cadd\"\n",
      "[1] \"HRC_snplist: 1; TOPMed_snplist: 24; Annot_gene: 25; Bed: 25\"\n"
     ]
    }
   ],
   "source": [
    "for(rsq in c(3, 8)){\n",
    "    for(maf in c(\"1\", \"05\", \"01\")){\n",
    "        for(cadd in c(\"\", \"_cadd\")){\n",
    "            print(sprintf(\"==========RSQ 0.%d; MAF 00%s; CADD filtering:%s\", rsq, maf, cadd))\n",
    "            annot_fname <- sprintf(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/hrc_topmed/hrc_topmed_168206ids_rsq0%d_maf00%s%s_annot.csv.gz\", \n",
    "                                   rsq, maf, cadd)\n",
    "            annot_gene <- fread(annot_fname) %>% filter(Gene.refGene == \"PCSK9\")\n",
    "\n",
    "            hrc_snplist <- annot_gene %>% filter(source == \"hrc\") %>% pull(ID_hg19)\n",
    "            topmed_snplist <- annot_gene %>% filter(source == \"topmed\") %>% pull(ID)\n",
    "\n",
    "            hrc_fname <- sprintf(\"PCSK9_hrc_rsq0%d_maf00%s%s_extracted.traw\", rsq, maf, cadd)\n",
    "            hrc <- fread(hrc_fname)\n",
    "            hrc_names <- colnames(hrc)[7: dim(hrc)[2]] %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "            colnames(hrc)[7:dim(hrc)[2]] <- hrc_names[,2]\n",
    "\n",
    "            hrc_gene <- hrc %>% filter(SNP %in% hrc_snplist)\n",
    "            hrc_gene <- hrc_gene %>% remove_rownames() %>% select(-c(\"CHR\", \"(C)M\", \"POS\", \"COUNTED\", \"ALT\", \"SNP\"))\n",
    "            hrc_gene <- 2 - hrc_gene %>% select(sort(colnames(hrc_gene)))\n",
    "\n",
    "            topmed_fname <- sprintf(\"PCSK9_topmed_rsq0%d_maf00%s%s_extracted.traw\", rsq, maf, cadd)\n",
    "            topmed <- fread(topmed_fname)\n",
    "            topmed_names <- colnames(topmed)[7: dim(topmed)[2]] %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "            colnames(topmed)[7:dim(topmed)[2]] <- topmed_names[,3]\n",
    "\n",
    "            topmed_gene <- topmed %>% filter(SNP %in% topmed_snplist)\n",
    "            topmed_gene <- topmed_gene %>% remove_rownames() %>% select(-c(\"CHR\", \"(C)M\", \"POS\", \"COUNTED\", \"ALT\", \"SNP\"))\n",
    "            topmed_gene <- 2 - topmed_gene %>% select(sort(colnames(topmed_gene)))\n",
    "\n",
    "            if(exists(\"hrc_gene\") & exists(\"topmed_gene\")){\n",
    "                bed <- rbind(hrc_gene, topmed_gene)\n",
    "            } else if (exists(\"hrc_gene\")) {\n",
    "                bed <- hrc_gene\n",
    "            } else {\n",
    "                bed <- topmed_gene\n",
    "            }\n",
    "            # fwrite(bed, sprintf(\"PCSK9_hrc_topmed_rsq0%d_maf00%s%s_extracted.traw\", rsq, maf, cadd))\n",
    "            \n",
    "            print(sprintf(\"HRC_snplist: %d; TOPMed_snplist: %d; Annot_gene: %d; Bed: %d\", \n",
    "                          length(hrc_snplist), length(topmed_snplist), nrow(annot_gene), nrow(bed)))\n",
    "        } \n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d146c8e6-b23e-41f4-b5c2-5a731dd36551",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### APOC3 - exome + topmed + hrc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "ae36feaa-7fed-4f86-a37d-c0c4930e6e51",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "setwd(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/apoc3\")\n",
    "\n",
    "library(dplyr)\n",
    "library(genio)\n",
    "library(data.table)\n",
    "library(tidyverse)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "8d487fb1-06a9-4522-b519-f2449a49ed3f",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"==========RSQ 0.3; MAF 001; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: APOC3_exome_maf001_extracted.bim\n",
      "\n",
      "Reading: APOC3_exome_maf001_extracted.fam\n",
      "\n",
      "Reading: APOC3_exome_maf001_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_snplist: 3; Exome: 60; Annot_gene: 63; Bed: 63\"\n",
      "[1] \"==========RSQ 0.3; MAF 001; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: APOC3_exome_maf001_cadd_extracted.bim\n",
      "\n",
      "Reading: APOC3_exome_maf001_cadd_extracted.fam\n",
      "\n",
      "Reading: APOC3_exome_maf001_cadd_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_snplist: 0; Exome: 26; Annot_gene: 26; Bed: 26\"\n",
      "[1] \"==========RSQ 0.3; MAF 0005; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: APOC3_exome_maf0005_extracted.bim\n",
      "\n",
      "Reading: APOC3_exome_maf0005_extracted.fam\n",
      "\n",
      "Reading: APOC3_exome_maf0005_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_snplist: 3; Exome: 60; Annot_gene: 63; Bed: 63\"\n",
      "[1] \"==========RSQ 0.3; MAF 0005; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: APOC3_exome_maf0005_cadd_extracted.bim\n",
      "\n",
      "Reading: APOC3_exome_maf0005_cadd_extracted.fam\n",
      "\n",
      "Reading: APOC3_exome_maf0005_cadd_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_snplist: 0; Exome: 26; Annot_gene: 26; Bed: 26\"\n",
      "[1] \"==========RSQ 0.3; MAF 0001; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: APOC3_exome_maf0001_extracted.bim\n",
      "\n",
      "Reading: APOC3_exome_maf0001_extracted.fam\n",
      "\n",
      "Reading: APOC3_exome_maf0001_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_snplist: 3; Exome: 59; Annot_gene: 62; Bed: 62\"\n",
      "[1] \"==========RSQ 0.3; MAF 0001; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: APOC3_exome_maf0001_cadd_extracted.bim\n",
      "\n",
      "Reading: APOC3_exome_maf0001_cadd_extracted.fam\n",
      "\n",
      "Reading: APOC3_exome_maf0001_cadd_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_snplist: 0; Exome: 25; Annot_gene: 25; Bed: 25\"\n",
      "[1] \"==========RSQ 0.8; MAF 001; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: APOC3_exome_maf001_extracted.bim\n",
      "\n",
      "Reading: APOC3_exome_maf001_extracted.fam\n",
      "\n",
      "Reading: APOC3_exome_maf001_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_snplist: 0; Exome: 60; Annot_gene: 60; Bed: 60\"\n",
      "[1] \"==========RSQ 0.8; MAF 001; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: APOC3_exome_maf001_cadd_extracted.bim\n",
      "\n",
      "Reading: APOC3_exome_maf001_cadd_extracted.fam\n",
      "\n",
      "Reading: APOC3_exome_maf001_cadd_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_snplist: 0; Exome: 26; Annot_gene: 26; Bed: 26\"\n",
      "[1] \"==========RSQ 0.8; MAF 0005; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: APOC3_exome_maf0005_extracted.bim\n",
      "\n",
      "Reading: APOC3_exome_maf0005_extracted.fam\n",
      "\n",
      "Reading: APOC3_exome_maf0005_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_snplist: 0; Exome: 60; Annot_gene: 60; Bed: 60\"\n",
      "[1] \"==========RSQ 0.8; MAF 0005; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: APOC3_exome_maf0005_cadd_extracted.bim\n",
      "\n",
      "Reading: APOC3_exome_maf0005_cadd_extracted.fam\n",
      "\n",
      "Reading: APOC3_exome_maf0005_cadd_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_snplist: 0; Exome: 26; Annot_gene: 26; Bed: 26\"\n",
      "[1] \"==========RSQ 0.8; MAF 0001; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: APOC3_exome_maf0001_extracted.bim\n",
      "\n",
      "Reading: APOC3_exome_maf0001_extracted.fam\n",
      "\n",
      "Reading: APOC3_exome_maf0001_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_snplist: 0; Exome: 59; Annot_gene: 59; Bed: 59\"\n",
      "[1] \"==========RSQ 0.8; MAF 0001; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: APOC3_exome_maf0001_cadd_extracted.bim\n",
      "\n",
      "Reading: APOC3_exome_maf0001_cadd_extracted.fam\n",
      "\n",
      "Reading: APOC3_exome_maf0001_cadd_extracted.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_snplist: 0; Exome: 25; Annot_gene: 25; Bed: 25\"\n"
     ]
    }
   ],
   "source": [
    "for(rsq in c(3, 8)){\n",
    "    for(maf in c(\"1\", \"05\", \"01\")){\n",
    "        for(cadd in c(\"\", \"_cadd\")){\n",
    "            print(sprintf(\"==========RSQ 0.%d; MAF 00%s; CADD filtering:%s\", rsq, maf, cadd))\n",
    "            \n",
    "            ## exome matrix\n",
    "            fname = sprintf(\"APOC3_exome_maf00%s%s_extracted\", maf, cadd)\n",
    "            bim_fname <- paste0(fname, '.bim')\n",
    "            fam_fname <- paste0(fname, '.fam')\n",
    "            bed_fname <- paste0(fname, '.bed')\n",
    "\n",
    "            bim <- read_bim(bim_fname)\n",
    "            fam <- read_fam(fam_fname)\n",
    "            bed <- read_bed(bed_fname, bim$id, fam$id)\n",
    "            bed <- bed[, order(colnames(bed))]\n",
    "            \n",
    "            annot_fname <- sprintf(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/hrc_topmed_exome/hrc_topmed_exome_168206ids_chr11_rsq0%d_maf00%s%s_annot.csv.gz\", \n",
    "                       rsq, maf, cadd)\n",
    "            annot_gene <- fread(annot_fname) %>% filter(Gene.refGene == \"APOC3\")\n",
    "\n",
    "            hrc_snplist <- annot_gene %>% filter(source == \"hrc\") %>% pull(ID_hg19)\n",
    "            topmed_snplist <- annot_gene %>% filter(source == \"topmed\") %>% pull(ID)\n",
    "\n",
    "            hrc_fname <- sprintf(\"APOC3_hrc_rsq0%d_maf00%s%s_extracted.traw\", rsq, maf, cadd)\n",
    "            if(file.exists(hrc_fname)){\n",
    "                hrc <- fread(hrc_fname)\n",
    "                hrc_names <- colnames(hrc)[7: dim(hrc)[2]] %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                colnames(hrc)[7:dim(hrc)[2]] <- hrc_names[,2]\n",
    "\n",
    "                hrc_gene <- hrc %>% filter(SNP %in% hrc_snplist)\n",
    "                hrc_gene <- hrc_gene %>% remove_rownames() %>% select(-c(\"CHR\", \"(C)M\", \"POS\", \"COUNTED\", \"ALT\", \"SNP\"))\n",
    "                hrc_gene <- 2 - hrc_gene %>% select(sort(colnames(hrc_gene)))\n",
    "            }\n",
    "            \n",
    "            topmed_fname <- sprintf(\"APOC3_topmed_rsq0%d_maf00%s%s_extracted.traw\", rsq, maf, cadd)\n",
    "            if(file.exists(topmed_fname)){\n",
    "                topmed_fname <- sprintf(\"APOC3_topmed_rsq0%d_maf00%s%s_extracted.traw\", rsq, maf, cadd)\n",
    "                topmed <- fread(topmed_fname)\n",
    "                topmed_names <- colnames(topmed)[7: dim(topmed)[2]] %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                colnames(topmed)[7:dim(topmed)[2]] <- topmed_names[,3]\n",
    "\n",
    "                topmed_gene <- topmed %>% filter(SNP %in% topmed_snplist)\n",
    "                topmed_gene <- topmed_gene %>% remove_rownames() %>% select(-c(\"CHR\", \"(C)M\", \"POS\", \"COUNTED\", \"ALT\", \"SNP\"))\n",
    "                topmed_gene <- 2 - topmed_gene %>% select(sort(colnames(topmed_gene)))\n",
    "            }\n",
    "            \n",
    "            if(exists(\"hrc_gene\") & exists(\"topmed_gene\")){\n",
    "                bed <- rbind(bed, hrc_gene, topmed_gene)\n",
    "                rm(hrc_gene)\n",
    "                rm(topmed_gene)\n",
    "            } else if (exists(\"hrc_gene\")) {\n",
    "                bed <- rbind(bed, hrc_gene)\n",
    "                rm(hrc_gene)\n",
    "            } else if (exists(\"topmed_gene\")){\n",
    "                bed <- rbind(bed, topmed_gene)\n",
    "                rm(topmed_gene)\n",
    "            }\n",
    "            # fwrite(bed, sprintf(\"APOC3_hrc_topmed_exome_rsq0%d_maf00%s%s_extracted.traw\", rsq, maf, cadd))\n",
    "            \n",
    "            print(sprintf(\"HRC_snplist: %d; TOPMed_snplist: %d; Exome: %d; Annot_gene: %d; Bed: %d\", \n",
    "                          length(hrc_snplist), length(topmed_snplist), nrow(bim), nrow(annot_gene), nrow(bed)))\n",
    "            rm(bed)\n",
    "        }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a1eb2f3e-f1c6-4f10-9f15-a42649632828",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### APOC3 - topmed + hrc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "e05dd335-c1e4-4b15-9cd2-3bd2b95cbc3d",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"==========RSQ 0.3; MAF 001; CADD filtering: \"\n",
      "[1] \"HRC_snplist: 1; TOPMed_snplist: 14; Annot_gene: 15; Bed: 15\"\n",
      "[1] \"==========RSQ 0.3; MAF 001; CADD filtering: _cadd\"\n",
      "[1] \"HRC_snplist: 1; TOPMed_snplist: 8; Annot_gene: 9; Bed: 9\"\n",
      "[1] \"==========RSQ 0.3; MAF 0005; CADD filtering: \"\n",
      "[1] \"HRC_snplist: 1; TOPMed_snplist: 14; Annot_gene: 15; Bed: 15\"\n",
      "[1] \"==========RSQ 0.3; MAF 0005; CADD filtering: _cadd\"\n",
      "[1] \"HRC_snplist: 1; TOPMed_snplist: 8; Annot_gene: 9; Bed: 9\"\n",
      "[1] \"==========RSQ 0.3; MAF 0001; CADD filtering: \"\n",
      "[1] \"HRC_snplist: 1; TOPMed_snplist: 13; Annot_gene: 14; Bed: 14\"\n",
      "[1] \"==========RSQ 0.3; MAF 0001; CADD filtering: _cadd\"\n",
      "[1] \"HRC_snplist: 1; TOPMed_snplist: 7; Annot_gene: 8; Bed: 8\"\n",
      "[1] \"==========RSQ 0.8; MAF 001; CADD filtering: \"\n",
      "[1] \"HRC_snplist: 1; TOPMed_snplist: 5; Annot_gene: 6; Bed: 6\"\n",
      "[1] \"==========RSQ 0.8; MAF 001; CADD filtering: _cadd\"\n",
      "[1] \"HRC_snplist: 1; TOPMed_snplist: 4; Annot_gene: 5; Bed: 5\"\n",
      "[1] \"==========RSQ 0.8; MAF 0005; CADD filtering: \"\n",
      "[1] \"HRC_snplist: 1; TOPMed_snplist: 5; Annot_gene: 6; Bed: 6\"\n",
      "[1] \"==========RSQ 0.8; MAF 0005; CADD filtering: _cadd\"\n",
      "[1] \"HRC_snplist: 1; TOPMed_snplist: 4; Annot_gene: 5; Bed: 5\"\n",
      "[1] \"==========RSQ 0.8; MAF 0001; CADD filtering: \"\n",
      "[1] \"HRC_snplist: 0; TOPMed_snplist: 4; Annot_gene: 4; Bed: 4\"\n",
      "[1] \"==========RSQ 0.8; MAF 0001; CADD filtering: _cadd\"\n",
      "[1] \"HRC_snplist: 0; TOPMed_snplist: 3; Annot_gene: 3; Bed: 3\"\n"
     ]
    }
   ],
   "source": [
    "for(rsq in c(3, 8)){\n",
    "    for(maf in c(\"1\", \"05\", \"01\")){\n",
    "        for(cadd in c(\"\", \"_cadd\")){\n",
    "            print(sprintf(\"==========RSQ 0.%d; MAF 00%s; CADD filtering: %s\", rsq, maf, cadd))\n",
    "            annot_fname <- sprintf(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/hrc_topmed/hrc_topmed_168206ids_chr11_rsq0%d_maf00%s%s_annot.csv.gz\", \n",
    "                                   rsq, maf, cadd)\n",
    "            annot_gene <- fread(annot_fname) %>% filter(Gene.refGene == \"APOC3\")\n",
    "\n",
    "            hrc_snplist <- annot_gene %>% filter(source == \"hrc\") %>% pull(ID_hg19)\n",
    "            topmed_snplist <- annot_gene %>% filter(source == \"topmed\") %>% pull(ID)\n",
    "\n",
    "            hrc_fname <- sprintf(\"APOC3_hrc_rsq0%d_maf00%s%s_extracted.traw\", rsq, maf, cadd)\n",
    "            if(file.exists(hrc_fname)){\n",
    "                hrc <- fread(hrc_fname)\n",
    "                hrc_names <- colnames(hrc)[7: dim(hrc)[2]] %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                colnames(hrc)[7:dim(hrc)[2]] <- hrc_names[,2]\n",
    "\n",
    "                hrc_gene <- hrc %>% filter(SNP %in% hrc_snplist)\n",
    "                hrc_gene <- hrc_gene %>% remove_rownames() %>% select(-c(\"CHR\", \"(C)M\", \"POS\", \"COUNTED\", \"ALT\", \"SNP\"))\n",
    "                hrc_gene <- 2 - hrc_gene %>% select(sort(colnames(hrc_gene)))\n",
    "            }\n",
    "\n",
    "            topmed_fname <- sprintf(\"APOC3_topmed_rsq0%d_maf00%s%s_extracted.traw\", rsq, maf, cadd)\n",
    "            if(file.exists(topmed_fname)){\n",
    "                topmed <- fread(topmed_fname)\n",
    "                topmed_names <- colnames(topmed)[7: dim(topmed)[2]] %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                colnames(topmed)[7:dim(topmed)[2]] <- topmed_names[,3]\n",
    "\n",
    "                topmed_gene <- topmed %>% filter(SNP %in% topmed_snplist)\n",
    "                topmed_gene <- topmed_gene %>% remove_rownames() %>% select(-c(\"CHR\", \"(C)M\", \"POS\", \"COUNTED\", \"ALT\", \"SNP\"))\n",
    "                topmed_gene <- 2 - topmed_gene %>% select(sort(colnames(topmed_gene)))\n",
    "            }\n",
    "            \n",
    "            if(exists(\"hrc_gene\") & exists(\"topmed_gene\")){\n",
    "                bed <- rbind(hrc_gene, topmed_gene)\n",
    "                rm(hrc_gene)\n",
    "                rm(topmed_gene)\n",
    "            } else if (exists(\"hrc_gene\")) {\n",
    "                bed <- hrc_gene\n",
    "                rm(hrc_gene)\n",
    "            } else {\n",
    "                bed <- topmed_gene\n",
    "                rm(topmed_gene)\n",
    "            }\n",
    "            \n",
    "            if(exists(\"bed\")){\n",
    "                # fwrite(bed, sprintf(\"APOC3_hrc_topmed_rsq0%d_maf00%s%s_extracted.traw\", rsq, maf, cadd))\n",
    "            }\n",
    "            \n",
    "            print(sprintf(\"HRC_snplist: %d; TOPMed_snplist: %d; Annot_gene: %d; Bed: %d\", \n",
    "                          length(hrc_snplist), length(topmed_snplist), nrow(annot_gene), nrow(bed)))\n",
    "            rm(bed)\n",
    "        }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4015f470-c66f-42c5-a48f-76e0bb314495",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "## Phenotype Analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "327b6ed1-d5c7-46c6-a7d7-dd267acb9869",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "We are going to be using Burden of Rare Variant ([BRV](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4459641/)) to analyze rare variant aggregate association for PCSK9 and APOC3. For phenotype analysis, we control for age, sex and 2 PC's."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b74829db-31be-4c55-b5b3-dabbac423253",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "### Functions and Packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "38b89472-3a0c-4605-9778-3069b2c8f3cd",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "library(dplyr)\n",
    "library(genio)\n",
    "library(data.table)\n",
    "library(tidyverse)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "3129d96d-bceb-407a-bdb9-c4a3b6c3bdf0",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "# mi_pheno <- read.csv(\"/home/tl3031/project/imputation-rvtest/analysis/imputation_aggregated_analysis/Unrealted_White_EU_Both_Exo_Impu_5209_MI_cases.phe.csv\")\n",
    "# mi_y <- mi_pheno %>% arrange(FID) %>% pull(MI)\n",
    "\n",
    "###########################################\n",
    "# trig_ldl_pheno <- read.csv(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/Trigly_LDL_Log10_168206ind.phe.csv\") %>% arrange(FID)\n",
    "# tg_y <- trig_ldl_pheno %>% arrange(FID) %>% pull(logarithm_base10_Trigli)\n",
    "# ldl_y <- trig_ldl_pheno  %>% arrange(FID) %>% pull(f.30780.0.0)\n",
    "\n",
    "###########################################\n",
    "ldl_df <- read.csv(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/UKBB_LDL_159759inds_4PCs.pheno\", sep = \"\\t\") %>% arrange(FID)\n",
    "ldl_y <- ldl_df %>% pull(f.30780.0.0)\n",
    "ldl_id <- ldl_df %>% pull(IID)\n",
    "ldl_cov <- ldl_df %>% select(sex, age, PC1, PC2)\n",
    "\n",
    "tg_df <- read.csv(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/UKBB_TG_159759inds_4PCs.pheno\", sep = \"\\t\") %>% arrange(FID)\n",
    "tg_y <- tg_df %>% pull(logarithm_base10_Trigli)\n",
    "tg_id <- tg_df %>% pull(IID)\n",
    "tg_cov <- tg_df %>% select(sex, age, PC1, PC2)\n",
    "\n",
    "rm(ldl_df, tg_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "a46975bb-1da7-4146-bc3c-69239154c6d4",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "# The BRV function\n",
    "BRV <- function(X_mat, y, cov, option = \"binomial\"){\n",
    "    X_mat <- as.matrix(X_mat)\n",
    "    y <- as.vector(y)\n",
    "\n",
    "    ## cout number of rare variants for each individual\n",
    "    if(ncol(X_mat) == 1){\n",
    "        X_mat_new = X_mat\n",
    "        print(head(X_mat_new))\n",
    "    } else {\n",
    "        X_mat_new  = rowSums(X_mat, na.rm=TRUE) %>% as.matrix()\n",
    "    }\n",
    "    \n",
    "    age <- cov$age\n",
    "    sex <- cov$sex\n",
    "    pc1 <- cov$PC1\n",
    "    pc2 <- cov$PC2\n",
    "\n",
    "    ## obtain p value by logistic regression\n",
    "    mat.fit <- glm(y ~ X_mat_new + age + sex + pc1 + pc2, family = option)\n",
    "    zstat <- coef(summary(mat.fit))[2, 3]\n",
    "    pval <- coef(summary(mat.fit))[2, 4]\n",
    "\n",
    "    return(list(zstat = zstat, pval = pval))\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "9143d131-dc72-4333-be2b-0001551cd063",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "## Formatting imputed data matrix\n",
    "format_traw <- function(df, id, option=\"merge\"){\n",
    "    if(option != \"merge\"){\n",
    "        X <- df %>% \n",
    "            select(-c(1:6)) %>% \n",
    "            t() %>% as.data.frame() %>% \n",
    "            tibble::rownames_to_column(\"FID_full\")\n",
    "        FID <- sapply(strsplit(X$FID_full,\"_\"), `[`, 2)\n",
    "        X <- X %>%\n",
    "            mutate(FID = FID) %>%\n",
    "            arrange(FID) %>% \n",
    "            filter(FID %in% id) %>%\n",
    "            select(-FID_full, -FID) %>% \n",
    "            as.matrix()\n",
    "        X <- 2 - X\n",
    "    } else {\n",
    "        X <- df %>% \n",
    "            t() %>% as.data.frame() %>% \n",
    "            arrange(as.numeric(V1)) %>% \n",
    "            filter(V1 %in% id) %>%\n",
    "            select(-V1) %>% \n",
    "            as.matrix()\n",
    "    }\n",
    "    \n",
    "    return(X)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ff62b4f2-83cd-4023-8421-9f5a4fcd3749",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### PCSK9 v.s. LDL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "b1a555bd-d389-4008-9825-cdb3a0d559fa",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: PCSK9_exome_maf001_extracted.bim\n",
      "\n",
      "Reading: PCSK9_exome_maf001_extracted.fam\n",
      "\n",
      "Reading: PCSK9_exome_maf001_extracted.bed\n",
      "\n",
      "Reading: PCSK9_exome_maf0005_extracted.bim\n",
      "\n",
      "Reading: PCSK9_exome_maf0005_extracted.fam\n",
      "\n",
      "Reading: PCSK9_exome_maf0005_extracted.bed\n",
      "\n",
      "Reading: PCSK9_exome_maf0001_extracted.bim\n",
      "\n",
      "Reading: PCSK9_exome_maf0001_extracted.fam\n",
      "\n",
      "Reading: PCSK9_exome_maf0001_extracted.bed\n",
      "\n"
     ]
    }
   ],
   "source": [
    "setwd(\"/mnt/vast/hpc/csg/tl3031/imputation-rvtest/analysis/imputation_aggregated_analysis/pcsk9\")\n",
    "\n",
    "result_df <- data.frame(data.frame(matrix(ncol = 5, nrow = 0)))\n",
    "colnames(result_df) <- c(\"data\", \"rsqs\", \"mafs\", \"num_var\", \"pvals\")\n",
    "\n",
    "for(maf in c(\"1\", \"05\", \"01\")){ \n",
    "    exome_bim <- read_bim(sprintf(\"PCSK9_exome_maf00%s_extracted.bim\", maf))\n",
    "    exome_fam <- read_fam(sprintf(\"PCSK9_exome_maf00%s_extracted.fam\", maf))\n",
    "    exome <- read_bed(sprintf(\"PCSK9_exome_maf00%s_extracted.bed\", maf), exome_bim$id, exome_fam$id)\n",
    "    exome_X <- exome %>% t() %>% as.data.frame() %>% tibble::rownames_to_column(\"FID\") %>% \n",
    "            mutate(FID = as.numeric(FID)) %>% arrange(FID) %>% filter(FID %in% ldl_id) %>% select(-FID) %>% as.matrix()\n",
    "    result_exome <- BRV(exome_X, ldl_y, ldl_cov, \"gaussian\")\n",
    "    \n",
    "    for(rsq in c(3, 8)){\n",
    "        hrc_fname <- sprintf(\"PCSK9_hrc_rsq0%d_maf00%s_extracted.traw\", rsq, maf)\n",
    "        topmed_fname <- sprintf(\"PCSK9_topmed_rsq0%d_maf00%s_extracted.traw\", rsq, maf)\n",
    "        hrc_topmed_fname <- sprintf(\"PCSK9_hrc_topmed_rsq0%d_maf00%s_extracted.traw\", rsq, maf)\n",
    "        hrc_topmed_exome_fname <- sprintf(\"PCSK9_hrc_topmed_exome_rsq0%d_maf00%s_extracted.traw\", rsq, maf)\n",
    "\n",
    "        hrc <- fread(hrc_fname)\n",
    "        hrc_X <- format_traw(hrc, ldl_id, \"hrc\")\n",
    "        result_hrc <- BRV(hrc_X, ldl_y, ldl_cov, \"gaussian\")\n",
    "\n",
    "        topmed <- fread(topmed_fname)\n",
    "        topmed_X <- format_traw(topmed, ldl_id, \"topmed\")\n",
    "        result_topmed <- BRV(topmed_X, ldl_y, ldl_cov, \"gaussian\")\n",
    "\n",
    "        hrc_topmed <- fread(hrc_topmed_fname)\n",
    "        hrc_topmed_X <- format_traw(hrc_topmed, ldl_id)\n",
    "        result_hrc_topmed <- BRV(hrc_topmed_X, ldl_y, ldl_cov, \"gaussian\")\n",
    "\n",
    "        hrc_topmed_exome <- fread(hrc_topmed_exome_fname)\n",
    "        hrc_topmed_exome_X <- format_traw(hrc_topmed_exome, ldl_id)\n",
    "        result_hrc_topmed_exome <- BRV(hrc_topmed_exome_X, ldl_y, ldl_cov, \"gaussian\")\n",
    "\n",
    "        data <- c(\"hrc\", \"topmed\", \"hrc_topmed\", \"hrc_topmed_exome\")\n",
    "        rsqs <- rep(rsq/10, 4)\n",
    "        mafs <- rep(paste0(\"0.0\", maf), 4)\n",
    "        pvals <- c(result_hrc$pval, result_topmed$pval, result_hrc_topmed$pval, result_hrc_topmed_exome$pval)\n",
    "        num_var <- c(ncol(hrc_X), ncol(topmed_X), ncol(hrc_topmed_X), ncol(hrc_topmed_exome_X))\n",
    "        sub_df <- data.frame(data, rsqs, mafs, num_var, pvals)\n",
    "        result_df <- rbind(result_df, sub_df)\n",
    "    }\n",
    "    result_df <- rbind(data.frame(data = \"exome\", rsqs = \"NA\", mafs = paste0(\"0.0\", maf), num_var = ncol(exome_X), pvals = result_exome$pval), result_df)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "cf4f5985-85db-4acb-a196-1a246a127bcb",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 27 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>data</th><th scope=col>rsqs</th><th scope=col>mafs</th><th scope=col>num_var</th><th scope=col>pvals</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.01 </td><td>394</td><td>5.473985e-71</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.01 </td><td> 16</td><td>1.944983e-12</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.01 </td><td>  6</td><td>6.993942e-11</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.01 </td><td>109</td><td>1.384075e-39</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.01 </td><td> 47</td><td>1.769128e-34</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.01 </td><td>110</td><td>1.121818e-38</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.01 </td><td> 48</td><td>1.972979e-34</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.01 </td><td>403</td><td>1.684064e-70</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.01 </td><td>396</td><td>7.159298e-71</td></tr>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.005</td><td>394</td><td>5.473985e-71</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.005</td><td> 16</td><td>1.944983e-12</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.005</td><td>  6</td><td>6.993942e-11</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.005</td><td>109</td><td>1.384075e-39</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.005</td><td> 47</td><td>1.769128e-34</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.005</td><td>110</td><td>1.121818e-38</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.005</td><td> 48</td><td>1.972979e-34</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.005</td><td>403</td><td>1.684064e-70</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.005</td><td>396</td><td>7.159298e-71</td></tr>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.001</td><td>393</td><td>9.347929e-53</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.001</td><td> 15</td><td>2.387199e-09</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.001</td><td>  6</td><td>6.993942e-11</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.001</td><td>108</td><td>1.040078e-27</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.001</td><td> 46</td><td>2.013611e-22</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.001</td><td>109</td><td>7.965593e-27</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.001</td><td> 47</td><td>2.473826e-22</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.001</td><td>402</td><td>2.584292e-52</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.001</td><td>395</td><td>1.200541e-52</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 27 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " data & rsqs & mafs & num\\_var & pvals\\\\\n",
       " <fct> & <chr> & <chr> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t exome            & NA  & 0.01  & 394 & 5.473985e-71\\\\\n",
       "\t hrc              & 0.3 & 0.01  &  16 & 1.944983e-12\\\\\n",
       "\t hrc              & 0.8 & 0.01  &   6 & 6.993942e-11\\\\\n",
       "\t topmed           & 0.3 & 0.01  & 109 & 1.384075e-39\\\\\n",
       "\t topmed           & 0.8 & 0.01  &  47 & 1.769128e-34\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.01  & 110 & 1.121818e-38\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.01  &  48 & 1.972979e-34\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.01  & 403 & 1.684064e-70\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.01  & 396 & 7.159298e-71\\\\\n",
       "\t exome            & NA  & 0.005 & 394 & 5.473985e-71\\\\\n",
       "\t hrc              & 0.3 & 0.005 &  16 & 1.944983e-12\\\\\n",
       "\t hrc              & 0.8 & 0.005 &   6 & 6.993942e-11\\\\\n",
       "\t topmed           & 0.3 & 0.005 & 109 & 1.384075e-39\\\\\n",
       "\t topmed           & 0.8 & 0.005 &  47 & 1.769128e-34\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.005 & 110 & 1.121818e-38\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.005 &  48 & 1.972979e-34\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.005 & 403 & 1.684064e-70\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.005 & 396 & 7.159298e-71\\\\\n",
       "\t exome            & NA  & 0.001 & 393 & 9.347929e-53\\\\\n",
       "\t hrc              & 0.3 & 0.001 &  15 & 2.387199e-09\\\\\n",
       "\t hrc              & 0.8 & 0.001 &   6 & 6.993942e-11\\\\\n",
       "\t topmed           & 0.3 & 0.001 & 108 & 1.040078e-27\\\\\n",
       "\t topmed           & 0.8 & 0.001 &  46 & 2.013611e-22\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.001 & 109 & 7.965593e-27\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.001 &  47 & 2.473826e-22\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.001 & 402 & 2.584292e-52\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.001 & 395 & 1.200541e-52\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 27 × 5\n",
       "\n",
       "| data &lt;fct&gt; | rsqs &lt;chr&gt; | mafs &lt;chr&gt; | num_var &lt;int&gt; | pvals &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| exome            | NA  | 0.01  | 394 | 5.473985e-71 |\n",
       "| hrc              | 0.3 | 0.01  |  16 | 1.944983e-12 |\n",
       "| hrc              | 0.8 | 0.01  |   6 | 6.993942e-11 |\n",
       "| topmed           | 0.3 | 0.01  | 109 | 1.384075e-39 |\n",
       "| topmed           | 0.8 | 0.01  |  47 | 1.769128e-34 |\n",
       "| hrc_topmed       | 0.3 | 0.01  | 110 | 1.121818e-38 |\n",
       "| hrc_topmed       | 0.8 | 0.01  |  48 | 1.972979e-34 |\n",
       "| hrc_topmed_exome | 0.3 | 0.01  | 403 | 1.684064e-70 |\n",
       "| hrc_topmed_exome | 0.8 | 0.01  | 396 | 7.159298e-71 |\n",
       "| exome            | NA  | 0.005 | 394 | 5.473985e-71 |\n",
       "| hrc              | 0.3 | 0.005 |  16 | 1.944983e-12 |\n",
       "| hrc              | 0.8 | 0.005 |   6 | 6.993942e-11 |\n",
       "| topmed           | 0.3 | 0.005 | 109 | 1.384075e-39 |\n",
       "| topmed           | 0.8 | 0.005 |  47 | 1.769128e-34 |\n",
       "| hrc_topmed       | 0.3 | 0.005 | 110 | 1.121818e-38 |\n",
       "| hrc_topmed       | 0.8 | 0.005 |  48 | 1.972979e-34 |\n",
       "| hrc_topmed_exome | 0.3 | 0.005 | 403 | 1.684064e-70 |\n",
       "| hrc_topmed_exome | 0.8 | 0.005 | 396 | 7.159298e-71 |\n",
       "| exome            | NA  | 0.001 | 393 | 9.347929e-53 |\n",
       "| hrc              | 0.3 | 0.001 |  15 | 2.387199e-09 |\n",
       "| hrc              | 0.8 | 0.001 |   6 | 6.993942e-11 |\n",
       "| topmed           | 0.3 | 0.001 | 108 | 1.040078e-27 |\n",
       "| topmed           | 0.8 | 0.001 |  46 | 2.013611e-22 |\n",
       "| hrc_topmed       | 0.3 | 0.001 | 109 | 7.965593e-27 |\n",
       "| hrc_topmed       | 0.8 | 0.001 |  47 | 2.473826e-22 |\n",
       "| hrc_topmed_exome | 0.3 | 0.001 | 402 | 2.584292e-52 |\n",
       "| hrc_topmed_exome | 0.8 | 0.001 | 395 | 1.200541e-52 |\n",
       "\n"
      ],
      "text/plain": [
       "   data             rsqs mafs  num_var pvals       \n",
       "1  exome            NA   0.01  394     5.473985e-71\n",
       "2  hrc              0.3  0.01   16     1.944983e-12\n",
       "3  hrc              0.8  0.01    6     6.993942e-11\n",
       "4  topmed           0.3  0.01  109     1.384075e-39\n",
       "5  topmed           0.8  0.01   47     1.769128e-34\n",
       "6  hrc_topmed       0.3  0.01  110     1.121818e-38\n",
       "7  hrc_topmed       0.8  0.01   48     1.972979e-34\n",
       "8  hrc_topmed_exome 0.3  0.01  403     1.684064e-70\n",
       "9  hrc_topmed_exome 0.8  0.01  396     7.159298e-71\n",
       "10 exome            NA   0.005 394     5.473985e-71\n",
       "11 hrc              0.3  0.005  16     1.944983e-12\n",
       "12 hrc              0.8  0.005   6     6.993942e-11\n",
       "13 topmed           0.3  0.005 109     1.384075e-39\n",
       "14 topmed           0.8  0.005  47     1.769128e-34\n",
       "15 hrc_topmed       0.3  0.005 110     1.121818e-38\n",
       "16 hrc_topmed       0.8  0.005  48     1.972979e-34\n",
       "17 hrc_topmed_exome 0.3  0.005 403     1.684064e-70\n",
       "18 hrc_topmed_exome 0.8  0.005 396     7.159298e-71\n",
       "19 exome            NA   0.001 393     9.347929e-53\n",
       "20 hrc              0.3  0.001  15     2.387199e-09\n",
       "21 hrc              0.8  0.001   6     6.993942e-11\n",
       "22 topmed           0.3  0.001 108     1.040078e-27\n",
       "23 topmed           0.8  0.001  46     2.013611e-22\n",
       "24 hrc_topmed       0.3  0.001 109     7.965593e-27\n",
       "25 hrc_topmed       0.8  0.001  47     2.473826e-22\n",
       "26 hrc_topmed_exome 0.3  0.001 402     2.584292e-52\n",
       "27 hrc_topmed_exome 0.8  0.001 395     1.200541e-52"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "result_df %>% \n",
    "    mutate(data = factor(data, levels = c(\"exome\", \"hrc\", \"topmed\", \"hrc_topmed\", \"hrc_topmed_exome\"))) %>% \n",
    "    arrange(desc(as.numeric(mafs)), data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "447e8f31-687a-4ce3-823f-907d496ac558",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: PCSK9_exome_maf001_cadd_extracted.bim\n",
      "\n",
      "Reading: PCSK9_exome_maf001_cadd_extracted.fam\n",
      "\n",
      "Reading: PCSK9_exome_maf001_cadd_extracted.bed\n",
      "\n",
      "Reading: PCSK9_exome_maf0005_cadd_extracted.bim\n",
      "\n",
      "Reading: PCSK9_exome_maf0005_cadd_extracted.fam\n",
      "\n",
      "Reading: PCSK9_exome_maf0005_cadd_extracted.bed\n",
      "\n",
      "Reading: PCSK9_exome_maf0001_cadd_extracted.bim\n",
      "\n",
      "Reading: PCSK9_exome_maf0001_cadd_extracted.fam\n",
      "\n",
      "Reading: PCSK9_exome_maf0001_cadd_extracted.bed\n",
      "\n"
     ]
    }
   ],
   "source": [
    "result_df <- data.frame(data.frame(matrix(ncol = 5, nrow = 0)))\n",
    "colnames(result_df) <- c(\"data\", \"rsqs\", \"maf\", \"num_var\", \"pvals\")\n",
    "\n",
    "for(maf in c(\"1\", \"05\", \"01\")){\n",
    "    exome_bim <- read_bim(sprintf(\"PCSK9_exome_maf00%s_cadd_extracted.bim\", maf))\n",
    "    exome_fam <- read_fam(sprintf(\"PCSK9_exome_maf00%s_cadd_extracted.fam\", maf))\n",
    "    exome <- read_bed(sprintf(\"PCSK9_exome_maf00%s_cadd_extracted.bed\", maf), exome_bim$id, exome_fam$id)\n",
    "    exome_X <- exome %>% t() %>% as.data.frame() %>% tibble::rownames_to_column(\"FID\") %>% \n",
    "            mutate(FID = as.numeric(FID)) %>% arrange(FID) %>% filter(FID %in% ldl_id) %>% select(-FID) %>% as.matrix()\n",
    "    result_exome <- BRV(exome_X, ldl_y, ldl_cov, \"gaussian\")\n",
    "    \n",
    "    for(rsq in c(3, 8)){\n",
    "        hrc_fname <- sprintf(\"PCSK9_hrc_rsq0%d_maf00%s_cadd_extracted.traw\", rsq, maf)\n",
    "        topmed_fname <- sprintf(\"PCSK9_topmed_rsq0%d_maf00%s_cadd_extracted.traw\", rsq, maf)\n",
    "        hrc_topmed_fname <- sprintf(\"PCSK9_hrc_topmed_rsq0%d_maf00%s_cadd_extracted.traw\", rsq, maf)\n",
    "        hrc_topmed_exome_fname <- sprintf(\"PCSK9_hrc_topmed_exome_rsq0%d_maf00%s_cadd_extracted.traw\", rsq, maf)\n",
    "\n",
    "        hrc <- fread(hrc_fname)\n",
    "        hrc_X <- format_traw(hrc, ldl_id, \"hrc\")\n",
    "        result_hrc <- BRV(hrc_X, ldl_y, ldl_cov, \"gaussian\")\n",
    "\n",
    "        topmed <- fread(topmed_fname)\n",
    "        topmed_X <- format_traw(topmed, ldl_id, \"topmed\")\n",
    "        result_topmed <- BRV(topmed_X, ldl_y, ldl_cov, \"gaussian\")\n",
    "\n",
    "        hrc_topmed <- fread(hrc_topmed_fname)\n",
    "        hrc_topmed_X <- format_traw(hrc_topmed, ldl_id)\n",
    "        result_hrc_topmed <- BRV(hrc_topmed_X, ldl_y, ldl_cov, \"gaussian\")\n",
    "            \n",
    "        hrc_topmed_exome <- fread(hrc_topmed_exome_fname)\n",
    "        hrc_topmed_exome_X <- format_traw(hrc_topmed_exome, ldl_id)\n",
    "        result_hrc_topmed_exome <- BRV(hrc_topmed_exome_X, ldl_y, ldl_cov, \"gaussian\")\n",
    "            \n",
    "        data <- c(\"hrc\", \"topmed\", \"hrc_topmed\", \"hrc_topmed_exome\")\n",
    "        rsqs <- rep(rsq/10, 4)\n",
    "        mafs <- rep(paste0(\"0.0\", maf), 4)\n",
    "        pvals <- c(result_hrc$pval, result_topmed$pval, result_hrc_topmed$pval, result_hrc_topmed_exome$pval)\n",
    "        num_var <- c(ncol(hrc_X), ncol(topmed_X), ncol(hrc_topmed_X), ncol(hrc_topmed_exome_X))\n",
    "        sub_df <- data.frame(data, rsqs, mafs, num_var, pvals)\n",
    "        result_df <- rbind(result_df, sub_df)\n",
    "    }\n",
    "    result_df <- rbind(data.frame(data = \"exome\", rsqs = \"NA\", mafs = paste0(\"0.0\", maf), num_var = ncol(exome_X), pvals = result_exome$pval), result_df)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "63dbcb72-fc6e-4d98-8cf7-1fbb94ae1f9e",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 27 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>data</th><th scope=col>rsqs</th><th scope=col>mafs</th><th scope=col>num_var</th><th scope=col>pvals</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.01 </td><td>200</td><td>8.246876e-93</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.01 </td><td>  6</td><td>1.912681e-12</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.01 </td><td>  2</td><td>4.563676e-15</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.01 </td><td> 61</td><td>2.297162e-50</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.01 </td><td> 25</td><td>6.112257e-45</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.01 </td><td> 62</td><td>1.269456e-50</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.01 </td><td> 26</td><td>1.855966e-45</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.01 </td><td>202</td><td>6.598130e-93</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.01 </td><td>201</td><td>8.034693e-93</td></tr>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.005</td><td>200</td><td>8.246876e-93</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.005</td><td>  6</td><td>1.912681e-12</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.005</td><td>  2</td><td>4.563676e-15</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.005</td><td> 61</td><td>2.297162e-50</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.005</td><td> 25</td><td>6.112257e-45</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.005</td><td> 62</td><td>1.269456e-50</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.005</td><td> 26</td><td>1.855966e-45</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.005</td><td>202</td><td>6.598130e-93</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.005</td><td>201</td><td>8.034693e-93</td></tr>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.001</td><td>199</td><td>2.525363e-74</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.001</td><td>  5</td><td>5.727402e-12</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.001</td><td>  2</td><td>4.563676e-15</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.001</td><td> 60</td><td>2.952016e-38</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.001</td><td> 24</td><td>3.027417e-34</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.001</td><td> 61</td><td>1.587594e-38</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.001</td><td> 25</td><td>8.632543e-35</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.001</td><td>201</td><td>2.013182e-74</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.001</td><td>200</td><td>2.456874e-74</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 27 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " data & rsqs & mafs & num\\_var & pvals\\\\\n",
       " <fct> & <chr> & <chr> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t exome            & NA  & 0.01  & 200 & 8.246876e-93\\\\\n",
       "\t hrc              & 0.3 & 0.01  &   6 & 1.912681e-12\\\\\n",
       "\t hrc              & 0.8 & 0.01  &   2 & 4.563676e-15\\\\\n",
       "\t topmed           & 0.3 & 0.01  &  61 & 2.297162e-50\\\\\n",
       "\t topmed           & 0.8 & 0.01  &  25 & 6.112257e-45\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.01  &  62 & 1.269456e-50\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.01  &  26 & 1.855966e-45\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.01  & 202 & 6.598130e-93\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.01  & 201 & 8.034693e-93\\\\\n",
       "\t exome            & NA  & 0.005 & 200 & 8.246876e-93\\\\\n",
       "\t hrc              & 0.3 & 0.005 &   6 & 1.912681e-12\\\\\n",
       "\t hrc              & 0.8 & 0.005 &   2 & 4.563676e-15\\\\\n",
       "\t topmed           & 0.3 & 0.005 &  61 & 2.297162e-50\\\\\n",
       "\t topmed           & 0.8 & 0.005 &  25 & 6.112257e-45\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.005 &  62 & 1.269456e-50\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.005 &  26 & 1.855966e-45\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.005 & 202 & 6.598130e-93\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.005 & 201 & 8.034693e-93\\\\\n",
       "\t exome            & NA  & 0.001 & 199 & 2.525363e-74\\\\\n",
       "\t hrc              & 0.3 & 0.001 &   5 & 5.727402e-12\\\\\n",
       "\t hrc              & 0.8 & 0.001 &   2 & 4.563676e-15\\\\\n",
       "\t topmed           & 0.3 & 0.001 &  60 & 2.952016e-38\\\\\n",
       "\t topmed           & 0.8 & 0.001 &  24 & 3.027417e-34\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.001 &  61 & 1.587594e-38\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.001 &  25 & 8.632543e-35\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.001 & 201 & 2.013182e-74\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.001 & 200 & 2.456874e-74\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 27 × 5\n",
       "\n",
       "| data &lt;fct&gt; | rsqs &lt;chr&gt; | mafs &lt;chr&gt; | num_var &lt;int&gt; | pvals &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| exome            | NA  | 0.01  | 200 | 8.246876e-93 |\n",
       "| hrc              | 0.3 | 0.01  |   6 | 1.912681e-12 |\n",
       "| hrc              | 0.8 | 0.01  |   2 | 4.563676e-15 |\n",
       "| topmed           | 0.3 | 0.01  |  61 | 2.297162e-50 |\n",
       "| topmed           | 0.8 | 0.01  |  25 | 6.112257e-45 |\n",
       "| hrc_topmed       | 0.3 | 0.01  |  62 | 1.269456e-50 |\n",
       "| hrc_topmed       | 0.8 | 0.01  |  26 | 1.855966e-45 |\n",
       "| hrc_topmed_exome | 0.3 | 0.01  | 202 | 6.598130e-93 |\n",
       "| hrc_topmed_exome | 0.8 | 0.01  | 201 | 8.034693e-93 |\n",
       "| exome            | NA  | 0.005 | 200 | 8.246876e-93 |\n",
       "| hrc              | 0.3 | 0.005 |   6 | 1.912681e-12 |\n",
       "| hrc              | 0.8 | 0.005 |   2 | 4.563676e-15 |\n",
       "| topmed           | 0.3 | 0.005 |  61 | 2.297162e-50 |\n",
       "| topmed           | 0.8 | 0.005 |  25 | 6.112257e-45 |\n",
       "| hrc_topmed       | 0.3 | 0.005 |  62 | 1.269456e-50 |\n",
       "| hrc_topmed       | 0.8 | 0.005 |  26 | 1.855966e-45 |\n",
       "| hrc_topmed_exome | 0.3 | 0.005 | 202 | 6.598130e-93 |\n",
       "| hrc_topmed_exome | 0.8 | 0.005 | 201 | 8.034693e-93 |\n",
       "| exome            | NA  | 0.001 | 199 | 2.525363e-74 |\n",
       "| hrc              | 0.3 | 0.001 |   5 | 5.727402e-12 |\n",
       "| hrc              | 0.8 | 0.001 |   2 | 4.563676e-15 |\n",
       "| topmed           | 0.3 | 0.001 |  60 | 2.952016e-38 |\n",
       "| topmed           | 0.8 | 0.001 |  24 | 3.027417e-34 |\n",
       "| hrc_topmed       | 0.3 | 0.001 |  61 | 1.587594e-38 |\n",
       "| hrc_topmed       | 0.8 | 0.001 |  25 | 8.632543e-35 |\n",
       "| hrc_topmed_exome | 0.3 | 0.001 | 201 | 2.013182e-74 |\n",
       "| hrc_topmed_exome | 0.8 | 0.001 | 200 | 2.456874e-74 |\n",
       "\n"
      ],
      "text/plain": [
       "   data             rsqs mafs  num_var pvals       \n",
       "1  exome            NA   0.01  200     8.246876e-93\n",
       "2  hrc              0.3  0.01    6     1.912681e-12\n",
       "3  hrc              0.8  0.01    2     4.563676e-15\n",
       "4  topmed           0.3  0.01   61     2.297162e-50\n",
       "5  topmed           0.8  0.01   25     6.112257e-45\n",
       "6  hrc_topmed       0.3  0.01   62     1.269456e-50\n",
       "7  hrc_topmed       0.8  0.01   26     1.855966e-45\n",
       "8  hrc_topmed_exome 0.3  0.01  202     6.598130e-93\n",
       "9  hrc_topmed_exome 0.8  0.01  201     8.034693e-93\n",
       "10 exome            NA   0.005 200     8.246876e-93\n",
       "11 hrc              0.3  0.005   6     1.912681e-12\n",
       "12 hrc              0.8  0.005   2     4.563676e-15\n",
       "13 topmed           0.3  0.005  61     2.297162e-50\n",
       "14 topmed           0.8  0.005  25     6.112257e-45\n",
       "15 hrc_topmed       0.3  0.005  62     1.269456e-50\n",
       "16 hrc_topmed       0.8  0.005  26     1.855966e-45\n",
       "17 hrc_topmed_exome 0.3  0.005 202     6.598130e-93\n",
       "18 hrc_topmed_exome 0.8  0.005 201     8.034693e-93\n",
       "19 exome            NA   0.001 199     2.525363e-74\n",
       "20 hrc              0.3  0.001   5     5.727402e-12\n",
       "21 hrc              0.8  0.001   2     4.563676e-15\n",
       "22 topmed           0.3  0.001  60     2.952016e-38\n",
       "23 topmed           0.8  0.001  24     3.027417e-34\n",
       "24 hrc_topmed       0.3  0.001  61     1.587594e-38\n",
       "25 hrc_topmed       0.8  0.001  25     8.632543e-35\n",
       "26 hrc_topmed_exome 0.3  0.001 201     2.013182e-74\n",
       "27 hrc_topmed_exome 0.8  0.001 200     2.456874e-74"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "result_df %>% \n",
    "    mutate(data = factor(data, levels = c(\"exome\", \"hrc\", \"topmed\", \"hrc_topmed\", \"hrc_topmed_exome\"))) %>% \n",
    "    arrange(desc(as.numeric(mafs)), data)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1023be46-7d3a-4fad-b525-240481c70e80",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "#### PCSK9: chr1:55052701:C:T (hg38)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "b9d5206b-b654-4789-9335-b9d59fafe742",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "'chr1:55052701:C:T'"
      ],
      "text/latex": [
       "'chr1:55052701:C:T'"
      ],
      "text/markdown": [
       "'chr1:55052701:C:T'"
      ],
      "text/plain": [
       "[1] \"chr1:55052701:C:T\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Chr</th><th scope=col>Start</th><th scope=col>End</th><th scope=col>Ref</th><th scope=col>Alt</th><th scope=col>Func.refGene</th><th scope=col>Gene.refGene</th><th scope=col>ExonicFunc.refGene</th><th scope=col>MAF_nfe_exome</th><th scope=col>REVEL_score</th><th scope=col>ID_hg38</th><th scope=col>ID</th><th scope=col>Function</th><th scope=col>RawScore</th><th scope=col>PHRED</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>55052701</td><td>55052701</td><td>C</td><td>T</td><td>exonic</td><td>PCSK9</td><td>nonsynonymous SNV</td><td>0.001</td><td>0.615</td><td>chr1:55052701:C:T</td><td>chr1:55052701:C:T</td><td>missense</td><td>3.457351</td><td>24.6</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 15\n",
       "\\begin{tabular}{lllllllllllllll}\n",
       " Chr & Start & End & Ref & Alt & Func.refGene & Gene.refGene & ExonicFunc.refGene & MAF\\_nfe\\_exome & REVEL\\_score & ID\\_hg38 & ID & Function & RawScore & PHRED\\\\\n",
       " <int> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t 1 & 55052701 & 55052701 & C & T & exonic & PCSK9 & nonsynonymous SNV & 0.001 & 0.615 & chr1:55052701:C:T & chr1:55052701:C:T & missense & 3.457351 & 24.6\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 15\n",
       "\n",
       "| Chr &lt;int&gt; | Start &lt;int&gt; | End &lt;int&gt; | Ref &lt;chr&gt; | Alt &lt;chr&gt; | Func.refGene &lt;chr&gt; | Gene.refGene &lt;chr&gt; | ExonicFunc.refGene &lt;chr&gt; | MAF_nfe_exome &lt;dbl&gt; | REVEL_score &lt;chr&gt; | ID_hg38 &lt;chr&gt; | ID &lt;chr&gt; | Function &lt;chr&gt; | RawScore &lt;dbl&gt; | PHRED &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 55052701 | 55052701 | C | T | exonic | PCSK9 | nonsynonymous SNV | 0.001 | 0.615 | chr1:55052701:C:T | chr1:55052701:C:T | missense | 3.457351 | 24.6 |\n",
       "\n"
      ],
      "text/plain": [
       "  Chr Start    End      Ref Alt Func.refGene Gene.refGene ExonicFunc.refGene\n",
       "1 1   55052701 55052701 C   T   exonic       PCSK9        nonsynonymous SNV \n",
       "  MAF_nfe_exome REVEL_score ID_hg38           ID                Function\n",
       "1 0.001         0.615       chr1:55052701:C:T chr1:55052701:C:T missense\n",
       "  RawScore PHRED\n",
       "1 3.457351 24.6 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "pcsk9_id_maf001 <- fread(\"PCSK9_exome_maf001_annot.csv.gz\")$ID_hg38\n",
    "pcsk9_id_maf0001 <- fread(\"PCSK9_exome_maf0001_annot.csv.gz\")$ID_hg38\n",
    "setdiff(pcsk9_id_maf001, pcsk9_id_maf0001)\n",
    "\n",
    "fread(\"PCSK9_exome_maf001_annot.csv.gz\") %>% filter(ID_hg38 == 'chr1:55052701:C:T')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4c87a3f8-b946-43b9-8ccc-1052f9713c2d",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "We are interested in this variant, because we see that after removal, the p-value has increased a lot. Therefore, we are going to conduct a single varaint association testing for this variants alone using exome and imputed data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "b48ea397-b076-40c1-b262-1b13c4d655d0",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: PCSK9_exome_maf001_extracted.bim\n",
      "\n",
      "Reading: PCSK9_exome_maf001_extracted.fam\n",
      "\n",
      "Reading: PCSK9_exome_maf001_extracted.bed\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "8.52997529387925e-21"
      ],
      "text/latex": [
       "8.52997529387925e-21"
      ],
      "text/markdown": [
       "8.52997529387925e-21"
      ],
      "text/plain": [
       "[1] 8.529975e-21"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## exome data\n",
    "exome_bim <- read_bim(\"PCSK9_exome_maf001_extracted.bim\")\n",
    "exome_fam <- read_fam(\"PCSK9_exome_maf001_extracted.fam\")\n",
    "exome <- read_bed(\"PCSK9_exome_maf001_extracted.bed\", exome_bim$id, exome_fam$id)\n",
    "\n",
    "exome_X <- data.frame(ID = names(exome[\"chr1:55052701:C:T\",]), value=exome[\"chr1:55052701:C:T\",], row.names=NULL)\n",
    "exome_X <- exome_X %>% arrange(as.numeric(ID)) %>% filter(ID %in% ldl_id) %>% select(-ID) %>% as.matrix()\n",
    "\n",
    "exome.fit <- glm(ldl_y ~ exome_X + ldl_cov$age + ldl_cov$sex + ldl_cov$PC1 + ldl_cov$PC2, family = \"gaussian\")\n",
    "coef(summary(exome.fit))[2, 4]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "7aa5fe18-a4f4-40a8-97a0-77f0d9fca848",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 17</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Chr</th><th scope=col>Start</th><th scope=col>End</th><th scope=col>Ref</th><th scope=col>Alt</th><th scope=col>Func.refGene</th><th scope=col>Gene.refGene</th><th scope=col>ExonicFunc.refGene</th><th scope=col>MAF_nfe_exome</th><th scope=col>REVEL_score</th><th scope=col>ID_hg38</th><th scope=col>ID_hg19</th><th scope=col>ID</th><th scope=col>Function</th><th scope=col>RawScore</th><th scope=col>PHRED</th><th scope=col>R2</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>55052701</td><td>55052701</td><td>C</td><td>T</td><td>exonic</td><td>PCSK9</td><td>nonsynonymous SNV</td><td>0.001</td><td>0.615</td><td>chr1:55052701:C:T</td><td>chr1:55518374:C:T</td><td>chr1:55052701:C:T</td><td>missense</td><td>3.341231</td><td>24.5</td><td>0.677861</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 17\n",
       "\\begin{tabular}{lllllllllllllllll}\n",
       " Chr & Start & End & Ref & Alt & Func.refGene & Gene.refGene & ExonicFunc.refGene & MAF\\_nfe\\_exome & REVEL\\_score & ID\\_hg38 & ID\\_hg19 & ID & Function & RawScore & PHRED & R2\\\\\n",
       " <int> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t 1 & 55052701 & 55052701 & C & T & exonic & PCSK9 & nonsynonymous SNV & 0.001 & 0.615 & chr1:55052701:C:T & chr1:55518374:C:T & chr1:55052701:C:T & missense & 3.341231 & 24.5 & 0.677861\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 17\n",
       "\n",
       "| Chr &lt;int&gt; | Start &lt;int&gt; | End &lt;int&gt; | Ref &lt;chr&gt; | Alt &lt;chr&gt; | Func.refGene &lt;chr&gt; | Gene.refGene &lt;chr&gt; | ExonicFunc.refGene &lt;chr&gt; | MAF_nfe_exome &lt;dbl&gt; | REVEL_score &lt;chr&gt; | ID_hg38 &lt;chr&gt; | ID_hg19 &lt;chr&gt; | ID &lt;chr&gt; | Function &lt;chr&gt; | RawScore &lt;dbl&gt; | PHRED &lt;dbl&gt; | R2 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 55052701 | 55052701 | C | T | exonic | PCSK9 | nonsynonymous SNV | 0.001 | 0.615 | chr1:55052701:C:T | chr1:55518374:C:T | chr1:55052701:C:T | missense | 3.341231 | 24.5 | 0.677861 |\n",
       "\n"
      ],
      "text/plain": [
       "  Chr Start    End      Ref Alt Func.refGene Gene.refGene ExonicFunc.refGene\n",
       "1 1   55052701 55052701 C   T   exonic       PCSK9        nonsynonymous SNV \n",
       "  MAF_nfe_exome REVEL_score ID_hg38           ID_hg19          \n",
       "1 0.001         0.615       chr1:55052701:C:T chr1:55518374:C:T\n",
       "  ID                Function RawScore PHRED R2      \n",
       "1 chr1:55052701:C:T missense 3.341231 24.5  0.677861"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0.00014734666556777"
      ],
      "text/latex": [
       "0.00014734666556777"
      ],
      "text/markdown": [
       "0.00014734666556777"
      ],
      "text/plain": [
       "[1] 0.0001473467"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## hrc\n",
    "fread(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/hrc_topmed/hrc_168206ids_chr1_rsq03_maf001_annot.csv.gz\") %>%\n",
    "filter(ID_hg19 == \"chr1:55518374:C:T\")\n",
    "\n",
    "hrc <- fread(\"PCSK9_hrc_rsq03_maf001_extracted.traw\") %>% filter(SNP == \"chr1:55518374:C:T\")\n",
    "hrc_X <- format_traw(hrc, ldl_id, \"hrc\")\n",
    "\n",
    "hrc.fit <- glm(ldl_y ~ hrc_X + ldl_cov$age + ldl_cov$sex + ldl_cov$PC1 + ldl_cov$PC2, family = \"gaussian\")\n",
    "coef(summary(hrc.fit))[2, 4]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "425179ad-742a-4936-8077-172ac896e041",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 17</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Chr</th><th scope=col>Start</th><th scope=col>End</th><th scope=col>Ref</th><th scope=col>Alt</th><th scope=col>Func.refGene</th><th scope=col>Gene.refGene</th><th scope=col>ExonicFunc.refGene</th><th scope=col>MAF_nfe_exome</th><th scope=col>REVEL_score</th><th scope=col>ID_hg38</th><th scope=col>ID_hg19</th><th scope=col>ID</th><th scope=col>Function</th><th scope=col>RawScore</th><th scope=col>PHRED</th><th scope=col>R2</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>55052701</td><td>55052701</td><td>C</td><td>T</td><td>exonic</td><td>PCSK9</td><td>nonsynonymous SNV</td><td>0.001</td><td>0.615</td><td>chr1:55052701:C:T</td><td>chr1:55052701:C:T</td><td>chr1:55052701:C:T</td><td>missense</td><td>3.457351</td><td>24.6</td><td>0.815682</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 17\n",
       "\\begin{tabular}{lllllllllllllllll}\n",
       " Chr & Start & End & Ref & Alt & Func.refGene & Gene.refGene & ExonicFunc.refGene & MAF\\_nfe\\_exome & REVEL\\_score & ID\\_hg38 & ID\\_hg19 & ID & Function & RawScore & PHRED & R2\\\\\n",
       " <int> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t 1 & 55052701 & 55052701 & C & T & exonic & PCSK9 & nonsynonymous SNV & 0.001 & 0.615 & chr1:55052701:C:T & chr1:55052701:C:T & chr1:55052701:C:T & missense & 3.457351 & 24.6 & 0.815682\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 17\n",
       "\n",
       "| Chr &lt;int&gt; | Start &lt;int&gt; | End &lt;int&gt; | Ref &lt;chr&gt; | Alt &lt;chr&gt; | Func.refGene &lt;chr&gt; | Gene.refGene &lt;chr&gt; | ExonicFunc.refGene &lt;chr&gt; | MAF_nfe_exome &lt;dbl&gt; | REVEL_score &lt;chr&gt; | ID_hg38 &lt;chr&gt; | ID_hg19 &lt;chr&gt; | ID &lt;chr&gt; | Function &lt;chr&gt; | RawScore &lt;dbl&gt; | PHRED &lt;dbl&gt; | R2 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 55052701 | 55052701 | C | T | exonic | PCSK9 | nonsynonymous SNV | 0.001 | 0.615 | chr1:55052701:C:T | chr1:55052701:C:T | chr1:55052701:C:T | missense | 3.457351 | 24.6 | 0.815682 |\n",
       "\n"
      ],
      "text/plain": [
       "  Chr Start    End      Ref Alt Func.refGene Gene.refGene ExonicFunc.refGene\n",
       "1 1   55052701 55052701 C   T   exonic       PCSK9        nonsynonymous SNV \n",
       "  MAF_nfe_exome REVEL_score ID_hg38           ID_hg19          \n",
       "1 0.001         0.615       chr1:55052701:C:T chr1:55052701:C:T\n",
       "  ID                Function RawScore PHRED R2      \n",
       "1 chr1:55052701:C:T missense 3.457351 24.6  0.815682"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "4.5832816186623e-14"
      ],
      "text/latex": [
       "4.5832816186623e-14"
      ],
      "text/markdown": [
       "4.5832816186623e-14"
      ],
      "text/plain": [
       "[1] 4.583282e-14"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fread(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/hrc_topmed/topmed_168206ids_chr1_rsq03_maf001_annot.csv.gz\") %>%\n",
    "filter(ID_hg19 == \"chr1:55052701:C:T\")\n",
    "\n",
    "topmed <- fread(\"PCSK9_topmed_rsq03_maf001_extracted.traw\") %>% filter(SNP == \"chr1:55052701:C:T\")\n",
    "topmed_X <- format_traw(topmed, ldl_id, \"topmed\")\n",
    "\n",
    "topmed.fit <- glm(ldl_y ~ topmed_X + ldl_cov$age + ldl_cov$sex + ldl_cov$PC1 + ldl_cov$PC2, family = \"gaussian\")\n",
    "coef(summary(topmed.fit))[2, 4]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "49c05ddf-d21c-44c6-9b95-6cec27250fb9",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "#### PCSK9: HRC dataset inspection\n",
    "\n",
    "We observe that for less variant, there is a more significant p-value. Therefore, we further investigated variants in HRC dataset."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "519cfabf-fc95-466a-b49f-7cedd0585bbb",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "annot <- fread(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/hrc/hrc_chr1_rsq03_hg19_hg38_maf001_LOF_missense_cadd_annot.csv.gz\")\n",
    "pcsk9_hrc_rsq03 <- annot %>% filter(Gene.refGene == \"PCSK9\") %>% pull(ID_hg19)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "8a55094d-8e3d-4257-8c66-e4641c0ee150",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "annot <- fread(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/hrc/hrc_chr1_rsq08_hg19_hg38_maf001_LOF_missense_cadd_annot.csv.gz\")\n",
    "pcsk9_hrc_rsq08 <- annot %>% filter(Gene.refGene == \"PCSK9\") %>% pull(ID_hg19)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "12c9b0df-6717-4c10-a61d-e55ad4d8d9ac",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "annot <- fread(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/hrc_topmed/hrc_168206ids_chr1_rsq03_maf001_annot.csv.gz\")\n",
    "diff_lst <- setdiff(pcsk9_hrc_rsq03, pcsk9_hrc_rsq08)\n",
    "hrc_mat <- fread(\"PCSK9_hrc_rsq03_maf001_extracted.traw\")\n",
    "\n",
    "df <- data.frame(data.frame(matrix(ncol = 4, nrow = 0)))\n",
    "colnames(df) <- c(\"variant\", \"rsqs\", \"functions\", \"pvals\")\n",
    "for(snp in diff_lst){\n",
    "    snp_info <- annot %>% filter(ID_hg19 == snp)\n",
    "    \n",
    "    hrc_X <- format_traw(hrc_mat %>% filter(SNP == snp), ldl_id, \"hrc\")\n",
    "    hrc.fit <- glm(ldl_y ~ hrc_X + ldl_cov$age + ldl_cov$sex + ldl_cov$PC1 + ldl_cov$PC2, family = \"gaussian\")\n",
    "    \n",
    "    subdf <- data.frame(variant = snp_info$ID, \n",
    "                        rsqs = snp_info$R2,\n",
    "                        functions = snp_info$Function,\n",
    "                        pvals = coef(summary(hrc.fit))[2, 4])\n",
    "    df <- rbind(df, subdf) \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "ea7e614d-4ea0-493a-a562-2e0cf5baff82",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 4 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>variant</th><th scope=col>rsqs</th><th scope=col>functions</th><th scope=col>pvals</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1:55046549:C:G</td><td>0.652429</td><td>LoF     </td><td>0.5391590013</td></tr>\n",
       "\t<tr><td>chr1:55052701:C:T</td><td>0.677861</td><td>missense</td><td>0.0001473467</td></tr>\n",
       "\t<tr><td>chr1:55056028:C:A</td><td>0.605831</td><td>missense</td><td>0.5496690045</td></tr>\n",
       "\t<tr><td>chr1:55057403:C:T</td><td>0.536730</td><td>missense</td><td>0.5307091797</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 4 × 4\n",
       "\\begin{tabular}{llll}\n",
       " variant & rsqs & functions & pvals\\\\\n",
       " <chr> & <dbl> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t chr1:55046549:C:G & 0.652429 & LoF      & 0.5391590013\\\\\n",
       "\t chr1:55052701:C:T & 0.677861 & missense & 0.0001473467\\\\\n",
       "\t chr1:55056028:C:A & 0.605831 & missense & 0.5496690045\\\\\n",
       "\t chr1:55057403:C:T & 0.536730 & missense & 0.5307091797\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 4 × 4\n",
       "\n",
       "| variant &lt;chr&gt; | rsqs &lt;dbl&gt; | functions &lt;chr&gt; | pvals &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| chr1:55046549:C:G | 0.652429 | LoF      | 0.5391590013 |\n",
       "| chr1:55052701:C:T | 0.677861 | missense | 0.0001473467 |\n",
       "| chr1:55056028:C:A | 0.605831 | missense | 0.5496690045 |\n",
       "| chr1:55057403:C:T | 0.536730 | missense | 0.5307091797 |\n",
       "\n"
      ],
      "text/plain": [
       "  variant           rsqs     functions pvals       \n",
       "1 chr1:55046549:C:G 0.652429 LoF       0.5391590013\n",
       "2 chr1:55052701:C:T 0.677861 missense  0.0001473467\n",
       "3 chr1:55056028:C:A 0.605831 missense  0.5496690045\n",
       "4 chr1:55057403:C:T 0.536730 missense  0.5307091797"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "64e44ee9-b2e6-47cd-81e7-e5384db5fbcf",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "annot <- fread(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/hrc_topmed/hrc_168206ids_chr1_rsq03_maf001_annot.csv.gz\")\n",
    "diff_lst <- pcsk9_hrc_rsq08\n",
    "hrc_mat <- fread(\"PCSK9_hrc_rsq03_maf001_extracted.traw\")\n",
    "\n",
    "df <- data.frame(data.frame(matrix(ncol = 4, nrow = 0)))\n",
    "colnames(df) <- c(\"variant\", \"rsqs\", \"functions\", \"pvals\")\n",
    "for(snp in diff_lst){\n",
    "    snp_info <- annot %>% filter(ID_hg19 == snp)\n",
    "    \n",
    "    hrc_X <- format_traw(hrc_mat %>% filter(SNP == snp), ldl_id, \"hrc\")\n",
    "    hrc.fit <- glm(ldl_y ~ hrc_X + ldl_cov$age + ldl_cov$sex + ldl_cov$PC1 + ldl_cov$PC2, family = \"gaussian\")\n",
    "    \n",
    "    subdf <- data.frame(variant = snp_info$ID, \n",
    "                        rsqs = snp_info$R2,\n",
    "                        functions = snp_info$Function,\n",
    "                        pvals = coef(summary(hrc.fit))[2, 4])\n",
    "    df <- rbind(df, subdf) \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "f063e5f2-8f3b-4ae5-a6cf-c67f8054cf7f",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>variant</th><th scope=col>rsqs</th><th scope=col>functions</th><th scope=col>pvals</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1:55063542:C:A</td><td>0.825654</td><td>LoF     </td><td>1.193677e-01</td></tr>\n",
       "\t<tr><td>chr1:55061557:G:A</td><td>0.850970</td><td>splicing</td><td>1.200053e-14</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 4\n",
       "\\begin{tabular}{llll}\n",
       " variant & rsqs & functions & pvals\\\\\n",
       " <chr> & <dbl> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t chr1:55063542:C:A & 0.825654 & LoF      & 1.193677e-01\\\\\n",
       "\t chr1:55061557:G:A & 0.850970 & splicing & 1.200053e-14\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 4\n",
       "\n",
       "| variant &lt;chr&gt; | rsqs &lt;dbl&gt; | functions &lt;chr&gt; | pvals &lt;dbl&gt; |\n",
       "|---|---|---|---|\n",
       "| chr1:55063542:C:A | 0.825654 | LoF      | 1.193677e-01 |\n",
       "| chr1:55061557:G:A | 0.850970 | splicing | 1.200053e-14 |\n",
       "\n"
      ],
      "text/plain": [
       "  variant           rsqs     functions pvals       \n",
       "1 chr1:55063542:C:A 0.825654 LoF       1.193677e-01\n",
       "2 chr1:55061557:G:A 0.850970 splicing  1.200053e-14"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1a23cc7d-0b1c-4751-941c-9dcd48bf3ba2",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "### APOC3 v.s. TG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "810a052c-b68e-4c71-8510-3b75e48d8dba",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "setwd(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/apoc3\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "1cf3e5d9-6067-4bee-8761-12a776222236",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: APOC3_exome_maf001_extracted.bim\n",
      "\n",
      "Reading: APOC3_exome_maf001_extracted.fam\n",
      "\n",
      "Reading: APOC3_exome_maf001_extracted.bed\n",
      "\n",
      "Reading: APOC3_exome_maf0005_extracted.bim\n",
      "\n",
      "Reading: APOC3_exome_maf0005_extracted.fam\n",
      "\n",
      "Reading: APOC3_exome_maf0005_extracted.bed\n",
      "\n",
      "Reading: APOC3_exome_maf0001_extracted.bim\n",
      "\n",
      "Reading: APOC3_exome_maf0001_extracted.fam\n",
      "\n",
      "Reading: APOC3_exome_maf0001_extracted.bed\n",
      "\n"
     ]
    }
   ],
   "source": [
    "result_df <- data.frame(data.frame(matrix(ncol = 5, nrow = 0)))\n",
    "colnames(result_df) <- c(\"data\", \"rsqs\", \"mafs\", \"num_var\", \"pvals\")\n",
    "\n",
    "for(maf in c(\"1\", \"05\", \"01\")){ \n",
    "    exome_bim <- read_bim(sprintf(\"APOC3_exome_maf00%s_extracted.bim\", maf))\n",
    "    exome_fam <- read_fam(sprintf(\"APOC3_exome_maf00%s_extracted.fam\", maf))\n",
    "    exome <- read_bed(sprintf(\"APOC3_exome_maf00%s_extracted.bed\", maf), exome_bim$id, exome_fam$id)\n",
    "    exome_X <- exome %>% t() %>% as.data.frame() %>% tibble::rownames_to_column(\"FID\") %>% \n",
    "            mutate(FID = as.numeric(FID)) %>% arrange(FID) %>% filter(FID %in% tg_id) %>% select(-FID) %>% as.matrix()\n",
    "    result_exome <- BRV(exome_X, tg_y, tg_cov,\"gaussian\")\n",
    "    \n",
    "    for(rsq in c(3, 8)){\n",
    "        hrc_fname <- sprintf(\"APOC3_hrc_rsq0%d_maf00%s_extracted.traw\", rsq, maf)\n",
    "        topmed_fname <- sprintf(\"APOC3_topmed_rsq0%d_maf00%s_extracted.traw\", rsq, maf)\n",
    "        hrc_topmed_fname <- sprintf(\"APOC3_hrc_topmed_rsq0%d_maf00%s_extracted.traw\", rsq, maf)\n",
    "        hrc_topmed_exome_fname <- sprintf(\"APOC3_hrc_topmed_exome_rsq0%d_maf00%s_extracted.traw\", rsq, maf)\n",
    "\n",
    "        if(file.exists(hrc_fname)){\n",
    "            hrc <- fread(hrc_fname)\n",
    "            hrc_X <- format_traw(hrc, tg_id, \"hrc\")\n",
    "            result_hrc <- BRV(hrc_X, tg_y, tg_cov, \"gaussian\")\n",
    "        } else {\n",
    "            hrc_X <- matrix()\n",
    "            result_hrc$pval <- NA\n",
    "        }\n",
    "        \n",
    "        if(file.exists(topmed_fname)){\n",
    "            topmed <- fread(topmed_fname)\n",
    "            topmed_X <- format_traw(topmed, tg_id, \"topmed\")\n",
    "            result_topmed <- BRV(topmed_X, tg_y, tg_cov, \"gaussian\")\n",
    "        } else {\n",
    "            topmed_X <- matrix()\n",
    "            result_topmed$pval <- NA\n",
    "        }\n",
    "\n",
    "        if(file.exists(hrc_topmed_fname)){\n",
    "            hrc_topmed <- fread(hrc_topmed_fname)\n",
    "            hrc_topmed_X <- format_traw(hrc_topmed, tg_id)\n",
    "            result_hrc_topmed <- BRV(hrc_topmed_X, tg_y, tg_cov, \"gaussian\")\n",
    "        } else {\n",
    "            hrc_topmed_X <- matrix()\n",
    "            result_hrc_topmed$pval <- NA\n",
    "        }\n",
    "\n",
    "        if(file.exists(hrc_topmed_exome_fname)){\n",
    "            hrc_topmed_exome <- fread(hrc_topmed_exome_fname)\n",
    "            hrc_topmed_exome_X <- format_traw(hrc_topmed_exome, tg_id)\n",
    "            result_hrc_topmed_exome <- BRV(hrc_topmed_exome_X, tg_y, tg_cov, \"gaussian\")\n",
    "        } else {\n",
    "            hrc_topmed_exome_X <- matrix()\n",
    "            result_hrc_topmed_exome$pval <- NA\n",
    "        }\n",
    "\n",
    "        data <- c(\"hrc\", \"topmed\", \"hrc_topmed\", \"hrc_topmed_exome\")\n",
    "        rsqs <- rep(rsq/10, 4)\n",
    "        mafs <- rep(paste0(\"0.0\", maf), 4)\n",
    "        pvals <- c(result_hrc$pval, result_topmed$pval, result_hrc_topmed$pval, result_hrc_topmed_exome$pval)\n",
    "        num_var <- c(ncol(hrc_X), ncol(topmed_X), ncol(hrc_topmed_X), ncol(hrc_topmed_exome_X))\n",
    "        sub_df <- data.frame(data, rsqs, mafs, num_var, pvals)\n",
    "        result_df <- rbind(result_df, sub_df)\n",
    "    }\n",
    "    result_df <- rbind(data.frame(data = \"exome\", rsqs = \"NA\", mafs = paste0(\"0.0\", maf), num_var = ncol(exome_X), pvals = result_exome$pval), result_df)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "147affcc-7c8f-45c2-8884-6937c2f786b5",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 27 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>data</th><th scope=col>rsqs</th><th scope=col>mafs</th><th scope=col>num_var</th><th scope=col>pvals</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.01 </td><td>60</td><td>3.073400e-234</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.01 </td><td> 3</td><td>3.376964e-182</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.01 </td><td> 2</td><td>9.967896e-170</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.01 </td><td>15</td><td>4.292663e-185</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.01 </td><td> 5</td><td>7.820732e-178</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.01 </td><td>15</td><td>2.382023e-188</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.01 </td><td> 6</td><td>1.891710e-187</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.01 </td><td>63</td><td>2.277336e-234</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.01 </td><td>60</td><td>3.073400e-234</td></tr>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.005</td><td>60</td><td>3.073400e-234</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.005</td><td> 3</td><td>3.376964e-182</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.005</td><td> 2</td><td>9.967896e-170</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.005</td><td>15</td><td>4.292663e-185</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.005</td><td> 5</td><td>7.820732e-178</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.005</td><td>15</td><td>2.382023e-188</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.005</td><td> 6</td><td>1.891710e-187</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.005</td><td>63</td><td>2.277336e-234</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.005</td><td>60</td><td>3.073400e-234</td></tr>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.001</td><td>59</td><td> 5.468090e-32</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.001</td><td> 2</td><td> 3.385530e-24</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.001</td><td> 1</td><td>           NA</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.001</td><td>14</td><td> 8.652242e-17</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.001</td><td> 4</td><td> 2.226380e-03</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.001</td><td>14</td><td> 2.375515e-19</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.001</td><td> 4</td><td> 2.226380e-03</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.001</td><td>62</td><td> 2.909126e-32</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.001</td><td>59</td><td> 5.468090e-32</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 27 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " data & rsqs & mafs & num\\_var & pvals\\\\\n",
       " <fct> & <chr> & <chr> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t exome            & NA  & 0.01  & 60 & 3.073400e-234\\\\\n",
       "\t hrc              & 0.3 & 0.01  &  3 & 3.376964e-182\\\\\n",
       "\t hrc              & 0.8 & 0.01  &  2 & 9.967896e-170\\\\\n",
       "\t topmed           & 0.3 & 0.01  & 15 & 4.292663e-185\\\\\n",
       "\t topmed           & 0.8 & 0.01  &  5 & 7.820732e-178\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.01  & 15 & 2.382023e-188\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.01  &  6 & 1.891710e-187\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.01  & 63 & 2.277336e-234\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.01  & 60 & 3.073400e-234\\\\\n",
       "\t exome            & NA  & 0.005 & 60 & 3.073400e-234\\\\\n",
       "\t hrc              & 0.3 & 0.005 &  3 & 3.376964e-182\\\\\n",
       "\t hrc              & 0.8 & 0.005 &  2 & 9.967896e-170\\\\\n",
       "\t topmed           & 0.3 & 0.005 & 15 & 4.292663e-185\\\\\n",
       "\t topmed           & 0.8 & 0.005 &  5 & 7.820732e-178\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.005 & 15 & 2.382023e-188\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.005 &  6 & 1.891710e-187\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.005 & 63 & 2.277336e-234\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.005 & 60 & 3.073400e-234\\\\\n",
       "\t exome            & NA  & 0.001 & 59 &  5.468090e-32\\\\\n",
       "\t hrc              & 0.3 & 0.001 &  2 &  3.385530e-24\\\\\n",
       "\t hrc              & 0.8 & 0.001 &  1 &            NA\\\\\n",
       "\t topmed           & 0.3 & 0.001 & 14 &  8.652242e-17\\\\\n",
       "\t topmed           & 0.8 & 0.001 &  4 &  2.226380e-03\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.001 & 14 &  2.375515e-19\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.001 &  4 &  2.226380e-03\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.001 & 62 &  2.909126e-32\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.001 & 59 &  5.468090e-32\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 27 × 5\n",
       "\n",
       "| data &lt;fct&gt; | rsqs &lt;chr&gt; | mafs &lt;chr&gt; | num_var &lt;int&gt; | pvals &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| exome            | NA  | 0.01  | 60 | 3.073400e-234 |\n",
       "| hrc              | 0.3 | 0.01  |  3 | 3.376964e-182 |\n",
       "| hrc              | 0.8 | 0.01  |  2 | 9.967896e-170 |\n",
       "| topmed           | 0.3 | 0.01  | 15 | 4.292663e-185 |\n",
       "| topmed           | 0.8 | 0.01  |  5 | 7.820732e-178 |\n",
       "| hrc_topmed       | 0.3 | 0.01  | 15 | 2.382023e-188 |\n",
       "| hrc_topmed       | 0.8 | 0.01  |  6 | 1.891710e-187 |\n",
       "| hrc_topmed_exome | 0.3 | 0.01  | 63 | 2.277336e-234 |\n",
       "| hrc_topmed_exome | 0.8 | 0.01  | 60 | 3.073400e-234 |\n",
       "| exome            | NA  | 0.005 | 60 | 3.073400e-234 |\n",
       "| hrc              | 0.3 | 0.005 |  3 | 3.376964e-182 |\n",
       "| hrc              | 0.8 | 0.005 |  2 | 9.967896e-170 |\n",
       "| topmed           | 0.3 | 0.005 | 15 | 4.292663e-185 |\n",
       "| topmed           | 0.8 | 0.005 |  5 | 7.820732e-178 |\n",
       "| hrc_topmed       | 0.3 | 0.005 | 15 | 2.382023e-188 |\n",
       "| hrc_topmed       | 0.8 | 0.005 |  6 | 1.891710e-187 |\n",
       "| hrc_topmed_exome | 0.3 | 0.005 | 63 | 2.277336e-234 |\n",
       "| hrc_topmed_exome | 0.8 | 0.005 | 60 | 3.073400e-234 |\n",
       "| exome            | NA  | 0.001 | 59 |  5.468090e-32 |\n",
       "| hrc              | 0.3 | 0.001 |  2 |  3.385530e-24 |\n",
       "| hrc              | 0.8 | 0.001 |  1 |            NA |\n",
       "| topmed           | 0.3 | 0.001 | 14 |  8.652242e-17 |\n",
       "| topmed           | 0.8 | 0.001 |  4 |  2.226380e-03 |\n",
       "| hrc_topmed       | 0.3 | 0.001 | 14 |  2.375515e-19 |\n",
       "| hrc_topmed       | 0.8 | 0.001 |  4 |  2.226380e-03 |\n",
       "| hrc_topmed_exome | 0.3 | 0.001 | 62 |  2.909126e-32 |\n",
       "| hrc_topmed_exome | 0.8 | 0.001 | 59 |  5.468090e-32 |\n",
       "\n"
      ],
      "text/plain": [
       "   data             rsqs mafs  num_var pvals        \n",
       "1  exome            NA   0.01  60      3.073400e-234\n",
       "2  hrc              0.3  0.01   3      3.376964e-182\n",
       "3  hrc              0.8  0.01   2      9.967896e-170\n",
       "4  topmed           0.3  0.01  15      4.292663e-185\n",
       "5  topmed           0.8  0.01   5      7.820732e-178\n",
       "6  hrc_topmed       0.3  0.01  15      2.382023e-188\n",
       "7  hrc_topmed       0.8  0.01   6      1.891710e-187\n",
       "8  hrc_topmed_exome 0.3  0.01  63      2.277336e-234\n",
       "9  hrc_topmed_exome 0.8  0.01  60      3.073400e-234\n",
       "10 exome            NA   0.005 60      3.073400e-234\n",
       "11 hrc              0.3  0.005  3      3.376964e-182\n",
       "12 hrc              0.8  0.005  2      9.967896e-170\n",
       "13 topmed           0.3  0.005 15      4.292663e-185\n",
       "14 topmed           0.8  0.005  5      7.820732e-178\n",
       "15 hrc_topmed       0.3  0.005 15      2.382023e-188\n",
       "16 hrc_topmed       0.8  0.005  6      1.891710e-187\n",
       "17 hrc_topmed_exome 0.3  0.005 63      2.277336e-234\n",
       "18 hrc_topmed_exome 0.8  0.005 60      3.073400e-234\n",
       "19 exome            NA   0.001 59       5.468090e-32\n",
       "20 hrc              0.3  0.001  2       3.385530e-24\n",
       "21 hrc              0.8  0.001  1                 NA\n",
       "22 topmed           0.3  0.001 14       8.652242e-17\n",
       "23 topmed           0.8  0.001  4       2.226380e-03\n",
       "24 hrc_topmed       0.3  0.001 14       2.375515e-19\n",
       "25 hrc_topmed       0.8  0.001  4       2.226380e-03\n",
       "26 hrc_topmed_exome 0.3  0.001 62       2.909126e-32\n",
       "27 hrc_topmed_exome 0.8  0.001 59       5.468090e-32"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "result_df %>% \n",
    "    mutate(data = factor(data, levels = c(\"exome\", \"hrc\", \"topmed\", \"hrc_topmed\", \"hrc_topmed_exome\"))) %>% \n",
    "    arrange(desc(as.numeric(mafs)), data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "8356218a-8e06-44ba-a005-9055676f8111",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: APOC3_exome_maf001_cadd_extracted.bim\n",
      "\n",
      "Reading: APOC3_exome_maf001_cadd_extracted.fam\n",
      "\n",
      "Reading: APOC3_exome_maf001_cadd_extracted.bed\n",
      "\n",
      "Reading: APOC3_exome_maf0005_cadd_extracted.bim\n",
      "\n",
      "Reading: APOC3_exome_maf0005_cadd_extracted.fam\n",
      "\n",
      "Reading: APOC3_exome_maf0005_cadd_extracted.bed\n",
      "\n",
      "Reading: APOC3_exome_maf0001_cadd_extracted.bim\n",
      "\n",
      "Reading: APOC3_exome_maf0001_cadd_extracted.fam\n",
      "\n",
      "Reading: APOC3_exome_maf0001_cadd_extracted.bed\n",
      "\n"
     ]
    }
   ],
   "source": [
    "result_df <- data.frame(data.frame(matrix(ncol = 5, nrow = 0)))\n",
    "colnames(result_df) <- c(\"data\", \"rsqs\", \"mafs\", \"num_var\", \"pvals\")\n",
    "\n",
    "for(maf in c(\"1\", \"05\", \"01\")){ \n",
    "    exome_bim <- read_bim(sprintf(\"APOC3_exome_maf00%s_cadd_extracted.bim\", maf))\n",
    "    exome_fam <- read_fam(sprintf(\"APOC3_exome_maf00%s_cadd_extracted.fam\", maf))\n",
    "    exome <- read_bed(sprintf(\"APOC3_exome_maf00%s_cadd_extracted.bed\", maf), exome_bim$id, exome_fam$id)\n",
    "    exome_X <- exome %>% t() %>% as.data.frame() %>% tibble::rownames_to_column(\"FID\") %>% \n",
    "            mutate(FID = as.numeric(FID)) %>% arrange(FID) %>% filter(FID %in% tg_id) %>% select(-FID) %>% as.matrix()\n",
    "    result_exome <- BRV(exome_X, tg_y, tg_cov, \"gaussian\")\n",
    "    \n",
    "    for(rsq in c(3, 8)){\n",
    "        hrc_fname <- sprintf(\"APOC3_hrc_rsq0%d_maf00%s_cadd_extracted.traw\", rsq, maf)\n",
    "        topmed_fname <- sprintf(\"APOC3_topmed_rsq0%d_maf00%s_cadd_extracted.traw\", rsq, maf)\n",
    "        hrc_topmed_fname <- sprintf(\"APOC3_hrc_topmed_rsq0%d_maf00%s_cadd_extracted.traw\", rsq, maf)\n",
    "        hrc_topmed_exome_fname <- sprintf(\"APOC3_hrc_topmed_exome_rsq0%d_maf00%s_cadd_extracted.traw\", rsq, maf)\n",
    "\n",
    "        if(file.exists(hrc_fname)){\n",
    "            hrc <- fread(hrc_fname)\n",
    "            hrc_X <- format_traw(hrc, tg_id, \"hrc\")\n",
    "            result_hrc <- BRV(hrc_X, tg_y, tg_cov, \"gaussian\")\n",
    "        } else {\n",
    "            hrc_X <- matrix()\n",
    "            result_hrc$pval <- NA\n",
    "        }\n",
    "        \n",
    "        if(file.exists(topmed_fname)){\n",
    "            topmed <- fread(topmed_fname)\n",
    "            topmed_X <- format_traw(topmed, tg_id, \"topmed\")\n",
    "            result_topmed <- BRV(topmed_X, tg_y, tg_cov, \"gaussian\")\n",
    "        } else {\n",
    "            topmed_X <- matrix()\n",
    "            result_topmed$pval <- NA\n",
    "        }\n",
    "\n",
    "        if(file.exists(hrc_topmed_fname)){\n",
    "            hrc_topmed <- fread(hrc_topmed_fname)\n",
    "            hrc_topmed_X <- format_traw(hrc_topmed, tg_id)\n",
    "            result_hrc_topmed <- BRV(hrc_topmed_X, tg_y, tg_cov, \"gaussian\")\n",
    "        } else {\n",
    "            hrc_topmed_X <- matrix()\n",
    "            result_hrc_topmed$pval <- NA\n",
    "        }\n",
    "\n",
    "        if(file.exists(hrc_topmed_exome_fname)){\n",
    "            hrc_topmed_exome <- fread(hrc_topmed_exome_fname)\n",
    "            hrc_topmed_exome_X <- format_traw(hrc_topmed_exome, tg_id)\n",
    "            result_hrc_topmed_exome <- BRV(hrc_topmed_exome_X, tg_y, tg_cov, \"gaussian\")\n",
    "        } else {\n",
    "            hrc_topmed_exome_X <- matrix()\n",
    "            result_hrc_topmed_exome$pval <- NA\n",
    "        }\n",
    "\n",
    "        data <- c(\"hrc\", \"topmed\", \"hrc_topmed\", \"hrc_topmed_exome\")\n",
    "        rsqs <- rep(rsq/10, 4)\n",
    "        mafs <- rep(paste0(\"0.0\", maf), 4)\n",
    "        pvals <- c(result_hrc$pval, result_topmed$pval, result_hrc_topmed$pval, result_hrc_topmed_exome$pval)\n",
    "        num_var <- c(ncol(hrc_X), ncol(topmed_X), ncol(hrc_topmed_X), ncol(hrc_topmed_exome_X))\n",
    "        sub_df <- data.frame(data, rsqs, mafs, num_var, pvals)\n",
    "        result_df <- rbind(result_df, sub_df)\n",
    "    }\n",
    "    result_df <- rbind(data.frame(data = \"exome\", rsqs = \"NA\", mafs = paste0(\"0.0\", maf), num_var = ncol(exome_X), pvals = result_exome$pval), result_df)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "9abc5e21-0b9f-46d0-a721-6b14d25f1b23",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 27 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>data</th><th scope=col>rsqs</th><th scope=col>mafs</th><th scope=col>num_var</th><th scope=col>pvals</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.01 </td><td>26</td><td>1.050377e-262</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.01 </td><td> 3</td><td>3.376964e-182</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.01 </td><td> 2</td><td>9.967896e-170</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.01 </td><td> 9</td><td>1.220032e-190</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.01 </td><td> 4</td><td>1.304211e-179</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.01 </td><td> 9</td><td>7.033698e-194</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.01 </td><td> 5</td><td>4.544786e-189</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.01 </td><td>26</td><td>1.050377e-262</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.01 </td><td>26</td><td>1.050377e-262</td></tr>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.005</td><td>26</td><td>1.050377e-262</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.005</td><td> 3</td><td>3.376964e-182</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.005</td><td> 2</td><td>9.967896e-170</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.005</td><td> 9</td><td>1.220032e-190</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.005</td><td> 4</td><td>1.304211e-179</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.005</td><td> 9</td><td>7.033698e-194</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.005</td><td> 5</td><td>4.544786e-189</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.005</td><td>26</td><td>1.050377e-262</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.005</td><td>26</td><td>1.050377e-262</td></tr>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.001</td><td>25</td><td> 9.616993e-48</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.001</td><td> 2</td><td> 3.385530e-24</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.001</td><td> 1</td><td>           NA</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.001</td><td> 8</td><td> 8.245356e-19</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.001</td><td> 3</td><td> 3.530366e-04</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.001</td><td> 8</td><td> 1.413871e-21</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.001</td><td> 3</td><td> 3.530366e-04</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.001</td><td>25</td><td> 9.616993e-48</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.001</td><td>25</td><td> 9.616993e-48</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 27 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " data & rsqs & mafs & num\\_var & pvals\\\\\n",
       " <fct> & <chr> & <chr> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t exome            & NA  & 0.01  & 26 & 1.050377e-262\\\\\n",
       "\t hrc              & 0.3 & 0.01  &  3 & 3.376964e-182\\\\\n",
       "\t hrc              & 0.8 & 0.01  &  2 & 9.967896e-170\\\\\n",
       "\t topmed           & 0.3 & 0.01  &  9 & 1.220032e-190\\\\\n",
       "\t topmed           & 0.8 & 0.01  &  4 & 1.304211e-179\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.01  &  9 & 7.033698e-194\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.01  &  5 & 4.544786e-189\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.01  & 26 & 1.050377e-262\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.01  & 26 & 1.050377e-262\\\\\n",
       "\t exome            & NA  & 0.005 & 26 & 1.050377e-262\\\\\n",
       "\t hrc              & 0.3 & 0.005 &  3 & 3.376964e-182\\\\\n",
       "\t hrc              & 0.8 & 0.005 &  2 & 9.967896e-170\\\\\n",
       "\t topmed           & 0.3 & 0.005 &  9 & 1.220032e-190\\\\\n",
       "\t topmed           & 0.8 & 0.005 &  4 & 1.304211e-179\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.005 &  9 & 7.033698e-194\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.005 &  5 & 4.544786e-189\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.005 & 26 & 1.050377e-262\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.005 & 26 & 1.050377e-262\\\\\n",
       "\t exome            & NA  & 0.001 & 25 &  9.616993e-48\\\\\n",
       "\t hrc              & 0.3 & 0.001 &  2 &  3.385530e-24\\\\\n",
       "\t hrc              & 0.8 & 0.001 &  1 &            NA\\\\\n",
       "\t topmed           & 0.3 & 0.001 &  8 &  8.245356e-19\\\\\n",
       "\t topmed           & 0.8 & 0.001 &  3 &  3.530366e-04\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.001 &  8 &  1.413871e-21\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.001 &  3 &  3.530366e-04\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.001 & 25 &  9.616993e-48\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.001 & 25 &  9.616993e-48\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 27 × 5\n",
       "\n",
       "| data &lt;fct&gt; | rsqs &lt;chr&gt; | mafs &lt;chr&gt; | num_var &lt;int&gt; | pvals &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| exome            | NA  | 0.01  | 26 | 1.050377e-262 |\n",
       "| hrc              | 0.3 | 0.01  |  3 | 3.376964e-182 |\n",
       "| hrc              | 0.8 | 0.01  |  2 | 9.967896e-170 |\n",
       "| topmed           | 0.3 | 0.01  |  9 | 1.220032e-190 |\n",
       "| topmed           | 0.8 | 0.01  |  4 | 1.304211e-179 |\n",
       "| hrc_topmed       | 0.3 | 0.01  |  9 | 7.033698e-194 |\n",
       "| hrc_topmed       | 0.8 | 0.01  |  5 | 4.544786e-189 |\n",
       "| hrc_topmed_exome | 0.3 | 0.01  | 26 | 1.050377e-262 |\n",
       "| hrc_topmed_exome | 0.8 | 0.01  | 26 | 1.050377e-262 |\n",
       "| exome            | NA  | 0.005 | 26 | 1.050377e-262 |\n",
       "| hrc              | 0.3 | 0.005 |  3 | 3.376964e-182 |\n",
       "| hrc              | 0.8 | 0.005 |  2 | 9.967896e-170 |\n",
       "| topmed           | 0.3 | 0.005 |  9 | 1.220032e-190 |\n",
       "| topmed           | 0.8 | 0.005 |  4 | 1.304211e-179 |\n",
       "| hrc_topmed       | 0.3 | 0.005 |  9 | 7.033698e-194 |\n",
       "| hrc_topmed       | 0.8 | 0.005 |  5 | 4.544786e-189 |\n",
       "| hrc_topmed_exome | 0.3 | 0.005 | 26 | 1.050377e-262 |\n",
       "| hrc_topmed_exome | 0.8 | 0.005 | 26 | 1.050377e-262 |\n",
       "| exome            | NA  | 0.001 | 25 |  9.616993e-48 |\n",
       "| hrc              | 0.3 | 0.001 |  2 |  3.385530e-24 |\n",
       "| hrc              | 0.8 | 0.001 |  1 |            NA |\n",
       "| topmed           | 0.3 | 0.001 |  8 |  8.245356e-19 |\n",
       "| topmed           | 0.8 | 0.001 |  3 |  3.530366e-04 |\n",
       "| hrc_topmed       | 0.3 | 0.001 |  8 |  1.413871e-21 |\n",
       "| hrc_topmed       | 0.8 | 0.001 |  3 |  3.530366e-04 |\n",
       "| hrc_topmed_exome | 0.3 | 0.001 | 25 |  9.616993e-48 |\n",
       "| hrc_topmed_exome | 0.8 | 0.001 | 25 |  9.616993e-48 |\n",
       "\n"
      ],
      "text/plain": [
       "   data             rsqs mafs  num_var pvals        \n",
       "1  exome            NA   0.01  26      1.050377e-262\n",
       "2  hrc              0.3  0.01   3      3.376964e-182\n",
       "3  hrc              0.8  0.01   2      9.967896e-170\n",
       "4  topmed           0.3  0.01   9      1.220032e-190\n",
       "5  topmed           0.8  0.01   4      1.304211e-179\n",
       "6  hrc_topmed       0.3  0.01   9      7.033698e-194\n",
       "7  hrc_topmed       0.8  0.01   5      4.544786e-189\n",
       "8  hrc_topmed_exome 0.3  0.01  26      1.050377e-262\n",
       "9  hrc_topmed_exome 0.8  0.01  26      1.050377e-262\n",
       "10 exome            NA   0.005 26      1.050377e-262\n",
       "11 hrc              0.3  0.005  3      3.376964e-182\n",
       "12 hrc              0.8  0.005  2      9.967896e-170\n",
       "13 topmed           0.3  0.005  9      1.220032e-190\n",
       "14 topmed           0.8  0.005  4      1.304211e-179\n",
       "15 hrc_topmed       0.3  0.005  9      7.033698e-194\n",
       "16 hrc_topmed       0.8  0.005  5      4.544786e-189\n",
       "17 hrc_topmed_exome 0.3  0.005 26      1.050377e-262\n",
       "18 hrc_topmed_exome 0.8  0.005 26      1.050377e-262\n",
       "19 exome            NA   0.001 25       9.616993e-48\n",
       "20 hrc              0.3  0.001  2       3.385530e-24\n",
       "21 hrc              0.8  0.001  1                 NA\n",
       "22 topmed           0.3  0.001  8       8.245356e-19\n",
       "23 topmed           0.8  0.001  3       3.530366e-04\n",
       "24 hrc_topmed       0.3  0.001  8       1.413871e-21\n",
       "25 hrc_topmed       0.8  0.001  3       3.530366e-04\n",
       "26 hrc_topmed_exome 0.3  0.001 25       9.616993e-48\n",
       "27 hrc_topmed_exome 0.8  0.001 25       9.616993e-48"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "result_df %>% \n",
    "    mutate(data = factor(data, levels = c(\"exome\", \"hrc\", \"topmed\", \"hrc_topmed\", \"hrc_topmed_exome\"))) %>% \n",
    "    arrange(desc(as.numeric(mafs)), data)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c758871c-4b96-41dc-b9ff-cd3d9932e91b",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "source": [
    "#### APOC3: chr11:116830638\n",
    "\n",
    "We specifically looked into this variant, for the same reason we did for the PCSK9 variant"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "e6045d37-c525-4f6c-a38d-c6724ce1e336",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading: APOC3_exome_maf001_extracted.bim\n",
      "\n",
      "Reading: APOC3_exome_maf001_extracted.fam\n",
      "\n",
      "Reading: APOC3_exome_maf001_extracted.bed\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "2.83630024818732e-219"
      ],
      "text/latex": [
       "2.83630024818732e-219"
      ],
      "text/markdown": [
       "2.83630024818732e-219"
      ],
      "text/plain": [
       "[1] 2.8363e-219"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "exome_bim <- read_bim(\"APOC3_exome_maf001_extracted.bim\")\n",
    "exome_fam <- read_fam(\"APOC3_exome_maf001_extracted.fam\")\n",
    "exome <- read_bed(\"APOC3_exome_maf001_extracted.bed\", exome_bim$id, exome_fam$id)\n",
    "\n",
    "exome_X <- data.frame(ID = names(exome[\"chr11:116830638:G:A\",]), value=exome[\"chr11:116830638:G:A\",], row.names=NULL)\n",
    "exome_X <- exome_X %>% arrange(as.numeric(ID)) %>% filter(ID %in% tg_id) %>% select(-ID) %>% as.matrix()\n",
    "\n",
    "exome.fit <- glm(tg_y ~ exome_X + tg_cov$age + tg_cov$sex + tg_cov$PC1 + tg_cov$PC2, family = \"gaussian\")\n",
    "coef(summary(exome.fit))[2, 4]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "0b50c929-924f-4f0d-882d-1594c5a6f2a2",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 17</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Chr</th><th scope=col>Start</th><th scope=col>End</th><th scope=col>Ref</th><th scope=col>Alt</th><th scope=col>Func.refGene</th><th scope=col>Gene.refGene</th><th scope=col>ExonicFunc.refGene</th><th scope=col>MAF_nfe_exome</th><th scope=col>REVEL_score</th><th scope=col>ID_hg38</th><th scope=col>ID_hg19</th><th scope=col>ID</th><th scope=col>Function</th><th scope=col>RawScore</th><th scope=col>PHRED</th><th scope=col>R2</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>11</td><td>116830638</td><td>116830638</td><td>G</td><td>A</td><td>splicing</td><td>APOC3</td><td>.</td><td>0.0023</td><td>.</td><td>chr11:116830638:G:A</td><td>chr11:116701354:G:A</td><td>chr11:116830638:G:A</td><td>splicing</td><td>5.166115</td><td>33</td><td>0.836202</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 17\n",
       "\\begin{tabular}{lllllllllllllllll}\n",
       " Chr & Start & End & Ref & Alt & Func.refGene & Gene.refGene & ExonicFunc.refGene & MAF\\_nfe\\_exome & REVEL\\_score & ID\\_hg38 & ID\\_hg19 & ID & Function & RawScore & PHRED & R2\\\\\n",
       " <int> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t 11 & 116830638 & 116830638 & G & A & splicing & APOC3 & . & 0.0023 & . & chr11:116830638:G:A & chr11:116701354:G:A & chr11:116830638:G:A & splicing & 5.166115 & 33 & 0.836202\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 17\n",
       "\n",
       "| Chr &lt;int&gt; | Start &lt;int&gt; | End &lt;int&gt; | Ref &lt;chr&gt; | Alt &lt;chr&gt; | Func.refGene &lt;chr&gt; | Gene.refGene &lt;chr&gt; | ExonicFunc.refGene &lt;chr&gt; | MAF_nfe_exome &lt;dbl&gt; | REVEL_score &lt;chr&gt; | ID_hg38 &lt;chr&gt; | ID_hg19 &lt;chr&gt; | ID &lt;chr&gt; | Function &lt;chr&gt; | RawScore &lt;dbl&gt; | PHRED &lt;dbl&gt; | R2 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 11 | 116830638 | 116830638 | G | A | splicing | APOC3 | . | 0.0023 | . | chr11:116830638:G:A | chr11:116701354:G:A | chr11:116830638:G:A | splicing | 5.166115 | 33 | 0.836202 |\n",
       "\n"
      ],
      "text/plain": [
       "  Chr Start     End       Ref Alt Func.refGene Gene.refGene ExonicFunc.refGene\n",
       "1 11  116830638 116830638 G   A   splicing     APOC3        .                 \n",
       "  MAF_nfe_exome REVEL_score ID_hg38             ID_hg19            \n",
       "1 0.0023        .           chr11:116830638:G:A chr11:116701354:G:A\n",
       "  ID                  Function RawScore PHRED R2      \n",
       "1 chr11:116830638:G:A splicing 5.166115 33    0.836202"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "1.1744304895572e-161"
      ],
      "text/latex": [
       "1.1744304895572e-161"
      ],
      "text/markdown": [
       "1.1744304895572e-161"
      ],
      "text/plain": [
       "[1] 1.17443e-161"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fread(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/hrc_topmed/hrc_168206ids_chr11_rsq03_maf001_annot.csv.gz\") %>% \n",
    "    filter(ID_hg19 == \"chr11:116701354:G:A\")\n",
    "\n",
    "hrc <- fread(\"APOC3_hrc_rsq03_maf001_extracted.traw\") %>% filter(SNP == \"chr11:116701354:G:A\")\n",
    "hrc_X <- format_traw(hrc, tg_id, \"hrc\")\n",
    "\n",
    "hrc.fit <- glm(tg_y ~ hrc_X + tg_cov$age + tg_cov$sex + tg_cov$PC1 + tg_cov$PC2, family = \"gaussian\")\n",
    "coef(summary(hrc.fit))[2, 4]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "c481aab3-a1bb-4ebe-a765-1e7203e42392",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 17</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Chr</th><th scope=col>Start</th><th scope=col>End</th><th scope=col>Ref</th><th scope=col>Alt</th><th scope=col>Func.refGene</th><th scope=col>Gene.refGene</th><th scope=col>ExonicFunc.refGene</th><th scope=col>MAF_nfe_exome</th><th scope=col>REVEL_score</th><th scope=col>ID_hg38</th><th scope=col>ID_hg19</th><th scope=col>ID</th><th scope=col>Function</th><th scope=col>RawScore</th><th scope=col>PHRED</th><th scope=col>R2</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>11</td><td>116830638</td><td>116830638</td><td>G</td><td>A</td><td>splicing</td><td>APOC3</td><td>.</td><td>0.0023</td><td>.</td><td>chr11:116830638:G:A</td><td>chr11:116830638:G:A</td><td>chr11:116830638:G:A</td><td>splicing</td><td>4.947405</td><td>33</td><td>0.859247</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 17\n",
       "\\begin{tabular}{lllllllllllllllll}\n",
       " Chr & Start & End & Ref & Alt & Func.refGene & Gene.refGene & ExonicFunc.refGene & MAF\\_nfe\\_exome & REVEL\\_score & ID\\_hg38 & ID\\_hg19 & ID & Function & RawScore & PHRED & R2\\\\\n",
       " <int> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t 11 & 116830638 & 116830638 & G & A & splicing & APOC3 & . & 0.0023 & . & chr11:116830638:G:A & chr11:116830638:G:A & chr11:116830638:G:A & splicing & 4.947405 & 33 & 0.859247\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 17\n",
       "\n",
       "| Chr &lt;int&gt; | Start &lt;int&gt; | End &lt;int&gt; | Ref &lt;chr&gt; | Alt &lt;chr&gt; | Func.refGene &lt;chr&gt; | Gene.refGene &lt;chr&gt; | ExonicFunc.refGene &lt;chr&gt; | MAF_nfe_exome &lt;dbl&gt; | REVEL_score &lt;chr&gt; | ID_hg38 &lt;chr&gt; | ID_hg19 &lt;chr&gt; | ID &lt;chr&gt; | Function &lt;chr&gt; | RawScore &lt;dbl&gt; | PHRED &lt;dbl&gt; | R2 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 11 | 116830638 | 116830638 | G | A | splicing | APOC3 | . | 0.0023 | . | chr11:116830638:G:A | chr11:116830638:G:A | chr11:116830638:G:A | splicing | 4.947405 | 33 | 0.859247 |\n",
       "\n"
      ],
      "text/plain": [
       "  Chr Start     End       Ref Alt Func.refGene Gene.refGene ExonicFunc.refGene\n",
       "1 11  116830638 116830638 G   A   splicing     APOC3        .                 \n",
       "  MAF_nfe_exome REVEL_score ID_hg38             ID_hg19            \n",
       "1 0.0023        .           chr11:116830638:G:A chr11:116830638:G:A\n",
       "  ID                  Function RawScore PHRED R2      \n",
       "1 chr11:116830638:G:A splicing 4.947405 33    0.859247"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "4.21312932485019e-177"
      ],
      "text/latex": [
       "4.21312932485019e-177"
      ],
      "text/markdown": [
       "4.21312932485019e-177"
      ],
      "text/plain": [
       "[1] 4.213129e-177"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fread(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/hrc_topmed/topmed_168206ids_chr11_rsq03_maf001_annot.csv.gz\") %>% \n",
    "    filter(ID_hg38 == \"chr11:116830638:G:A\")\n",
    "\n",
    "topmed <- fread(\"APOC3_topmed_rsq03_maf001_extracted.traw\") %>% filter(SNP == \"chr11:116830638:G:A\")\n",
    "topmed_X <- format_traw(topmed, tg_id, \"topmed\")\n",
    "\n",
    "topmed.fit <- glm(tg_y ~ topmed_X + tg_cov$age + tg_cov$sex + tg_cov$PC1 + tg_cov$PC2, family = \"gaussian\")\n",
    "coef(summary(topmed.fit))[2,4]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "997af33d-49a2-4f40-9ec0-76220a55bbfd",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8aa498dc-4d4a-4f35-b696-d6a1620d2a68",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a0e55034-eca9-4dc4-be71-5d45ef8e7bbd",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "049fb1a8-a706-45d3-924f-307db6ab7c6d",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65d1ae8a-10e8-4cdc-90c4-45f7610f30da",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "SoS",
   "language": "sos",
   "name": "sos"
  },
  "language_info": {
   "codemirror_mode": "sos",
   "file_extension": ".sos",
   "mimetype": "text/x-sos",
   "name": "sos",
   "nbconvert_exporter": "sos_notebook.converter.SoS_Exporter",
   "pygments_lexer": "sos"
  },
  "sos": {
   "kernels": [
    [
     "Bash",
     "bash",
     "Bash",
     "#E6EEFF",
     "shell"
    ],
    [
     "R",
     "ir",
     "R",
     "#DCDCDA",
     "r"
    ]
   ],
   "version": "0.24.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
