{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "3388eb60-271b-492f-b015-05d1b081691f",
   "metadata": {
    "kernel": "SoS",
    "tags": []
   },
   "source": [
    "# PCSK9 vs. APOC3\n",
    "\n",
    "This notebook records [BRV](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4459641/) analysis result for PCSK9 v.s LDL and APOC3 v.s. TG"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b486533c-8a01-4884-b060-3830c23c6116",
   "metadata": {
    "kernel": "SoS",
    "tags": []
   },
   "source": [
    "## Extract PCSK9 + APOC3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "76dea2bc-5912-4425-8fec-80915b7a7a5e",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(dplyr)\n",
    "library(data.table)\n",
    "setwd(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c0c78a1b-dbde-4be9-8cac-2a2f9ca1dc04",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### Exome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e8ac29ed-6226-4947-bcce-c9cf66c0a3b3",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "for(chr in c(1,11)){\n",
    "    gene <- ifelse(chr==1, \"PCSK9\", \"APOC3\")\n",
    "    \n",
    "    for(maf in c(0.01, 0.005, 0.001)){\n",
    "        maf_c <- gsub(\"\\\\.\", \"\", as.character(maf))\n",
    "        \n",
    "        annot <- read.csv(sprintf(\"./exome/ukb23156_c%i.merged.filtered.hg38.hg38_multianno_formatted_sel_col_maf%s_LOF_missense.csv.gz\", chr, maf_c)) %>% \n",
    "            filter(Gene.refGene == gene)\n",
    "        annot %>% fwrite(sprintf(\"./%s/%s_exome_maf%s_annot.csv.gz\", tolower(gene), gene, maf_c))\n",
    "        annot %>% select(ID_hg38) %>% fwrite(sprintf(\"./%s/%s_exome_maf%s_extractlist_snplist\", tolower(gene), gene, maf_c), col.names = FALSE)\n",
    "        \n",
    "        annot_cadd <- read.csv(sprintf(\"./exome/ukb23156_c%i.merged.filtered.hg38.hg38_multianno_formatted_sel_col_maf%s_LOF_missense_cadd.csv.gz\", chr, maf_c)) %>% \n",
    "            filter(Gene.refGene == gene)\n",
    "        annot_cadd %>% fwrite(sprintf(\"./%s/%s_exome_maf%s_cadd_annot.csv.gz\", tolower(gene), gene, maf_c))\n",
    "        annot_cadd %>% select(ID_hg38) %>% fwrite(sprintf(\"./%s/%s_exome_maf%s_cadd_extractlist_snplist\", tolower(gene), gene, maf_c), col.names = FALSE)\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "136a6c79-3924-4bc0-be27-e15c446bac07",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### HRC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "8af5537b-0624-4548-8272-297b75b0b900",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "for(chr in c(1,11)){\n",
    "    gene <- ifelse(chr==1, \"PCSK9\", \"APOC3\")\n",
    "    \n",
    "    for(rsq in c(3, 8)){\n",
    "        for(maf in c(0.01, 0.005, 0.001)){\n",
    "            maf_c <- gsub(\"\\\\.\", \"\", as.character(maf))\n",
    "\n",
    "            annot <- read.csv(sprintf(\"./hrc/hrc_chr%i_rsq0%i_hg19_hg38_maf%s_LOF_missense_annot.csv.gz\", chr, rsq, maf_c)) %>% \n",
    "                filter(Gene.refGene == gene)\n",
    "            annot %>% fwrite(sprintf(\"./%s/%s_hrc_rsq0%i_maf%s_annot.csv.gz\", tolower(gene), gene, rsq, maf_c))\n",
    "            annot %>% select(ID_hg19) %>% fwrite(sprintf(\"./%s/%s_hrc_rsq0%i_maf%s_extractlist_snplist\", tolower(gene), gene, rsq, maf_c), col.names = FALSE)\n",
    "        }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "08d7d176-1f71-4412-8468-3a00a6d8e0d7",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "for(chr in c(1, 11)){\n",
    "    gene <- ifelse(chr==1, \"PCSK9\", \"APOC3\")\n",
    "    \n",
    "    for(rsq in c(3, 8)){\n",
    "        for(maf in c(0.01, 0.005, 0.001)){\n",
    "            maf_c <- gsub(\"\\\\.\", \"\", as.character(maf))\n",
    "\n",
    "            annot_cadd <- read.csv(sprintf(\"./hrc/hrc_chr%i_rsq0%i_hg19_hg38_maf%s_LOF_missense_cadd_annot.csv.gz\", chr, rsq, maf_c)) %>% \n",
    "                filter(Gene.refGene == gene)\n",
    "            annot_cadd %>% fwrite(sprintf(\"./%s/%s_hrc_rsq0%i_maf%s_cadd_annot.csv.gz\", tolower(gene), gene, rsq, maf_c))\n",
    "            annot_cadd %>% select(ID_hg19) %>% fwrite(sprintf(\"./%s/%s_hrc_rsq0%i_maf%s_cadd_extractlist_snplist\", tolower(gene), gene, rsq, maf_c), col.names = FALSE)\n",
    "        }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "36daeda2-8c72-4161-8569-6f5b35e3b234",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### TOPMed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "f6c94c59-734f-47ee-89e0-d451136cb573",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "for(chr in c(1, 11)){\n",
    "    gene <- ifelse(chr==1, \"PCSK9\", \"APOC3\")\n",
    "    \n",
    "    for(rsq in c(3, 8)){\n",
    "        for(maf in c(0.01, 0.005, 0.001)){\n",
    "            maf_c <- gsub(\"\\\\.\", \"\", as.character(maf))\n",
    "\n",
    "            annot <- read.csv(sprintf(\"./topmed_v3/topmed_chr%i_rsq0%i_hg38_hg38_maf%s_LOF_missense_annot.csv.gz\", chr, rsq, maf_c)) %>% \n",
    "                filter(Gene.refGene == gene)\n",
    "            annot %>% fwrite(sprintf(\"./%s/%s_topmed_v3_rsq0%i_maf%s_annot.csv.gz\", tolower(gene), gene, rsq, maf_c))\n",
    "            annot %>% select(ID_hg38) %>% fwrite(sprintf(\"./%s/%s_topmed_v3_rsq0%i_maf%s_extractlist_snplist\", tolower(gene), gene, rsq, maf_c), col.names = FALSE)\n",
    "        }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "42536aeb-453f-4600-8935-0ad8479a53c9",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "for(chr in c(1, 11)){\n",
    "    gene <- ifelse(chr==1, \"PCSK9\", \"APOC3\")\n",
    "    for(rsq in c(3, 8)){\n",
    "        for(maf in c(0.01, 0.005, 0.001)){\n",
    "            maf_c <- gsub(\"\\\\.\", \"\", as.character(maf))\n",
    "\n",
    "            annot_cadd <- read.csv(sprintf(\"./topmed_v3/topmed_chr%i_rsq0%i_hg38_hg38_maf%s_LOF_missense_cadd_annot.csv.gz\", chr, rsq, maf_c)) %>% \n",
    "                filter(Gene.refGene == gene)\n",
    "            annot_cadd %>% fwrite(sprintf(\"./%s/%s_topmed_v3_rsq0%i_maf%s_cadd_annot.csv.gz\", tolower(gene), gene, rsq, maf_c))\n",
    "            annot_cadd %>% select(ID_hg38) %>% fwrite(sprintf(\"./%s/%s_topmed_v3_rsq0%i_maf%s_cadd_extractlist_snplist\", tolower(gene), gene, rsq, maf_c), col.names = FALSE)\n",
    "        }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f1f3d44f-3483-465c-9cc3-e9ac54d7dd88",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "### Extract"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "14718091-0b4f-4460-b2ed-323515e32bdc",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_exome_maf001_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c1_maf001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_exome_maf001_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out PCSK9_exome_maf001_extracted\n",
      "\n",
      "Start time: Fri Jul  5 13:50:10 2024\n",
      "515666 MiB RAM detected, ~490443 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c1_maf001_LOF_missense_extracted.fam.\n",
      "478357 variants loaded from\n",
      "../exome/ukb23156_c1_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 394 variants remaining.\n",
      "394 variants remaining after main filters.\n",
      "Writing PCSK9_exome_maf001_extracted.fam ... done.\n",
      "Writing PCSK9_exome_maf001_extracted.bim ... done.\n",
      "Writing PCSK9_exome_maf001_extracted.bed ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_exome_maf001_extracted.raw written.\n",
      "End time: Fri Jul  5 13:50:20 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq03_maf001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq03_maf001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_hrc_rsq03_maf001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq03_maf001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:50:20 2024\n",
      "515666 MiB RAM detected, ~490429 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq03_maf001_LOF_missense_extracted.fam.\n",
      "20551 variants loaded from\n",
      "../hrc/hrc_chr1_rsq03_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 16 variants remaining.\n",
      "16 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq03_maf001_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf001_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf001_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_hrc_rsq03_maf001_extracted.raw written.\n",
      "End time: Fri Jul  5 13:50:22 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_v3_rsq03_maf001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/topmed_chr1_rsq03_maf001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_topmed_v3_rsq03_maf001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_v3_rsq03_maf001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:50:22 2024\n",
      "515666 MiB RAM detected, ~490420 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed_v3/topmed_chr1_rsq03_maf001_LOF_missense_extracted.fam.\n",
      "141967 variants loaded from\n",
      "../topmed_v3/topmed_chr1_rsq03_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 124 variants remaining.\n",
      "124 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_v3_rsq03_maf001_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_v3_rsq03_maf001_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_v3_rsq03_maf001_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_topmed_v3_rsq03_maf001_extracted.raw written.\n",
      "End time: Fri Jul  5 13:50:22 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq08_maf001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq08_maf001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_hrc_rsq08_maf001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq08_maf001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:50:22 2024\n",
      "515666 MiB RAM detected, ~490416 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq08_maf001_LOF_missense_extracted.fam.\n",
      "8396 variants loaded from\n",
      "../hrc/hrc_chr1_rsq08_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 6 variants remaining.\n",
      "6 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq08_maf001_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf001_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf001_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_hrc_rsq08_maf001_extracted.raw written.\n",
      "End time: Fri Jul  5 13:50:23 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_v3_rsq08_maf001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/topmed_chr1_rsq08_maf001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_topmed_v3_rsq08_maf001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_v3_rsq08_maf001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:50:23 2024\n",
      "515666 MiB RAM detected, ~490411 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed_v3/topmed_chr1_rsq08_maf001_LOF_missense_extracted.fam.\n",
      "45669 variants loaded from\n",
      "../topmed_v3/topmed_chr1_rsq08_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 34 variants remaining.\n",
      "34 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_v3_rsq08_maf001_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_v3_rsq08_maf001_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_v3_rsq08_maf001_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_topmed_v3_rsq08_maf001_extracted.raw written.\n",
      "End time: Fri Jul  5 13:50:25 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_exome_maf0005_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c1_maf0005_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_exome_maf0005_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out PCSK9_exome_maf0005_extracted\n",
      "\n",
      "Start time: Fri Jul  5 13:50:25 2024\n",
      "515666 MiB RAM detected, ~490402 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c1_maf0005_LOF_missense_extracted.fam.\n",
      "477452 variants loaded from\n",
      "../exome/ukb23156_c1_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 394 variants remaining.\n",
      "394 variants remaining after main filters.\n",
      "Writing PCSK9_exome_maf0005_extracted.fam ... done.\n",
      "Writing PCSK9_exome_maf0005_extracted.bim ... done.\n",
      "Writing PCSK9_exome_maf0005_extracted.bed ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_exome_maf0005_extracted.raw written.\n",
      "End time: Fri Jul  5 13:50:35 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq03_maf0005_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq03_maf0005_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_hrc_rsq03_maf0005_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq03_maf0005_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:50:35 2024\n",
      "515666 MiB RAM detected, ~490351 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq03_maf0005_LOF_missense_extracted.fam.\n",
      "19743 variants loaded from\n",
      "../hrc/hrc_chr1_rsq03_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 16 variants remaining.\n",
      "16 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq03_maf0005_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf0005_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf0005_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_hrc_rsq03_maf0005_extracted.raw written.\n",
      "End time: Fri Jul  5 13:50:37 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_v3_rsq03_maf0005_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/topmed_chr1_rsq03_maf0005_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_topmed_v3_rsq03_maf0005_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_v3_rsq03_maf0005_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:50:37 2024\n",
      "515666 MiB RAM detected, ~490345 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed_v3/topmed_chr1_rsq03_maf0005_LOF_missense_extracted.fam.\n",
      "141141 variants loaded from\n",
      "../topmed_v3/topmed_chr1_rsq03_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 124 variants remaining.\n",
      "124 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_v3_rsq03_maf0005_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_v3_rsq03_maf0005_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_v3_rsq03_maf0005_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_topmed_v3_rsq03_maf0005_extracted.raw written.\n",
      "End time: Fri Jul  5 13:50:39 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq08_maf0005_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq08_maf0005_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_hrc_rsq08_maf0005_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq08_maf0005_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:50:40 2024\n",
      "515666 MiB RAM detected, ~490333 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq08_maf0005_LOF_missense_extracted.fam.\n",
      "7666 variants loaded from\n",
      "../hrc/hrc_chr1_rsq08_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 6 variants remaining.\n",
      "6 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq08_maf0005_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf0005_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf0005_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_hrc_rsq08_maf0005_extracted.raw written.\n",
      "End time: Fri Jul  5 13:50:41 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_v3_rsq08_maf0005_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/topmed_chr1_rsq08_maf0005_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_topmed_v3_rsq08_maf0005_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_v3_rsq08_maf0005_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:50:41 2024\n",
      "515666 MiB RAM detected, ~490326 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed_v3/topmed_chr1_rsq08_maf0005_LOF_missense_extracted.fam.\n",
      "44886 variants loaded from\n",
      "../topmed_v3/topmed_chr1_rsq08_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 34 variants remaining.\n",
      "34 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_v3_rsq08_maf0005_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_v3_rsq08_maf0005_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_v3_rsq08_maf0005_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_topmed_v3_rsq08_maf0005_extracted.raw written.\n",
      "End time: Fri Jul  5 13:50:43 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_exome_maf0001_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c1_maf0001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_exome_maf0001_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out PCSK9_exome_maf0001_extracted\n",
      "\n",
      "Start time: Fri Jul  5 13:50:43 2024\n",
      "515666 MiB RAM detected, ~490319 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c1_maf0001_LOF_missense_extracted.fam.\n",
      "473986 variants loaded from\n",
      "../exome/ukb23156_c1_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 393 variants remaining.\n",
      "393 variants remaining after main filters.\n",
      "Writing PCSK9_exome_maf0001_extracted.fam ... done.\n",
      "Writing PCSK9_exome_maf0001_extracted.bim ... done.\n",
      "Writing PCSK9_exome_maf0001_extracted.bed ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_exome_maf0001_extracted.raw written.\n",
      "End time: Fri Jul  5 13:50:52 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq03_maf0001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq03_maf0001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_hrc_rsq03_maf0001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq03_maf0001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:50:52 2024\n",
      "515666 MiB RAM detected, ~490399 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq03_maf0001_LOF_missense_extracted.fam.\n",
      "16591 variants loaded from\n",
      "../hrc/hrc_chr1_rsq03_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 15 variants remaining.\n",
      "15 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq03_maf0001_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf0001_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf0001_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_hrc_rsq03_maf0001_extracted.raw written.\n",
      "End time: Fri Jul  5 13:50:53 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_v3_rsq03_maf0001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/topmed_chr1_rsq03_maf0001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_topmed_v3_rsq03_maf0001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_v3_rsq03_maf0001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:50:53 2024\n",
      "515666 MiB RAM detected, ~490392 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed_v3/topmed_chr1_rsq03_maf0001_LOF_missense_extracted.fam.\n",
      "137927 variants loaded from\n",
      "../topmed_v3/topmed_chr1_rsq03_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 123 variants remaining.\n",
      "123 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_v3_rsq03_maf0001_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_v3_rsq03_maf0001_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_v3_rsq03_maf0001_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_topmed_v3_rsq03_maf0001_extracted.raw written.\n",
      "End time: Fri Jul  5 13:50:56 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq08_maf0001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq08_maf0001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_hrc_rsq08_maf0001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq08_maf0001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:50:56 2024\n",
      "515666 MiB RAM detected, ~490377 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq08_maf0001_LOF_missense_extracted.fam.\n",
      "5243 variants loaded from\n",
      "../hrc/hrc_chr1_rsq08_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 6 variants remaining.\n",
      "6 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq08_maf0001_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf0001_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf0001_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_hrc_rsq08_maf0001_extracted.raw written.\n",
      "End time: Fri Jul  5 13:50:58 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_v3_rsq08_maf0001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/topmed_chr1_rsq08_maf0001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_topmed_v3_rsq08_maf0001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_v3_rsq08_maf0001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:50:58 2024\n",
      "515666 MiB RAM detected, ~490369 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed_v3/topmed_chr1_rsq08_maf0001_LOF_missense_extracted.fam.\n",
      "42131 variants loaded from\n",
      "../topmed_v3/topmed_chr1_rsq08_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 33 variants remaining.\n",
      "33 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_v3_rsq08_maf0001_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_v3_rsq08_maf0001_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_v3_rsq08_maf0001_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_topmed_v3_rsq08_maf0001_extracted.raw written.\n",
      "End time: Fri Jul  5 13:50:59 2024\n"
     ]
    }
   ],
   "source": [
    "cd ~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/pcsk9\n",
    "\n",
    "module load Plink/2.00a\n",
    "\n",
    "for i in 1 05 01; do\n",
    "    plink2 \\\n",
    "        --bfile ../exome/ukb23156_c1_maf00${i}_LOF_missense_extracted \\\n",
    "        --extract PCSK9_exome_maf00${i}_extractlist_snplist \\\n",
    "        --make-bed --export A \\\n",
    "        --out PCSK9_exome_maf00${i}_extracted\n",
    "\n",
    "    for j in 3 8; do\n",
    "        plink2 \\\n",
    "            --bpfile ../hrc/hrc_chr1_rsq0${j}_maf00${i}_LOF_missense_extracted \\\n",
    "            --extract PCSK9_hrc_rsq0${j}_maf00${i}_extractlist_snplist \\\n",
    "            --make-bpgen --sort-vars \\\n",
    "            --export A \\\n",
    "            --out PCSK9_hrc_rsq0${j}_maf00${i}_extracted\n",
    "\n",
    "        plink2 \\\n",
    "            --bpfile ../topmed_v3/topmed_chr1_rsq0${j}_maf00${i}_LOF_missense_extracted \\\n",
    "            --extract PCSK9_topmed_v3_rsq0${j}_maf00${i}_extractlist_snplist \\\n",
    "            --make-bpgen --sort-vars \\\n",
    "            --export A \\\n",
    "            --out PCSK9_topmed_v3_rsq0${j}_maf00${i}_extracted\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "706717d5-8ee4-4657-9bac-77bc3bfc791a",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_exome_maf001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c1_maf001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_exome_maf001_cadd_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out PCSK9_exome_maf001_cadd_extracted\n",
      "\n",
      "Start time: Fri Jul  5 13:52:41 2024\n",
      "515666 MiB RAM detected, ~490275 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c1_maf001_LOF_missense_extracted.fam.\n",
      "478357 variants loaded from\n",
      "../exome/ukb23156_c1_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 200 variants remaining.\n",
      "200 variants remaining after main filters.\n",
      "Writing PCSK9_exome_maf001_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_exome_maf001_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_exome_maf001_cadd_extracted.bed ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_exome_maf001_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:52:41 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq03_maf001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq03_maf001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_hrc_rsq03_maf001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq03_maf001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:52:41 2024\n",
      "515666 MiB RAM detected, ~490271 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq03_maf001_LOF_missense_extracted.fam.\n",
      "20551 variants loaded from\n",
      "../hrc/hrc_chr1_rsq03_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 6 variants remaining.\n",
      "6 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq03_maf001_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf001_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf001_cadd_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_hrc_rsq03_maf001_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:52:41 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_v3_rsq03_maf001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/../topmed_v3/topmed_chr1_rsq03_maf001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_topmed_v3_rsq03_maf001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_v3_rsq03_maf001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:52:41 2024\n",
      "515666 MiB RAM detected, ~490271 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from\n",
      "../topmed_v3/../topmed_v3/topmed_chr1_rsq03_maf001_LOF_missense_extracted.fam.\n",
      "141967 variants loaded from\n",
      "../topmed_v3/../topmed_v3/topmed_chr1_rsq03_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 61 variants remaining.\n",
      "61 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_v3_rsq03_maf001_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_v3_rsq03_maf001_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_v3_rsq03_maf001_cadd_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_topmed_v3_rsq03_maf001_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:52:41 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq08_maf001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq08_maf001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_hrc_rsq08_maf001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq08_maf001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:52:41 2024\n",
      "515666 MiB RAM detected, ~490270 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq08_maf001_LOF_missense_extracted.fam.\n",
      "8396 variants loaded from\n",
      "../hrc/hrc_chr1_rsq08_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 2 variants remaining.\n",
      "2 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq08_maf001_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf001_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf001_cadd_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_hrc_rsq08_maf001_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:52:42 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_v3_rsq08_maf001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/../topmed_v3/topmed_chr1_rsq08_maf001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_topmed_v3_rsq08_maf001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_v3_rsq08_maf001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:52:42 2024\n",
      "515666 MiB RAM detected, ~490268 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from\n",
      "../topmed_v3/../topmed_v3/topmed_chr1_rsq08_maf001_LOF_missense_extracted.fam.\n",
      "45669 variants loaded from\n",
      "../topmed_v3/../topmed_v3/topmed_chr1_rsq08_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 15 variants remaining.\n",
      "15 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_v3_rsq08_maf001_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_v3_rsq08_maf001_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_v3_rsq08_maf001_cadd_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_topmed_v3_rsq08_maf001_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:52:42 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_exome_maf0005_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c1_maf0005_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_exome_maf0005_cadd_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out PCSK9_exome_maf0005_cadd_extracted\n",
      "\n",
      "Start time: Fri Jul  5 13:52:42 2024\n",
      "515666 MiB RAM detected, ~490270 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c1_maf0005_LOF_missense_extracted.fam.\n",
      "477452 variants loaded from\n",
      "../exome/ukb23156_c1_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 200 variants remaining.\n",
      "200 variants remaining after main filters.\n",
      "Writing PCSK9_exome_maf0005_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_exome_maf0005_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_exome_maf0005_cadd_extracted.bed ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_exome_maf0005_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:52:42 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq03_maf0005_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq03_maf0005_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_hrc_rsq03_maf0005_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq03_maf0005_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:52:42 2024\n",
      "515666 MiB RAM detected, ~490267 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq03_maf0005_LOF_missense_extracted.fam.\n",
      "19743 variants loaded from\n",
      "../hrc/hrc_chr1_rsq03_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 6 variants remaining.\n",
      "6 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq03_maf0005_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf0005_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf0005_cadd_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_hrc_rsq03_maf0005_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:52:42 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_v3_rsq03_maf0005_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/../topmed_v3/topmed_chr1_rsq03_maf0005_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_topmed_v3_rsq03_maf0005_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_v3_rsq03_maf0005_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:52:42 2024\n",
      "515666 MiB RAM detected, ~490269 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from\n",
      "../topmed_v3/../topmed_v3/topmed_chr1_rsq03_maf0005_LOF_missense_extracted.fam.\n",
      "141141 variants loaded from\n",
      "../topmed_v3/../topmed_v3/topmed_chr1_rsq03_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 61 variants remaining.\n",
      "61 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_v3_rsq03_maf0005_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_v3_rsq03_maf0005_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_v3_rsq03_maf0005_cadd_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_topmed_v3_rsq03_maf0005_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:52:42 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq08_maf0005_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq08_maf0005_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_hrc_rsq08_maf0005_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq08_maf0005_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:52:42 2024\n",
      "515666 MiB RAM detected, ~490266 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq08_maf0005_LOF_missense_extracted.fam.\n",
      "7666 variants loaded from\n",
      "../hrc/hrc_chr1_rsq08_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 2 variants remaining.\n",
      "2 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq08_maf0005_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf0005_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf0005_cadd_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_hrc_rsq08_maf0005_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:52:42 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_v3_rsq08_maf0005_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/../topmed_v3/topmed_chr1_rsq08_maf0005_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_topmed_v3_rsq08_maf0005_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_v3_rsq08_maf0005_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:52:42 2024\n",
      "515666 MiB RAM detected, ~490265 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from\n",
      "../topmed_v3/../topmed_v3/topmed_chr1_rsq08_maf0005_LOF_missense_extracted.fam.\n",
      "44886 variants loaded from\n",
      "../topmed_v3/../topmed_v3/topmed_chr1_rsq08_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 15 variants remaining.\n",
      "15 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_v3_rsq08_maf0005_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_v3_rsq08_maf0005_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_v3_rsq08_maf0005_cadd_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_topmed_v3_rsq08_maf0005_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:52:42 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_exome_maf0001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c1_maf0001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_exome_maf0001_cadd_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out PCSK9_exome_maf0001_cadd_extracted\n",
      "\n",
      "Start time: Fri Jul  5 13:52:42 2024\n",
      "515666 MiB RAM detected, ~490265 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c1_maf0001_LOF_missense_extracted.fam.\n",
      "473986 variants loaded from\n",
      "../exome/ukb23156_c1_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 199 variants remaining.\n",
      "199 variants remaining after main filters.\n",
      "Writing PCSK9_exome_maf0001_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_exome_maf0001_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_exome_maf0001_cadd_extracted.bed ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_exome_maf0001_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:52:43 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq03_maf0001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq03_maf0001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_hrc_rsq03_maf0001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq03_maf0001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:52:43 2024\n",
      "515666 MiB RAM detected, ~490265 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq03_maf0001_LOF_missense_extracted.fam.\n",
      "16591 variants loaded from\n",
      "../hrc/hrc_chr1_rsq03_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 5 variants remaining.\n",
      "5 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq03_maf0001_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf0001_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq03_maf0001_cadd_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_hrc_rsq03_maf0001_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:52:43 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_v3_rsq03_maf0001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/../topmed_v3/topmed_chr1_rsq03_maf0001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_topmed_v3_rsq03_maf0001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_v3_rsq03_maf0001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:52:43 2024\n",
      "515666 MiB RAM detected, ~490264 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from\n",
      "../topmed_v3/../topmed_v3/topmed_chr1_rsq03_maf0001_LOF_missense_extracted.fam.\n",
      "137927 variants loaded from\n",
      "../topmed_v3/../topmed_v3/topmed_chr1_rsq03_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 60 variants remaining.\n",
      "60 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_v3_rsq03_maf0001_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_v3_rsq03_maf0001_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_v3_rsq03_maf0001_cadd_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_topmed_v3_rsq03_maf0001_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:52:43 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_hrc_rsq08_maf0001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr1_rsq08_maf0001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_hrc_rsq08_maf0001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_hrc_rsq08_maf0001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:52:43 2024\n",
      "515666 MiB RAM detected, ~490263 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr1_rsq08_maf0001_LOF_missense_extracted.fam.\n",
      "5243 variants loaded from\n",
      "../hrc/hrc_chr1_rsq08_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 2 variants remaining.\n",
      "2 variants remaining after main filters.\n",
      "Writing PCSK9_hrc_rsq08_maf0001_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf0001_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_hrc_rsq08_maf0001_cadd_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_hrc_rsq08_maf0001_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:52:43 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PCSK9_topmed_v3_rsq08_maf0001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/../topmed_v3/topmed_chr1_rsq08_maf0001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract PCSK9_topmed_v3_rsq08_maf0001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out PCSK9_topmed_v3_rsq08_maf0001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:52:43 2024\n",
      "515666 MiB RAM detected, ~490262 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from\n",
      "../topmed_v3/../topmed_v3/topmed_chr1_rsq08_maf0001_LOF_missense_extracted.fam.\n",
      "42131 variants loaded from\n",
      "../topmed_v3/../topmed_v3/topmed_chr1_rsq08_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 14 variants remaining.\n",
      "14 variants remaining after main filters.\n",
      "Writing PCSK9_topmed_v3_rsq08_maf0001_cadd_extracted.bim ... done.\n",
      "Writing PCSK9_topmed_v3_rsq08_maf0001_cadd_extracted.fam ... done.\n",
      "Writing PCSK9_topmed_v3_rsq08_maf0001_cadd_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: PCSK9_topmed_v3_rsq08_maf0001_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:52:43 2024\n"
     ]
    }
   ],
   "source": [
    "for i in 1 05 01; do\n",
    "    plink2 \\\n",
    "        --bfile ../exome/ukb23156_c1_maf00${i}_LOF_missense_extracted \\\n",
    "        --extract PCSK9_exome_maf00${i}_cadd_extractlist_snplist \\\n",
    "        --make-bed --export A \\\n",
    "        --out PCSK9_exome_maf00${i}_cadd_extracted\n",
    "\n",
    "    for j in 3 8; do\n",
    "        plink2 \\\n",
    "            --bpfile ../hrc/hrc_chr1_rsq0${j}_maf00${i}_LOF_missense_extracted \\\n",
    "            --extract PCSK9_hrc_rsq0${j}_maf00${i}_cadd_extractlist_snplist \\\n",
    "            --make-bpgen --sort-vars \\\n",
    "            --export A \\\n",
    "            --out PCSK9_hrc_rsq0${j}_maf00${i}_cadd_extracted\n",
    "\n",
    "        plink2 \\\n",
    "            --bpfile ../topmed_v3/../topmed_v3/topmed_chr1_rsq0${j}_maf00${i}_LOF_missense_extracted \\\n",
    "            --extract PCSK9_topmed_v3_rsq0${j}_maf00${i}_cadd_extractlist_snplist \\\n",
    "            --make-bpgen --sort-vars \\\n",
    "            --export A \\\n",
    "            --out PCSK9_topmed_v3_rsq0${j}_maf00${i}_cadd_extracted\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "1cc93470-3364-43c2-8915-0152ba22e9c9",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_exome_maf001_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c11_maf001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_exome_maf001_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out APOC3_exome_maf001_extracted\n",
      "\n",
      "Start time: Fri Jul  5 13:52:52 2024\n",
      "515666 MiB RAM detected, ~490212 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c11_maf001_LOF_missense_extracted.fam.\n",
      "293784 variants loaded from\n",
      "../exome/ukb23156_c11_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 60 variants remaining.\n",
      "60 variants remaining after main filters.\n",
      "Writing APOC3_exome_maf001_extracted.fam ... done.\n",
      "Writing APOC3_exome_maf001_extracted.bim ... done.\n",
      "Writing APOC3_exome_maf001_extracted.bed ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_exome_maf001_extracted.raw written.\n",
      "End time: Fri Jul  5 13:52:57 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq03_maf001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq03_maf001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_hrc_rsq03_maf001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq03_maf001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:52:57 2024\n",
      "515666 MiB RAM detected, ~490185 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq03_maf001_LOF_missense_extracted.fam.\n",
      "13392 variants loaded from\n",
      "../hrc/hrc_chr11_rsq03_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 3 variants remaining.\n",
      "3 variants remaining after main filters.\n",
      "Writing APOC3_hrc_rsq03_maf001_extracted.bim ... done.\n",
      "Writing APOC3_hrc_rsq03_maf001_extracted.fam ... done.\n",
      "Writing APOC3_hrc_rsq03_maf001_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_hrc_rsq03_maf001_extracted.raw written.\n",
      "End time: Fri Jul  5 13:52:58 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_v3_rsq03_maf001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/topmed_chr11_rsq03_maf001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_topmed_v3_rsq03_maf001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_v3_rsq03_maf001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:52:58 2024\n",
      "515666 MiB RAM detected, ~490179 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed_v3/topmed_chr11_rsq03_maf001_LOF_missense_extracted.fam.\n",
      "88634 variants loaded from\n",
      "../topmed_v3/topmed_chr11_rsq03_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 18 variants remaining.\n",
      "18 variants remaining after main filters.\n",
      "Writing APOC3_topmed_v3_rsq03_maf001_extracted.bim ... done.\n",
      "Writing APOC3_topmed_v3_rsq03_maf001_extracted.fam ... done.\n",
      "Writing APOC3_topmed_v3_rsq03_maf001_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_topmed_v3_rsq03_maf001_extracted.raw written.\n",
      "End time: Fri Jul  5 13:53:00 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq08_maf001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq08_maf001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_hrc_rsq08_maf001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq08_maf001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:00 2024\n",
      "515666 MiB RAM detected, ~490168 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq08_maf001_LOF_missense_extracted.fam.\n",
      "5811 variants loaded from\n",
      "../hrc/hrc_chr11_rsq08_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 2 variants remaining.\n",
      "2 variants remaining after main filters.\n",
      "Writing APOC3_hrc_rsq08_maf001_extracted.bim ... done.\n",
      "Writing APOC3_hrc_rsq08_maf001_extracted.fam ... done.\n",
      "Writing APOC3_hrc_rsq08_maf001_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_hrc_rsq08_maf001_extracted.raw written.\n",
      "End time: Fri Jul  5 13:53:01 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_v3_rsq08_maf001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/topmed_chr11_rsq08_maf001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_topmed_v3_rsq08_maf001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_v3_rsq08_maf001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:01 2024\n",
      "515666 MiB RAM detected, ~490161 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed_v3/topmed_chr11_rsq08_maf001_LOF_missense_extracted.fam.\n",
      "29328 variants loaded from\n",
      "../topmed_v3/topmed_chr11_rsq08_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 4 variants remaining.\n",
      "4 variants remaining after main filters.\n",
      "Writing APOC3_topmed_v3_rsq08_maf001_extracted.bim ... done.\n",
      "Writing APOC3_topmed_v3_rsq08_maf001_extracted.fam ... done.\n",
      "Writing APOC3_topmed_v3_rsq08_maf001_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_topmed_v3_rsq08_maf001_extracted.raw written.\n",
      "End time: Fri Jul  5 13:53:03 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_exome_maf0005_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c11_maf0005_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_exome_maf0005_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out APOC3_exome_maf0005_extracted\n",
      "\n",
      "Start time: Fri Jul  5 13:53:03 2024\n",
      "515666 MiB RAM detected, ~490148 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c11_maf0005_LOF_missense_extracted.fam.\n",
      "293221 variants loaded from\n",
      "../exome/ukb23156_c11_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 60 variants remaining.\n",
      "60 variants remaining after main filters.\n",
      "Writing APOC3_exome_maf0005_extracted.fam ... done.\n",
      "Writing APOC3_exome_maf0005_extracted.bim ... done.\n",
      "Writing APOC3_exome_maf0005_extracted.bed ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_exome_maf0005_extracted.raw written.\n",
      "End time: Fri Jul  5 13:53:07 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq03_maf0005_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq03_maf0005_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_hrc_rsq03_maf0005_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq03_maf0005_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:07 2024\n",
      "515666 MiB RAM detected, ~490124 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq03_maf0005_LOF_missense_extracted.fam.\n",
      "12867 variants loaded from\n",
      "../hrc/hrc_chr11_rsq03_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 3 variants remaining.\n",
      "3 variants remaining after main filters.\n",
      "Writing APOC3_hrc_rsq03_maf0005_extracted.bim ... done.\n",
      "Writing APOC3_hrc_rsq03_maf0005_extracted.fam ... done.\n",
      "Writing APOC3_hrc_rsq03_maf0005_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_hrc_rsq03_maf0005_extracted.raw written.\n",
      "End time: Fri Jul  5 13:53:11 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_v3_rsq03_maf0005_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/topmed_chr11_rsq03_maf0005_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_topmed_v3_rsq03_maf0005_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_v3_rsq03_maf0005_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:11 2024\n",
      "515666 MiB RAM detected, ~490105 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed_v3/topmed_chr11_rsq03_maf0005_LOF_missense_extracted.fam.\n",
      "88110 variants loaded from\n",
      "../topmed_v3/topmed_chr11_rsq03_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 18 variants remaining.\n",
      "18 variants remaining after main filters.\n",
      "Writing APOC3_topmed_v3_rsq03_maf0005_extracted.bim ... done.\n",
      "Writing APOC3_topmed_v3_rsq03_maf0005_extracted.fam ... done.\n",
      "Writing APOC3_topmed_v3_rsq03_maf0005_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_topmed_v3_rsq03_maf0005_extracted.raw written.\n",
      "End time: Fri Jul  5 13:53:13 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq08_maf0005_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq08_maf0005_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_hrc_rsq08_maf0005_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq08_maf0005_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:13 2024\n",
      "515666 MiB RAM detected, ~490374 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq08_maf0005_LOF_missense_extracted.fam.\n",
      "5315 variants loaded from\n",
      "../hrc/hrc_chr11_rsq08_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 2 variants remaining.\n",
      "2 variants remaining after main filters.\n",
      "Writing APOC3_hrc_rsq08_maf0005_extracted.bim ... done.\n",
      "Writing APOC3_hrc_rsq08_maf0005_extracted.fam ... done.\n",
      "Writing APOC3_hrc_rsq08_maf0005_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_hrc_rsq08_maf0005_extracted.raw written.\n",
      "End time: Fri Jul  5 13:53:15 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_v3_rsq08_maf0005_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/topmed_chr11_rsq08_maf0005_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_topmed_v3_rsq08_maf0005_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_v3_rsq08_maf0005_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:15 2024\n",
      "515666 MiB RAM detected, ~490366 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed_v3/topmed_chr11_rsq08_maf0005_LOF_missense_extracted.fam.\n",
      "28822 variants loaded from\n",
      "../topmed_v3/topmed_chr11_rsq08_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 4 variants remaining.\n",
      "4 variants remaining after main filters.\n",
      "Writing APOC3_topmed_v3_rsq08_maf0005_extracted.bim ... done.\n",
      "Writing APOC3_topmed_v3_rsq08_maf0005_extracted.fam ... done.\n",
      "Writing APOC3_topmed_v3_rsq08_maf0005_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_topmed_v3_rsq08_maf0005_extracted.raw written.\n",
      "End time: Fri Jul  5 13:53:17 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_exome_maf0001_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c11_maf0001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_exome_maf0001_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out APOC3_exome_maf0001_extracted\n",
      "\n",
      "Start time: Fri Jul  5 13:53:17 2024\n",
      "515666 MiB RAM detected, ~490353 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c11_maf0001_LOF_missense_extracted.fam.\n",
      "291093 variants loaded from\n",
      "../exome/ukb23156_c11_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 59 variants remaining.\n",
      "59 variants remaining after main filters.\n",
      "Writing APOC3_exome_maf0001_extracted.fam ... done.\n",
      "Writing APOC3_exome_maf0001_extracted.bim ... done.\n",
      "Writing APOC3_exome_maf0001_extracted.bed ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_exome_maf0001_extracted.raw written.\n",
      "End time: Fri Jul  5 13:53:21 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq03_maf0001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq03_maf0001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_hrc_rsq03_maf0001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq03_maf0001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:21 2024\n",
      "515666 MiB RAM detected, ~490332 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq03_maf0001_LOF_missense_extracted.fam.\n",
      "10872 variants loaded from\n",
      "../hrc/hrc_chr11_rsq03_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 2 variants remaining.\n",
      "2 variants remaining after main filters.\n",
      "Writing APOC3_hrc_rsq03_maf0001_extracted.bim ... done.\n",
      "Writing APOC3_hrc_rsq03_maf0001_extracted.fam ... done.\n",
      "Writing APOC3_hrc_rsq03_maf0001_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_hrc_rsq03_maf0001_extracted.raw written.\n",
      "End time: Fri Jul  5 13:53:23 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_v3_rsq03_maf0001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/topmed_chr11_rsq03_maf0001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_topmed_v3_rsq03_maf0001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_v3_rsq03_maf0001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:23 2024\n",
      "515666 MiB RAM detected, ~490320 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed_v3/topmed_chr11_rsq03_maf0001_LOF_missense_extracted.fam.\n",
      "86113 variants loaded from\n",
      "../topmed_v3/topmed_chr11_rsq03_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 17 variants remaining.\n",
      "17 variants remaining after main filters.\n",
      "Writing APOC3_topmed_v3_rsq03_maf0001_extracted.bim ... done.\n",
      "Writing APOC3_topmed_v3_rsq03_maf0001_extracted.fam ... done.\n",
      "Writing APOC3_topmed_v3_rsq03_maf0001_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_topmed_v3_rsq03_maf0001_extracted.raw written.\n",
      "End time: Fri Jul  5 13:53:25 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq08_maf0001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq08_maf0001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_hrc_rsq08_maf0001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq08_maf0001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:25 2024\n",
      "515666 MiB RAM detected, ~490309 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq08_maf0001_LOF_missense_extracted.fam.\n",
      "3788 variants loaded from\n",
      "../hrc/hrc_chr11_rsq08_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 0 variants remaining.\n",
      "Error: No variants remaining after main filters.\n",
      "End time: Fri Jul  5 13:53:26 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_v3_rsq08_maf0001_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/topmed_chr11_rsq08_maf0001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_topmed_v3_rsq08_maf0001_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_v3_rsq08_maf0001_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:26 2024\n",
      "515666 MiB RAM detected, ~490302 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed_v3/topmed_chr11_rsq08_maf0001_LOF_missense_extracted.fam.\n",
      "27096 variants loaded from\n",
      "../topmed_v3/topmed_chr11_rsq08_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 3 variants remaining.\n",
      "3 variants remaining after main filters.\n",
      "Writing APOC3_topmed_v3_rsq08_maf0001_extracted.bim ... done.\n",
      "Writing APOC3_topmed_v3_rsq08_maf0001_extracted.fam ... done.\n",
      "Writing APOC3_topmed_v3_rsq08_maf0001_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_topmed_v3_rsq08_maf0001_extracted.raw written.\n",
      "End time: Fri Jul  5 13:53:28 2024\n"
     ]
    }
   ],
   "source": [
    "cd ~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/apoc3\n",
    "\n",
    "for i in 1 05 01; do\n",
    "    plink2 \\\n",
    "        --bfile ../exome/ukb23156_c11_maf00${i}_LOF_missense_extracted \\\n",
    "        --extract APOC3_exome_maf00${i}_extractlist_snplist \\\n",
    "        --make-bed --export A \\\n",
    "        --out APOC3_exome_maf00${i}_extracted\n",
    "\n",
    "    for j in 3 8; do\n",
    "        plink2 \\\n",
    "            --bpfile ../hrc/hrc_chr11_rsq0${j}_maf00${i}_LOF_missense_extracted \\\n",
    "            --extract APOC3_hrc_rsq0${j}_maf00${i}_extractlist_snplist \\\n",
    "            --make-bpgen --sort-vars \\\n",
    "            --export A \\\n",
    "            --out APOC3_hrc_rsq0${j}_maf00${i}_extracted\n",
    "\n",
    "        plink2 \\\n",
    "            --bpfile ../topmed_v3/topmed_chr11_rsq0${j}_maf00${i}_LOF_missense_extracted \\\n",
    "            --extract APOC3_topmed_v3_rsq0${j}_maf00${i}_extractlist_snplist \\\n",
    "            --make-bpgen --sort-vars \\\n",
    "            --export A \\\n",
    "            --out APOC3_topmed_v3_rsq0${j}_maf00${i}_extracted\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "db049040-a07b-457a-be12-0053daa19015",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_exome_maf001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c11_maf001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_exome_maf001_cadd_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out APOC3_exome_maf001_cadd_extracted\n",
      "\n",
      "Start time: Fri Jul  5 13:53:33 2024\n",
      "515666 MiB RAM detected, ~490265 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c11_maf001_LOF_missense_extracted.fam.\n",
      "293784 variants loaded from\n",
      "../exome/ukb23156_c11_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 26 variants remaining.\n",
      "26 variants remaining after main filters.\n",
      "Writing APOC3_exome_maf001_cadd_extracted.fam ... done.\n",
      "Writing APOC3_exome_maf001_cadd_extracted.bim ... done.\n",
      "Writing APOC3_exome_maf001_cadd_extracted.bed ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_exome_maf001_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:53:33 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq03_maf001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq03_maf001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_hrc_rsq03_maf001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq03_maf001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:33 2024\n",
      "515666 MiB RAM detected, ~490265 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq03_maf001_LOF_missense_extracted.fam.\n",
      "13392 variants loaded from\n",
      "../hrc/hrc_chr11_rsq03_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 3 variants remaining.\n",
      "3 variants remaining after main filters.\n",
      "Writing APOC3_hrc_rsq03_maf001_cadd_extracted.bim ... done.\n",
      "Writing APOC3_hrc_rsq03_maf001_cadd_extracted.fam ... done.\n",
      "Writing APOC3_hrc_rsq03_maf001_cadd_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_hrc_rsq03_maf001_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:53:33 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_v3_rsq03_maf001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/topmed_chr1_rsq03_maf001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_topmed_v3_rsq03_maf001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_v3_rsq03_maf001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:33 2024\n",
      "515666 MiB RAM detected, ~490262 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed_v3/topmed_chr1_rsq03_maf001_LOF_missense_extracted.fam.\n",
      "141967 variants loaded from\n",
      "../topmed_v3/topmed_chr1_rsq03_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 0 variants remaining.\n",
      "Error: No variants remaining after main filters.\n",
      "End time: Fri Jul  5 13:53:33 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq08_maf001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq08_maf001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_hrc_rsq08_maf001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq08_maf001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:33 2024\n",
      "515666 MiB RAM detected, ~490260 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq08_maf001_LOF_missense_extracted.fam.\n",
      "5811 variants loaded from\n",
      "../hrc/hrc_chr11_rsq08_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 2 variants remaining.\n",
      "2 variants remaining after main filters.\n",
      "Writing APOC3_hrc_rsq08_maf001_cadd_extracted.bim ... done.\n",
      "Writing APOC3_hrc_rsq08_maf001_cadd_extracted.fam ... done.\n",
      "Writing APOC3_hrc_rsq08_maf001_cadd_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_hrc_rsq08_maf001_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:53:33 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_v3_rsq08_maf001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/topmed_chr1_rsq08_maf001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_topmed_v3_rsq08_maf001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_v3_rsq08_maf001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:33 2024\n",
      "515666 MiB RAM detected, ~490260 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed_v3/topmed_chr1_rsq08_maf001_LOF_missense_extracted.fam.\n",
      "45669 variants loaded from\n",
      "../topmed_v3/topmed_chr1_rsq08_maf001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 0 variants remaining.\n",
      "Error: No variants remaining after main filters.\n",
      "End time: Fri Jul  5 13:53:33 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_exome_maf0005_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c11_maf0005_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_exome_maf0005_cadd_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out APOC3_exome_maf0005_cadd_extracted\n",
      "\n",
      "Start time: Fri Jul  5 13:53:33 2024\n",
      "515666 MiB RAM detected, ~490260 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c11_maf0005_LOF_missense_extracted.fam.\n",
      "293221 variants loaded from\n",
      "../exome/ukb23156_c11_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 26 variants remaining.\n",
      "26 variants remaining after main filters.\n",
      "Writing APOC3_exome_maf0005_cadd_extracted.fam ... done.\n",
      "Writing APOC3_exome_maf0005_cadd_extracted.bim ... done.\n",
      "Writing APOC3_exome_maf0005_cadd_extracted.bed ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_exome_maf0005_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:53:33 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq03_maf0005_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq03_maf0005_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_hrc_rsq03_maf0005_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq03_maf0005_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:33 2024\n",
      "515666 MiB RAM detected, ~490260 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq03_maf0005_LOF_missense_extracted.fam.\n",
      "12867 variants loaded from\n",
      "../hrc/hrc_chr11_rsq03_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 3 variants remaining.\n",
      "3 variants remaining after main filters.\n",
      "Writing APOC3_hrc_rsq03_maf0005_cadd_extracted.bim ... done.\n",
      "Writing APOC3_hrc_rsq03_maf0005_cadd_extracted.fam ... done.\n",
      "Writing APOC3_hrc_rsq03_maf0005_cadd_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_hrc_rsq03_maf0005_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:53:33 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_v3_rsq03_maf0005_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/topmed_chr1_rsq03_maf0005_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_topmed_v3_rsq03_maf0005_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_v3_rsq03_maf0005_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:33 2024\n",
      "515666 MiB RAM detected, ~490256 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed_v3/topmed_chr1_rsq03_maf0005_LOF_missense_extracted.fam.\n",
      "141141 variants loaded from\n",
      "../topmed_v3/topmed_chr1_rsq03_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 0 variants remaining.\n",
      "Error: No variants remaining after main filters.\n",
      "End time: Fri Jul  5 13:53:34 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq08_maf0005_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq08_maf0005_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_hrc_rsq08_maf0005_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq08_maf0005_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:34 2024\n",
      "515666 MiB RAM detected, ~490257 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq08_maf0005_LOF_missense_extracted.fam.\n",
      "5315 variants loaded from\n",
      "../hrc/hrc_chr11_rsq08_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 2 variants remaining.\n",
      "2 variants remaining after main filters.\n",
      "Writing APOC3_hrc_rsq08_maf0005_cadd_extracted.bim ... done.\n",
      "Writing APOC3_hrc_rsq08_maf0005_cadd_extracted.fam ... done.\n",
      "Writing APOC3_hrc_rsq08_maf0005_cadd_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_hrc_rsq08_maf0005_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:53:34 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_v3_rsq08_maf0005_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/topmed_chr1_rsq08_maf0005_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_topmed_v3_rsq08_maf0005_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_v3_rsq08_maf0005_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:34 2024\n",
      "515666 MiB RAM detected, ~490259 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed_v3/topmed_chr1_rsq08_maf0005_LOF_missense_extracted.fam.\n",
      "44886 variants loaded from\n",
      "../topmed_v3/topmed_chr1_rsq08_maf0005_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 0 variants remaining.\n",
      "Error: No variants remaining after main filters.\n",
      "End time: Fri Jul  5 13:53:34 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_exome_maf0001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bfile ../exome/ukb23156_c11_maf0001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_exome_maf0001_cadd_extractlist_snplist\n",
      "  --make-bed\n",
      "  --out APOC3_exome_maf0001_cadd_extracted\n",
      "\n",
      "Start time: Fri Jul  5 13:53:34 2024\n",
      "515666 MiB RAM detected, ~490258 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../exome/ukb23156_c11_maf0001_LOF_missense_extracted.fam.\n",
      "291093 variants loaded from\n",
      "../exome/ukb23156_c11_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 25 variants remaining.\n",
      "25 variants remaining after main filters.\n",
      "Writing APOC3_exome_maf0001_cadd_extracted.fam ... done.\n",
      "Writing APOC3_exome_maf0001_cadd_extracted.bim ... done.\n",
      "Writing APOC3_exome_maf0001_cadd_extracted.bed ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_exome_maf0001_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:53:34 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq03_maf0001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq03_maf0001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_hrc_rsq03_maf0001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq03_maf0001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:34 2024\n",
      "515666 MiB RAM detected, ~490258 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq03_maf0001_LOF_missense_extracted.fam.\n",
      "10872 variants loaded from\n",
      "../hrc/hrc_chr11_rsq03_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 2 variants remaining.\n",
      "2 variants remaining after main filters.\n",
      "Writing APOC3_hrc_rsq03_maf0001_cadd_extracted.bim ... done.\n",
      "Writing APOC3_hrc_rsq03_maf0001_cadd_extracted.fam ... done.\n",
      "Writing APOC3_hrc_rsq03_maf0001_cadd_extracted.pgen ... done.\n",
      "--export A pass 1/1: loading... 0writing... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "--export A: APOC3_hrc_rsq03_maf0001_cadd_extracted.raw written.\n",
      "End time: Fri Jul  5 13:53:34 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_v3_rsq03_maf0001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/topmed_chr1_rsq03_maf0001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_topmed_v3_rsq03_maf0001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_v3_rsq03_maf0001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:34 2024\n",
      "515666 MiB RAM detected, ~490254 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed_v3/topmed_chr1_rsq03_maf0001_LOF_missense_extracted.fam.\n",
      "137927 variants loaded from\n",
      "../topmed_v3/topmed_chr1_rsq03_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 0 variants remaining.\n",
      "Error: No variants remaining after main filters.\n",
      "End time: Fri Jul  5 13:53:34 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_hrc_rsq08_maf0001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../hrc/hrc_chr11_rsq08_maf0001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_hrc_rsq08_maf0001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_hrc_rsq08_maf0001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:34 2024\n",
      "515666 MiB RAM detected, ~490253 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../hrc/hrc_chr11_rsq08_maf0001_LOF_missense_extracted.fam.\n",
      "3788 variants loaded from\n",
      "../hrc/hrc_chr11_rsq08_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 0 variants remaining.\n",
      "Error: No variants remaining after main filters.\n",
      "End time: Fri Jul  5 13:53:34 2024\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to APOC3_topmed_v3_rsq08_maf0001_cadd_extracted.log.\n",
      "Options in effect:\n",
      "  --bpfile ../topmed_v3/topmed_chr1_rsq08_maf0001_LOF_missense_extracted\n",
      "  --export A\n",
      "  --extract APOC3_topmed_v3_rsq08_maf0001_cadd_extractlist_snplist\n",
      "  --make-bpgen\n",
      "  --out APOC3_topmed_v3_rsq08_maf0001_cadd_extracted\n",
      "  --sort-vars\n",
      "\n",
      "Start time: Fri Jul  5 13:53:34 2024\n",
      "515666 MiB RAM detected, ~490250 available; reserving 257833 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "168206 samples (0 females, 0 males, 168206 ambiguous; 168206 founders) loaded\n",
      "from ../topmed_v3/topmed_chr1_rsq08_maf0001_LOF_missense_extracted.fam.\n",
      "42131 variants loaded from\n",
      "../topmed_v3/topmed_chr1_rsq08_maf0001_LOF_missense_extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 0 variants remaining.\n",
      "Error: No variants remaining after main filters.\n",
      "End time: Fri Jul  5 13:53:34 2024\n"
     ]
    },
    {
     "ename": "",
     "evalue": "13",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "for i in 1 05 01; do\n",
    "    plink2 \\\n",
    "        --bfile ../exome/ukb23156_c11_maf00${i}_LOF_missense_extracted \\\n",
    "        --extract APOC3_exome_maf00${i}_cadd_extractlist_snplist \\\n",
    "        --make-bed --export A \\\n",
    "        --out APOC3_exome_maf00${i}_cadd_extracted\n",
    "\n",
    "    for j in 3 8; do\n",
    "        plink2 \\\n",
    "            --bpfile ../hrc/hrc_chr11_rsq0${j}_maf00${i}_LOF_missense_extracted \\\n",
    "            --extract APOC3_hrc_rsq0${j}_maf00${i}_cadd_extractlist_snplist \\\n",
    "            --make-bpgen --sort-vars \\\n",
    "            --export A \\\n",
    "            --out APOC3_hrc_rsq0${j}_maf00${i}_cadd_extracted\n",
    "\n",
    "        plink2 \\\n",
    "            --bpfile ../topmed_v3/topmed_chr1_rsq0${j}_maf00${i}_LOF_missense_extracted \\\n",
    "            --extract APOC3_topmed_v3_rsq0${j}_maf00${i}_cadd_extractlist_snplist \\\n",
    "            --make-bpgen --sort-vars \\\n",
    "            --export A \\\n",
    "            --out APOC3_topmed_v3_rsq0${j}_maf00${i}_cadd_extracted\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cac917a4-e325-48cb-8bd8-1e1a732199cc",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "## Make merged dataset"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "774d0c62-eaec-4012-b3f4-0ba48f98d2a4",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### PCSK9 - exome + topmed + hrc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "e5dbef7c-37d1-4335-ab60-d551ca60bb72",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────── tidyverse 1.3.2 ──\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.5.1     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 1.0.2\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.2.1     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.5.1\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.3.1     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 1.0.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.1.5     \n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdata.table\u001b[39m::\u001b[32mbetween()\u001b[39m masks \u001b[34mdplyr\u001b[39m::between()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m       masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdata.table\u001b[39m::\u001b[32mfirst()\u001b[39m   masks \u001b[34mdplyr\u001b[39m::first()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m          masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdata.table\u001b[39m::\u001b[32mlast()\u001b[39m    masks \u001b[34mdplyr\u001b[39m::last()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mtranspose()\u001b[39m    masks \u001b[34mdata.table\u001b[39m::transpose()\n"
     ]
    }
   ],
   "source": [
    "setwd(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/pcsk9\")\n",
    "\n",
    "library(dplyr)\n",
    "library(data.table)\n",
    "library(tidyverse)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "168cf009-8648-4ec3-be10-b6fe2502229f",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"==========RSQ 0.3; MAF 001; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 2; TOPMed_v3_snplist: 16; Exome: 394; Annot_gene: 412; Full dataframe: 412\"\n",
      "[1] \"==========RSQ 0.3; MAF 001; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 1; TOPMed_v3_snplist: 3; Exome: 200; Annot_gene: 204; Full dataframe: 204\"\n",
      "[1] \"==========RSQ 0.3; MAF 0005; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 2; TOPMed_v3_snplist: 16; Exome: 394; Annot_gene: 412; Full dataframe: 412\"\n",
      "[1] \"==========RSQ 0.3; MAF 0005; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 1; TOPMed_v3_snplist: 3; Exome: 200; Annot_gene: 204; Full dataframe: 204\"\n",
      "[1] \"==========RSQ 0.3; MAF 0001; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 2; TOPMed_v3_snplist: 16; Exome: 393; Annot_gene: 411; Full dataframe: 411\"\n",
      "[1] \"==========RSQ 0.3; MAF 0001; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 1; TOPMed_v3_snplist: 3; Exome: 199; Annot_gene: 203; Full dataframe: 203\"\n",
      "[1] \"==========RSQ 0.8; MAF 001; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_v3_snplist: 0; Exome: 394; Annot_gene: 394; Full dataframe: 394\"\n",
      "[1] \"==========RSQ 0.8; MAF 001; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_v3_snplist: 0; Exome: 200; Annot_gene: 200; Full dataframe: 200\"\n",
      "[1] \"==========RSQ 0.8; MAF 0005; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_v3_snplist: 0; Exome: 394; Annot_gene: 394; Full dataframe: 394\"\n",
      "[1] \"==========RSQ 0.8; MAF 0005; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_v3_snplist: 0; Exome: 200; Annot_gene: 200; Full dataframe: 200\"\n",
      "[1] \"==========RSQ 0.8; MAF 0001; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_v3_snplist: 0; Exome: 393; Annot_gene: 393; Full dataframe: 393\"\n",
      "[1] \"==========RSQ 0.8; MAF 0001; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_v3_snplist: 0; Exome: 199; Annot_gene: 199; Full dataframe: 199\"\n"
     ]
    }
   ],
   "source": [
    "for(rsq in c(3, 8)){\n",
    "    for(maf in c(\"1\", \"05\", \"01\")){\n",
    "        for(cadd in c(\"\", \"_cadd\")){\n",
    "            print(sprintf(\"==========RSQ 0.%d; MAF 00%s; CADD filtering:%s\", rsq, maf, cadd))\n",
    "            \n",
    "            annot_fname <- sprintf(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/hrc_topmed_exome/hrc_topmed_v3_exome_168206ids_rsq0%d_maf00%s%s_annot.csv.gz\", \n",
    "                       rsq, maf, cadd)\n",
    "            annot_gene <- read.table(annot_fname, sep = \",\", header = TRUE) %>% filter(Gene.refGene == \"PCSK9\") %>% unique()\n",
    "            \n",
    "            full_df <- read.table(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/id_only.csv\", sep = \",\", header = TRUE)\n",
    "            \n",
    "            exome_fname <- sprintf(\"PCSK9_exome_maf00%s%s_extracted.raw\", maf, cadd)\n",
    "            hrc_fname <- sprintf(\"PCSK9_hrc_rsq0%d_maf00%s%s_extracted.raw\", rsq, maf, cadd)\n",
    "            topmed_fname <- sprintf(\"PCSK9_topmed_v3_rsq0%d_maf00%s%s_extracted.raw\", rsq, maf, cadd)\n",
    "\n",
    "            hrc_snplist <- annot_gene %>% filter(source == \"hrc\") %>% pull(ID_hg19)\n",
    "            topmed_snplist <- annot_gene %>% filter(source == \"topmed\") %>% pull(ID)\n",
    "            \n",
    "            ## exome matrix\n",
    "            if(file.exists(exome_fname)){\n",
    "                exome <- read.table(exome_fname, sep = \"\\t\", header = TRUE) %>% arrange(FID, IID)\n",
    "                exome_names <- colnames(exome)[7: dim(exome)[2]] %>% \n",
    "                    stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                exome_names <- stringr::str_replace_all(exome_names[,1], \"\\\\.\", \":\")\n",
    "                colnames(exome)[7:dim(exome)[2]] <- exome_names\n",
    "\n",
    "                full_df <- left_join(full_df, exome)\n",
    "            }\n",
    "            \n",
    "            ## hrc matrix\n",
    "            if(file.exists(hrc_fname)){\n",
    "                hrc <- read.table(hrc_fname, sep = \"\\t\", header = TRUE)\n",
    "                hrc_names <- colnames(hrc)[7: dim(hrc)[2]] %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                hrc_names <- stringr::str_replace_all(hrc_names[,1], \"\\\\.\", \":\")\n",
    "                colnames(hrc)[7:dim(hrc)[2]] <- hrc_names\n",
    "\n",
    "                fid_iid <- hrc$IID %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                hrc$FID <- as.integer(fid_iid[,1])\n",
    "                hrc$IID <- as.integer(fid_iid[,2])\n",
    "\n",
    "                hrc <- hrc %>% arrange(FID, IID) %>% select(c(\"FID\", \"IID\", \"PAT\", \"MAT\", \"SEX\", \"PHENOTYPE\", hrc_snplist))\n",
    "                full_df <- left_join(full_df, hrc)\n",
    "            }\n",
    "            \n",
    "            ## topmed matrix\n",
    "            if(file.exists(topmed_fname)){\n",
    "                topmed <- read.table(topmed_fname, sep = \"\\t\", header = TRUE)\n",
    "                topmed_names <- colnames(topmed)[7: dim(topmed)[2]] %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                topmed_names <- stringr::str_replace_all(topmed_names[,1], \"\\\\.\", \":\")\n",
    "                colnames(topmed)[7:dim(topmed)[2]] <- topmed_names\n",
    "\n",
    "                fid_iid <- topmed$IID %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                topmed$FID <- as.integer(fid_iid[,1])\n",
    "                topmed$IID <- as.integer(fid_iid[,2])\n",
    "\n",
    "                topmed <- topmed %>% arrange(FID, IID) %>% select(c(\"FID\", \"IID\", \"PAT\", \"MAT\", \"SEX\", \"PHENOTYPE\", topmed_snplist))\n",
    "                full_df <- left_join(full_df, topmed)\n",
    "            }\n",
    "            \n",
    "            fwrite(full_df, sprintf(\"PCSK9_hrc_topmed_v3_exome_rsq0%d_maf00%s%s_extracted.raw\", rsq, maf, cadd))\n",
    "            \n",
    "            print(sprintf(\"HRC_snplist: %d; TOPMed_v3_snplist: %d; Exome: %d; Annot_gene: %d; Full dataframe: %d\", \n",
    "                          length(hrc_snplist), length(topmed_snplist), ncol(exome)-6, nrow(annot_gene), ncol(full_df)-6))\n",
    "            rm(full_df)\n",
    "        }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a06bc6c3-7631-42dd-9bca-7c7fdcfe841c",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### PCSK9 - topmed + hrc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "0b79f400-9b91-41ff-896f-165af12cb0ce",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"==========RSQ 0.3; MAF 001; CADD filtering: \"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 10; TOPMed_v3_snplist: 115; Annot_gene: 125; Full dataframe: 125\"\n",
      "[1] \"==========RSQ 0.3; MAF 001; CADD filtering: _cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 4; TOPMed_v3_snplist: 57; Annot_gene: 61; Full dataframe: 61\"\n",
      "[1] \"==========RSQ 0.3; MAF 0005; CADD filtering: \"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 10; TOPMed_v3_snplist: 115; Annot_gene: 125; Full dataframe: 125\"\n",
      "[1] \"==========RSQ 0.3; MAF 0005; CADD filtering: _cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 4; TOPMed_v3_snplist: 57; Annot_gene: 61; Full dataframe: 61\"\n",
      "[1] \"==========RSQ 0.3; MAF 0001; CADD filtering: \"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 10; TOPMed_v3_snplist: 114; Annot_gene: 124; Full dataframe: 124\"\n",
      "[1] \"==========RSQ 0.3; MAF 0001; CADD filtering: _cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 4; TOPMed_v3_snplist: 56; Annot_gene: 60; Full dataframe: 60\"\n",
      "[1] \"==========RSQ 0.8; MAF 001; CADD filtering: \"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 4; TOPMed_v3_snplist: 34; Annot_gene: 38; Full dataframe: 38\"\n",
      "[1] \"==========RSQ 0.8; MAF 001; CADD filtering: _cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 1; TOPMed_v3_snplist: 15; Annot_gene: 16; Full dataframe: 16\"\n",
      "[1] \"==========RSQ 0.8; MAF 0005; CADD filtering: \"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 4; TOPMed_v3_snplist: 34; Annot_gene: 38; Full dataframe: 38\"\n",
      "[1] \"==========RSQ 0.8; MAF 0005; CADD filtering: _cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 1; TOPMed_v3_snplist: 15; Annot_gene: 16; Full dataframe: 16\"\n",
      "[1] \"==========RSQ 0.8; MAF 0001; CADD filtering: \"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 4; TOPMed_v3_snplist: 33; Annot_gene: 37; Full dataframe: 37\"\n",
      "[1] \"==========RSQ 0.8; MAF 0001; CADD filtering: _cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 1; TOPMed_v3_snplist: 14; Annot_gene: 15; Full dataframe: 15\"\n"
     ]
    }
   ],
   "source": [
    "for(rsq in c(3, 8)){\n",
    "    for(maf in c(\"1\", \"05\", \"01\")){\n",
    "        for(cadd in c(\"\", \"_cadd\")){\n",
    "            print(sprintf(\"==========RSQ 0.%d; MAF 00%s; CADD filtering: %s\", rsq, maf, cadd))\n",
    "            annot_fname <- sprintf(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/hrc_topmed/hrc_topmed_v3_168206ids_rsq0%d_maf00%s%s_annot.csv.gz\", \n",
    "                                   rsq, maf, cadd)\n",
    "            annot_gene <- read.table(annot_fname, sep = \",\", header = TRUE) %>% filter(Gene.refGene == \"PCSK9\") %>% unique()\n",
    "            \n",
    "            full_df <- read.table(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/id_only.csv\", sep = \",\", header = TRUE)            \n",
    "            hrc_fname <- sprintf(\"PCSK9_hrc_rsq0%d_maf00%s%s_extracted.raw\", rsq, maf, cadd)\n",
    "            topmed_fname <- sprintf(\"PCSK9_topmed_v3_rsq0%d_maf00%s%s_extracted.raw\", rsq, maf, cadd)\n",
    "            \n",
    "            hrc_snplist <- annot_gene %>% filter(source == \"hrc\") %>% pull(ID_hg19)\n",
    "            topmed_snplist <- annot_gene %>% filter(source == \"topmed\") %>% pull(ID)\n",
    "\n",
    "            ## hrc matrix\n",
    "            if(file.exists(hrc_fname)){\n",
    "                hrc <- read.table(hrc_fname, sep = \"\\t\", header = TRUE)\n",
    "                hrc_names <- colnames(hrc)[7: dim(hrc)[2]] %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                hrc_names <- stringr::str_replace_all(hrc_names[,1], \"\\\\.\", \":\")\n",
    "                colnames(hrc)[7:dim(hrc)[2]] <- hrc_names\n",
    "\n",
    "                fid_iid <- hrc$IID %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                hrc$FID <- as.integer(fid_iid[,1])\n",
    "                hrc$IID <- as.integer(fid_iid[,2])\n",
    "\n",
    "                hrc <- hrc %>% arrange(FID, IID) %>% select(c(\"FID\", \"IID\", \"PAT\", \"MAT\", \"SEX\", \"PHENOTYPE\", hrc_snplist))\n",
    "                full_df <- left_join(full_df, hrc)\n",
    "            }\n",
    "            \n",
    "            ## topmed matrix\n",
    "            if(file.exists(topmed_fname)){\n",
    "                topmed <- read.table(topmed_fname, sep = \"\\t\", header = TRUE)\n",
    "                topmed_names <- colnames(topmed)[7: dim(topmed)[2]] %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                topmed_names <- stringr::str_replace_all(topmed_names[,1], \"\\\\.\", \":\")\n",
    "                colnames(topmed)[7:dim(topmed)[2]] <- topmed_names\n",
    "\n",
    "                fid_iid <- topmed$IID %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                topmed$FID <- as.integer(fid_iid[,1])\n",
    "                topmed$IID <- as.integer(fid_iid[,2])\n",
    "\n",
    "                topmed <- topmed %>% arrange(FID, IID) %>% select(c(\"FID\", \"IID\", \"PAT\", \"MAT\", \"SEX\", \"PHENOTYPE\", topmed_snplist))\n",
    "                full_df <- left_join(full_df, topmed)\n",
    "            }\n",
    "            \n",
    "            fwrite(full_df, sprintf(\"PCSK9_hrc_topmed_v3_rsq0%d_maf00%s%s_extracted.raw\", rsq, maf, cadd))\n",
    "            \n",
    "            print(sprintf(\"HRC_snplist: %d; TOPMed_v3_snplist: %d; Annot_gene: %d; Full dataframe: %d\", \n",
    "                          length(hrc_snplist), length(topmed_snplist), nrow(annot_gene), ncol(full_df)-6))\n",
    "        }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d146c8e6-b23e-41f4-b5c2-5a731dd36551",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### APOC3 - exome + topmed + hrc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "ae36feaa-7fed-4f86-a37d-c0c4930e6e51",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "setwd(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/apoc3\")\n",
    "\n",
    "library(dplyr)\n",
    "library(genio)\n",
    "library(data.table)\n",
    "library(tidyverse)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "8d487fb1-06a9-4522-b519-f2449a49ed3f",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"==========RSQ 0.3; MAF 001; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_v3_snplist: 5; Exome: 60; Annot_gene: 65; Full dataframe: 65\"\n",
      "[1] \"==========RSQ 0.3; MAF 001; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_v3_snplist: 2; Exome: 26; Annot_gene: 28; Full dataframe: 28\"\n",
      "[1] \"==========RSQ 0.3; MAF 0005; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_v3_snplist: 5; Exome: 60; Annot_gene: 65; Full dataframe: 65\"\n",
      "[1] \"==========RSQ 0.3; MAF 0005; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_v3_snplist: 2; Exome: 26; Annot_gene: 28; Full dataframe: 28\"\n",
      "[1] \"==========RSQ 0.3; MAF 0001; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_v3_snplist: 5; Exome: 59; Annot_gene: 64; Full dataframe: 64\"\n",
      "[1] \"==========RSQ 0.3; MAF 0001; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_v3_snplist: 2; Exome: 25; Annot_gene: 27; Full dataframe: 27\"\n",
      "[1] \"==========RSQ 0.8; MAF 001; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_v3_snplist: 0; Exome: 60; Annot_gene: 60; Full dataframe: 60\"\n",
      "[1] \"==========RSQ 0.8; MAF 001; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_v3_snplist: 0; Exome: 26; Annot_gene: 26; Full dataframe: 26\"\n",
      "[1] \"==========RSQ 0.8; MAF 0005; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_v3_snplist: 0; Exome: 60; Annot_gene: 60; Full dataframe: 60\"\n",
      "[1] \"==========RSQ 0.8; MAF 0005; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_v3_snplist: 0; Exome: 26; Annot_gene: 26; Full dataframe: 26\"\n",
      "[1] \"==========RSQ 0.8; MAF 0001; CADD filtering:\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_v3_snplist: 0; Exome: 59; Annot_gene: 59; Full dataframe: 59\"\n",
      "[1] \"==========RSQ 0.8; MAF 0001; CADD filtering:_cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_v3_snplist: 0; Exome: 25; Annot_gene: 25; Full dataframe: 25\"\n"
     ]
    }
   ],
   "source": [
    "for(rsq in c(3, 8)){\n",
    "    for(maf in c(\"1\", \"05\", \"01\")){\n",
    "        for(cadd in c(\"\", \"_cadd\")){\n",
    "            print(sprintf(\"==========RSQ 0.%d; MAF 00%s; CADD filtering:%s\", rsq, maf, cadd))\n",
    "            \n",
    "            annot_fname <- sprintf(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/hrc_topmed_exome/hrc_topmed_v3_exome_168206ids_chr11_rsq0%d_maf00%s%s_annot.csv.gz\", \n",
    "                       rsq, maf, cadd)\n",
    "            annot_gene <- read.table(annot_fname, sep = \",\", header = TRUE) %>% filter(Gene.refGene == \"APOC3\") %>% unique()\n",
    "            \n",
    "            full_df <- read.table(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/id_only.csv\", sep = \",\", header = TRUE)\n",
    "            \n",
    "            exome_fname <- sprintf(\"APOC3_exome_maf00%s%s_extracted.raw\", maf, cadd)\n",
    "            hrc_fname <- sprintf(\"APOC3_hrc_rsq0%d_maf00%s%s_extracted.raw\", rsq, maf, cadd)\n",
    "            topmed_fname <- sprintf(\"APOC3_topmed_v3_rsq0%d_maf00%s%s_extracted.raw\", rsq, maf, cadd)\n",
    "\n",
    "            hrc_snplist <- annot_gene %>% filter(source == \"hrc\") %>% pull(ID_hg19)\n",
    "            topmed_snplist <- annot_gene %>% filter(source == \"topmed\") %>% pull(ID)\n",
    "            \n",
    "            ## exome matrix\n",
    "            if(file.exists(exome_fname)){\n",
    "                exome <- read.table(exome_fname, sep = \"\\t\", header = TRUE) %>% arrange(FID, IID)\n",
    "                exome_names <- colnames(exome)[7: dim(exome)[2]] %>% \n",
    "                    stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                exome_names <- stringr::str_replace_all(exome_names[,1], \"\\\\.\", \":\")\n",
    "                colnames(exome)[7:dim(exome)[2]] <- exome_names\n",
    "\n",
    "                full_df <- left_join(full_df, exome)\n",
    "            }\n",
    "            \n",
    "            ## hrc matrix\n",
    "            if(file.exists(hrc_fname)){\n",
    "                hrc <- read.table(hrc_fname, sep = \"\\t\", header = TRUE)\n",
    "                hrc_names <- colnames(hrc)[7: dim(hrc)[2]] %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                hrc_names <- stringr::str_replace_all(hrc_names[,1], \"\\\\.\", \":\")\n",
    "                colnames(hrc)[7:dim(hrc)[2]] <- hrc_names\n",
    "\n",
    "                fid_iid <- hrc$IID %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                hrc$FID <- as.integer(fid_iid[,1])\n",
    "                hrc$IID <- as.integer(fid_iid[,2])\n",
    "\n",
    "                hrc <- hrc %>% arrange(FID, IID) %>% select(c(\"FID\", \"IID\", \"PAT\", \"MAT\", \"SEX\", \"PHENOTYPE\", hrc_snplist))\n",
    "                full_df <- left_join(full_df, hrc)\n",
    "            }\n",
    "            \n",
    "            ## topmed matrix\n",
    "            if(file.exists(topmed_fname)){\n",
    "                topmed <- read.table(topmed_fname, sep = \"\\t\", header = TRUE)\n",
    "                topmed_names <- colnames(topmed)[7: dim(topmed)[2]] %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                topmed_names <- stringr::str_replace_all(topmed_names[,1], \"\\\\.\", \":\")\n",
    "                colnames(topmed)[7:dim(topmed)[2]] <- topmed_names\n",
    "\n",
    "                fid_iid <- topmed$IID %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                topmed$FID <- as.integer(fid_iid[,1])\n",
    "                topmed$IID <- as.integer(fid_iid[,2])\n",
    "\n",
    "                topmed <- topmed %>% arrange(FID, IID) %>% select(c(\"FID\", \"IID\", \"PAT\", \"MAT\", \"SEX\", \"PHENOTYPE\", topmed_snplist))\n",
    "                full_df <- left_join(full_df, topmed)\n",
    "            }\n",
    "            \n",
    "            fwrite(full_df, sprintf(\"APOC3_hrc_topmed_v3_exome_rsq0%d_maf00%s%s_extracted.raw\", rsq, maf, cadd))\n",
    "            \n",
    "            print(sprintf(\"HRC_snplist: %d; TOPMed_v3_snplist: %d; Exome: %d; Annot_gene: %d; Full dataframe: %d\", \n",
    "                          length(hrc_snplist), length(topmed_snplist), ncol(exome)-6, nrow(annot_gene), ncol(full_df)-6))\n",
    "        }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a1eb2f3e-f1c6-4f10-9f15-a42649632828",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### APOC3 - topmed + hrc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "e05dd335-c1e4-4b15-9cd2-3bd2b95cbc3d",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"==========RSQ 0.3; MAF 001; CADD filtering: \"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 2; TOPMed_v3_snplist: 16; Annot_gene: 18; Full dataframe: 18\"\n",
      "[1] \"==========RSQ 0.3; MAF 001; CADD filtering: _cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 2; TOPMed_v3_snplist: 8; Annot_gene: 10; Full dataframe: 10\"\n",
      "[1] \"==========RSQ 0.3; MAF 0005; CADD filtering: \"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 2; TOPMed_v3_snplist: 16; Annot_gene: 18; Full dataframe: 18\"\n",
      "[1] \"==========RSQ 0.3; MAF 0005; CADD filtering: _cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 2; TOPMed_v3_snplist: 8; Annot_gene: 10; Full dataframe: 10\"\n",
      "[1] \"==========RSQ 0.3; MAF 0001; CADD filtering: \"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 2; TOPMed_v3_snplist: 15; Annot_gene: 17; Full dataframe: 17\"\n",
      "[1] \"==========RSQ 0.3; MAF 0001; CADD filtering: _cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 2; TOPMed_v3_snplist: 7; Annot_gene: 9; Full dataframe: 9\"\n",
      "[1] \"==========RSQ 0.8; MAF 001; CADD filtering: \"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 1; TOPMed_v3_snplist: 3; Annot_gene: 4; Full dataframe: 4\"\n",
      "[1] \"==========RSQ 0.8; MAF 001; CADD filtering: _cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 1; TOPMed_v3_snplist: 3; Annot_gene: 4; Full dataframe: 4\"\n",
      "[1] \"==========RSQ 0.8; MAF 0005; CADD filtering: \"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 1; TOPMed_v3_snplist: 3; Annot_gene: 4; Full dataframe: 4\"\n",
      "[1] \"==========RSQ 0.8; MAF 0005; CADD filtering: _cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n",
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 1; TOPMed_v3_snplist: 3; Annot_gene: 4; Full dataframe: 4\"\n",
      "[1] \"==========RSQ 0.8; MAF 0001; CADD filtering: \"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_v3_snplist: 3; Annot_gene: 3; Full dataframe: 3\"\n",
      "[1] \"==========RSQ 0.8; MAF 0001; CADD filtering: _cadd\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(FID, IID, PAT, MAT, SEX, PHENOTYPE)`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"HRC_snplist: 0; TOPMed_v3_snplist: 3; Annot_gene: 3; Full dataframe: 3\"\n"
     ]
    }
   ],
   "source": [
    "for(rsq in c(3, 8)){\n",
    "    for(maf in c(\"1\", \"05\", \"01\")){\n",
    "        for(cadd in c(\"\", \"_cadd\")){\n",
    "            print(sprintf(\"==========RSQ 0.%d; MAF 00%s; CADD filtering: %s\", rsq, maf, cadd))\n",
    "            annot_fname <- sprintf(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/hrc_topmed/hrc_topmed_v3_168206ids_chr11_rsq0%d_maf00%s%s_annot.csv.gz\", \n",
    "                                   rsq, maf, cadd)\n",
    "            annot_gene <- read.table(annot_fname, sep = \",\", header = TRUE) %>% filter(Gene.refGene == \"APOC3\") %>% unique()\n",
    "            \n",
    "            full_df <- read.table(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/id_only.csv\", sep = \",\", header = TRUE)\n",
    "            \n",
    "            hrc_fname <- sprintf(\"APOC3_hrc_rsq0%d_maf00%s%s_extracted.raw\", rsq, maf, cadd)\n",
    "            topmed_fname <- sprintf(\"APOC3_topmed_v3_rsq0%d_maf00%s%s_extracted.raw\", rsq, maf, cadd)\n",
    "            \n",
    "            hrc_snplist <- annot_gene %>% filter(source == \"hrc\") %>% pull(ID_hg19)\n",
    "            topmed_snplist <- annot_gene %>% filter(source == \"topmed\") %>% pull(ID)\n",
    "\n",
    "            ## hrc matrix\n",
    "            if(file.exists(hrc_fname)){\n",
    "                hrc <- read.table(hrc_fname, sep = \"\\t\", header = TRUE)\n",
    "                hrc_names <- colnames(hrc)[7: dim(hrc)[2]] %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                hrc_names <- stringr::str_replace_all(hrc_names[,1], \"\\\\.\", \":\")\n",
    "                colnames(hrc)[7:dim(hrc)[2]] <- hrc_names\n",
    "\n",
    "                fid_iid <- hrc$IID %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                hrc$FID <- as.integer(fid_iid[,1])\n",
    "                hrc$IID <- as.integer(fid_iid[,2])\n",
    "\n",
    "                hrc <- hrc %>% arrange(FID, IID) %>% select(c(\"FID\", \"IID\", \"PAT\", \"MAT\", \"SEX\", \"PHENOTYPE\", hrc_snplist))\n",
    "                full_df <- left_join(full_df, hrc)\n",
    "            }\n",
    "            \n",
    "            ## topmed matrix\n",
    "            if(file.exists(topmed_fname)){\n",
    "                topmed <- read.table(topmed_fname, sep = \"\\t\", header = TRUE)\n",
    "                topmed_names <- colnames(topmed)[7: dim(topmed)[2]] %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                topmed_names <- stringr::str_replace_all(topmed_names[,1], \"\\\\.\", \":\")\n",
    "                colnames(topmed)[7:dim(topmed)[2]] <- topmed_names\n",
    "\n",
    "                fid_iid <- topmed$IID %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "                topmed$FID <- as.integer(fid_iid[,1])\n",
    "                topmed$IID <- as.integer(fid_iid[,2])\n",
    "\n",
    "                topmed <- topmed %>% arrange(FID, IID) %>% select(c(\"FID\", \"IID\", \"PAT\", \"MAT\", \"SEX\", \"PHENOTYPE\", topmed_snplist))\n",
    "                full_df <- left_join(full_df, topmed)\n",
    "            }\n",
    "            \n",
    "            fwrite(full_df, sprintf(\"APOC3_hrc_topmed_v3_rsq0%d_maf00%s%s_extracted.raw\", rsq, maf, cadd))\n",
    "            \n",
    "            print(sprintf(\"HRC_snplist: %d; TOPMed_v3_snplist: %d; Annot_gene: %d; Full dataframe: %d\", \n",
    "                          length(hrc_snplist), length(topmed_snplist), nrow(annot_gene), ncol(full_df)-6))\n",
    "        }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4015f470-c66f-42c5-a48f-76e0bb314495",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "## Phenotype Analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "327b6ed1-d5c7-46c6-a7d7-dd267acb9869",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "We are going to be using Burden of Rare Variant ([BRV](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4459641/)) to analyze rare variant aggregate association for PCSK9 and APOC3. For phenotype analysis, we control for age, sex and 2 PC's."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b74829db-31be-4c55-b5b3-dabbac423253",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "### Functions and Packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "38b89472-3a0c-4605-9778-3069b2c8f3cd",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "library(dplyr)\n",
    "library(genio)\n",
    "library(data.table)\n",
    "library(tidyverse)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "3129d96d-bceb-407a-bdb9-c4a3b6c3bdf0",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "# mi_pheno <- read.csv(\"/home/tl3031/project/imputation-rvtest/analysis/imputation_aggregated_analysis/Unrealted_White_EU_Both_Exo_Impu_5209_MI_cases.phe.csv\")\n",
    "# mi_y <- mi_pheno %>% arrange(FID) %>% pull(MI)\n",
    "\n",
    "###########################################\n",
    "# trig_ldl_pheno <- read.csv(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/Trigly_LDL_Log10_168206ind.phe.csv\") %>% arrange(FID)\n",
    "# tg_y <- trig_ldl_pheno %>% arrange(FID) %>% pull(logarithm_base10_Trigli)\n",
    "# ldl_y <- trig_ldl_pheno  %>% arrange(FID) %>% pull(f.30780.0.0)\n",
    "\n",
    "###########################################\n",
    "ldl_df <- read.csv(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/UKBB_LDL_159759inds_4PCs.pheno\", sep = \"\\t\") %>% arrange(FID)\n",
    "ldl_y <- ldl_df %>% pull(f.30780.0.0)\n",
    "ldl_id <- ldl_df %>% pull(IID)\n",
    "ldl_cov <- ldl_df %>% select(sex, age, PC1, PC2)\n",
    "\n",
    "tg_df <- read.csv(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/UKBB_TG_159759inds_4PCs.pheno\", sep = \"\\t\") %>% arrange(FID)\n",
    "tg_y <- tg_df %>% pull(logarithm_base10_Trigli)\n",
    "tg_id <- tg_df %>% pull(IID)\n",
    "tg_cov <- tg_df %>% select(sex, age, PC1, PC2)\n",
    "\n",
    "rm(ldl_df, tg_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "a46975bb-1da7-4146-bc3c-69239154c6d4",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "# the BRV function\n",
    "BRV <- function(X_mat, y, cov, option = \"binomial\"){\n",
    "    X_mat <- as.matrix(X_mat)\n",
    "    y <- as.vector(y)\n",
    "\n",
    "    ## cout number of rare variants for each individual\n",
    "    if(ncol(X_mat) == 1){\n",
    "        X_mat_new = X_mat\n",
    "    } else {\n",
    "        X_mat_new  = rowSums(X_mat, na.rm=TRUE) %>% as.matrix()\n",
    "    }\n",
    "    \n",
    "    age <- cov$age\n",
    "    sex <- cov$sex\n",
    "    pc1 <- cov$PC1\n",
    "    pc2 <- cov$PC2\n",
    "\n",
    "    ## obtain p value by logistic regression\n",
    "    mat.fit <- glm(y ~ X_mat_new + age + sex + pc1 + pc2, family = option)\n",
    "    zstat <- coef(summary(mat.fit))[2, 3]\n",
    "    pval <- coef(summary(mat.fit))[2, 4]\n",
    "\n",
    "    return(list(zstat = zstat, pval = pval))\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ff62b4f2-83cd-4023-8421-9f5a4fcd3749",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### PCSK9 v.s. LDL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "bc338eb9-bf19-4f48-868e-15d5c0d8f2f4",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "setwd('/mnt/vast/hpc/csg/tl3031/imputation-rvtest/analysis/imputation_aggregated_analysis/pcsk9')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "b1a555bd-d389-4008-9825-cdb3a0d559fa",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "result_df <- data.frame(data.frame(matrix(ncol = 5, nrow = 0)))\n",
    "colnames(result_df) <- c(\"data\", \"rsqs\", \"mafs\", \"num_var\", \"pvals\")\n",
    "\n",
    "for(maf in c(\"1\", \"05\", \"01\")){\n",
    "    exome_fname <- sprintf(\"PCSK9_exome_maf00%s_extracted.raw\", maf)\n",
    "    if(file.exists(exome_fname)){\n",
    "        exome <- read.table(exome_fname, sep = \"\\t\", header = TRUE) %>% arrange(FID, IID)\n",
    "        exome_X <- exome %>% arrange(FID, IID) %>% filter(FID %in% ldl_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "        result_exome <- BRV(2 - exome_X, ldl_y, ldl_cov,\"gaussian\")\n",
    "    } else {\n",
    "        exome_X <- matrix()\n",
    "        result_exome$pval <- NA\n",
    "    }\n",
    "\n",
    "    for(rsq in c(3, 8)){\n",
    "        hrc_fname <- sprintf(\"PCSK9_hrc_rsq0%d_maf00%s_extracted.raw\", rsq, maf)\n",
    "        topmed_fname <- sprintf(\"PCSK9_topmed_v3_rsq0%d_maf00%s_extracted.raw\", rsq, maf)\n",
    "        hrc_topmed_fname <- sprintf(\"PCSK9_hrc_topmed_v3_rsq0%d_maf00%s_extracted.raw\", rsq, maf)\n",
    "        hrc_topmed_exome_fname <- sprintf(\"PCSK9_hrc_topmed_v3_exome_rsq0%d_maf00%s_extracted.raw\", rsq, maf)\n",
    "\n",
    "        if(file.exists(hrc_fname)){\n",
    "            hrc <- read.table(hrc_fname, sep = \"\\t\", header = TRUE)\n",
    "            \n",
    "            fid_iid <- hrc$IID %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "            hrc$FID <- as.integer(fid_iid[,1])\n",
    "            hrc$IID <- as.integer(fid_iid[,2])\n",
    "            hrc_X <- hrc %>% arrange(FID, IID) %>% filter(FID %in% ldl_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "\n",
    "            result_hrc <- BRV(2 - hrc_X, ldl_y, ldl_cov, \"gaussian\")\n",
    "\n",
    "        } else {\n",
    "            hrc_X <- matrix()\n",
    "            result_hrc$pval <- NA\n",
    "        }\n",
    "        \n",
    "        if(file.exists(topmed_fname)){\n",
    "            topmed <- read.table(topmed_fname, sep = \"\\t\", header = TRUE)\n",
    "            \n",
    "            fid_iid <- topmed$IID %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "            topmed$FID <- as.integer(fid_iid[,1])\n",
    "            topmed$IID <- as.integer(fid_iid[,2])\n",
    "\n",
    "            topmed_X <- topmed %>% arrange(FID, IID) %>% filter(FID %in% ldl_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "            result_topmed <- BRV(2 - topmed_X, ldl_y, ldl_cov, \"gaussian\")\n",
    "        } else {\n",
    "            topmed_X <- matrix()\n",
    "            result_topmed$pval <- NA\n",
    "        }\n",
    "\n",
    "        if(file.exists(hrc_topmed_fname)){\n",
    "            hrc_topmed <- read.table(hrc_topmed_fname, sep = \",\", header = TRUE)\n",
    "            hrc_topmed_X <- hrc_topmed %>% arrange(FID, IID) %>% filter(FID %in% ldl_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "            result_hrc_topmed <- BRV(2 - hrc_topmed_X, ldl_y, ldl_cov, \"gaussian\")\n",
    "        } else {\n",
    "            hrc_topmed_X <- matrix()\n",
    "            result_hrc_topmed$pval <- NA\n",
    "        }\n",
    "\n",
    "        if(file.exists(hrc_topmed_exome_fname)){\n",
    "            hrc_topmed_exome <- read.table(hrc_topmed_exome_fname, sep = \",\", header = TRUE)\n",
    "            hrc_topmed_exome_X <- hrc_topmed_exome %>% arrange(FID, IID) %>% filter(FID %in% ldl_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "            result_hrc_topmed_exome <- BRV(2 - hrc_topmed_exome_X, ldl_y, ldl_cov, \"gaussian\")\n",
    "        } else {\n",
    "            hrc_topmed_exome_X <- matrix()\n",
    "            result_hrc_topmed_exome$pval <- NA\n",
    "        }\n",
    "\n",
    "        data <- c(\"hrc\", \"topmed\", \"hrc_topmed\", \"hrc_topmed_exome\")\n",
    "        rsqs <- rep(rsq/10, 4)\n",
    "        mafs <- rep(paste0(\"0.0\", maf), 4)\n",
    "        pvals <- c(result_hrc$pval, result_topmed$pval, result_hrc_topmed$pval, result_hrc_topmed_exome$pval)\n",
    "        num_var <- c(ncol(hrc_X), ncol(topmed_X), ncol(hrc_topmed_X), ncol(hrc_topmed_exome_X))\n",
    "        sub_df <- data.frame(data, rsqs, mafs, num_var, pvals)\n",
    "        result_df <- rbind(result_df, sub_df)\n",
    "    }\n",
    "    result_df <- rbind(data.frame(data = \"exome\", rsqs = \"NA\", mafs = paste0(\"0.0\", maf), num_var = ncol(exome_X), pvals = result_exome$pval), result_df)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "cf4f5985-85db-4acb-a196-1a246a127bcb",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 27 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>data</th><th scope=col>rsqs</th><th scope=col>mafs</th><th scope=col>num_var</th><th scope=col>pvals</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.01 </td><td>394</td><td>5.473985e-71</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.01 </td><td> 16</td><td>1.944983e-12</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.01 </td><td>  6</td><td>6.993942e-11</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.01 </td><td>124</td><td>2.403941e-38</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.01 </td><td> 34</td><td>7.399230e-30</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.01 </td><td>125</td><td>1.146106e-39</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.01 </td><td> 38</td><td>5.148051e-32</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.01 </td><td>412</td><td>8.865311e-71</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.01 </td><td>394</td><td>5.473985e-71</td></tr>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.005</td><td>394</td><td>5.473985e-71</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.005</td><td> 16</td><td>1.944983e-12</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.005</td><td>  6</td><td>6.993942e-11</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.005</td><td>124</td><td>2.403941e-38</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.005</td><td> 34</td><td>7.399230e-30</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.005</td><td>125</td><td>1.146106e-39</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.005</td><td> 38</td><td>5.148051e-32</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.005</td><td>412</td><td>8.865311e-71</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.005</td><td>394</td><td>5.473985e-71</td></tr>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.001</td><td>393</td><td>9.347929e-53</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.001</td><td> 15</td><td>2.387199e-09</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.001</td><td>  6</td><td>6.993942e-11</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.001</td><td>123</td><td>5.232206e-26</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.001</td><td> 33</td><td>5.721054e-17</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.001</td><td>124</td><td>4.189196e-27</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.001</td><td> 37</td><td>3.530685e-19</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.001</td><td>411</td><td>1.368744e-52</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.001</td><td>393</td><td>9.347929e-53</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 27 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " data & rsqs & mafs & num\\_var & pvals\\\\\n",
       " <fct> & <chr> & <chr> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t exome            & NA  & 0.01  & 394 & 5.473985e-71\\\\\n",
       "\t hrc              & 0.3 & 0.01  &  16 & 1.944983e-12\\\\\n",
       "\t hrc              & 0.8 & 0.01  &   6 & 6.993942e-11\\\\\n",
       "\t topmed           & 0.3 & 0.01  & 124 & 2.403941e-38\\\\\n",
       "\t topmed           & 0.8 & 0.01  &  34 & 7.399230e-30\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.01  & 125 & 1.146106e-39\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.01  &  38 & 5.148051e-32\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.01  & 412 & 8.865311e-71\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.01  & 394 & 5.473985e-71\\\\\n",
       "\t exome            & NA  & 0.005 & 394 & 5.473985e-71\\\\\n",
       "\t hrc              & 0.3 & 0.005 &  16 & 1.944983e-12\\\\\n",
       "\t hrc              & 0.8 & 0.005 &   6 & 6.993942e-11\\\\\n",
       "\t topmed           & 0.3 & 0.005 & 124 & 2.403941e-38\\\\\n",
       "\t topmed           & 0.8 & 0.005 &  34 & 7.399230e-30\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.005 & 125 & 1.146106e-39\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.005 &  38 & 5.148051e-32\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.005 & 412 & 8.865311e-71\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.005 & 394 & 5.473985e-71\\\\\n",
       "\t exome            & NA  & 0.001 & 393 & 9.347929e-53\\\\\n",
       "\t hrc              & 0.3 & 0.001 &  15 & 2.387199e-09\\\\\n",
       "\t hrc              & 0.8 & 0.001 &   6 & 6.993942e-11\\\\\n",
       "\t topmed           & 0.3 & 0.001 & 123 & 5.232206e-26\\\\\n",
       "\t topmed           & 0.8 & 0.001 &  33 & 5.721054e-17\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.001 & 124 & 4.189196e-27\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.001 &  37 & 3.530685e-19\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.001 & 411 & 1.368744e-52\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.001 & 393 & 9.347929e-53\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 27 × 5\n",
       "\n",
       "| data &lt;fct&gt; | rsqs &lt;chr&gt; | mafs &lt;chr&gt; | num_var &lt;int&gt; | pvals &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| exome            | NA  | 0.01  | 394 | 5.473985e-71 |\n",
       "| hrc              | 0.3 | 0.01  |  16 | 1.944983e-12 |\n",
       "| hrc              | 0.8 | 0.01  |   6 | 6.993942e-11 |\n",
       "| topmed           | 0.3 | 0.01  | 124 | 2.403941e-38 |\n",
       "| topmed           | 0.8 | 0.01  |  34 | 7.399230e-30 |\n",
       "| hrc_topmed       | 0.3 | 0.01  | 125 | 1.146106e-39 |\n",
       "| hrc_topmed       | 0.8 | 0.01  |  38 | 5.148051e-32 |\n",
       "| hrc_topmed_exome | 0.3 | 0.01  | 412 | 8.865311e-71 |\n",
       "| hrc_topmed_exome | 0.8 | 0.01  | 394 | 5.473985e-71 |\n",
       "| exome            | NA  | 0.005 | 394 | 5.473985e-71 |\n",
       "| hrc              | 0.3 | 0.005 |  16 | 1.944983e-12 |\n",
       "| hrc              | 0.8 | 0.005 |   6 | 6.993942e-11 |\n",
       "| topmed           | 0.3 | 0.005 | 124 | 2.403941e-38 |\n",
       "| topmed           | 0.8 | 0.005 |  34 | 7.399230e-30 |\n",
       "| hrc_topmed       | 0.3 | 0.005 | 125 | 1.146106e-39 |\n",
       "| hrc_topmed       | 0.8 | 0.005 |  38 | 5.148051e-32 |\n",
       "| hrc_topmed_exome | 0.3 | 0.005 | 412 | 8.865311e-71 |\n",
       "| hrc_topmed_exome | 0.8 | 0.005 | 394 | 5.473985e-71 |\n",
       "| exome            | NA  | 0.001 | 393 | 9.347929e-53 |\n",
       "| hrc              | 0.3 | 0.001 |  15 | 2.387199e-09 |\n",
       "| hrc              | 0.8 | 0.001 |   6 | 6.993942e-11 |\n",
       "| topmed           | 0.3 | 0.001 | 123 | 5.232206e-26 |\n",
       "| topmed           | 0.8 | 0.001 |  33 | 5.721054e-17 |\n",
       "| hrc_topmed       | 0.3 | 0.001 | 124 | 4.189196e-27 |\n",
       "| hrc_topmed       | 0.8 | 0.001 |  37 | 3.530685e-19 |\n",
       "| hrc_topmed_exome | 0.3 | 0.001 | 411 | 1.368744e-52 |\n",
       "| hrc_topmed_exome | 0.8 | 0.001 | 393 | 9.347929e-53 |\n",
       "\n"
      ],
      "text/plain": [
       "   data             rsqs mafs  num_var pvals       \n",
       "1  exome            NA   0.01  394     5.473985e-71\n",
       "2  hrc              0.3  0.01   16     1.944983e-12\n",
       "3  hrc              0.8  0.01    6     6.993942e-11\n",
       "4  topmed           0.3  0.01  124     2.403941e-38\n",
       "5  topmed           0.8  0.01   34     7.399230e-30\n",
       "6  hrc_topmed       0.3  0.01  125     1.146106e-39\n",
       "7  hrc_topmed       0.8  0.01   38     5.148051e-32\n",
       "8  hrc_topmed_exome 0.3  0.01  412     8.865311e-71\n",
       "9  hrc_topmed_exome 0.8  0.01  394     5.473985e-71\n",
       "10 exome            NA   0.005 394     5.473985e-71\n",
       "11 hrc              0.3  0.005  16     1.944983e-12\n",
       "12 hrc              0.8  0.005   6     6.993942e-11\n",
       "13 topmed           0.3  0.005 124     2.403941e-38\n",
       "14 topmed           0.8  0.005  34     7.399230e-30\n",
       "15 hrc_topmed       0.3  0.005 125     1.146106e-39\n",
       "16 hrc_topmed       0.8  0.005  38     5.148051e-32\n",
       "17 hrc_topmed_exome 0.3  0.005 412     8.865311e-71\n",
       "18 hrc_topmed_exome 0.8  0.005 394     5.473985e-71\n",
       "19 exome            NA   0.001 393     9.347929e-53\n",
       "20 hrc              0.3  0.001  15     2.387199e-09\n",
       "21 hrc              0.8  0.001   6     6.993942e-11\n",
       "22 topmed           0.3  0.001 123     5.232206e-26\n",
       "23 topmed           0.8  0.001  33     5.721054e-17\n",
       "24 hrc_topmed       0.3  0.001 124     4.189196e-27\n",
       "25 hrc_topmed       0.8  0.001  37     3.530685e-19\n",
       "26 hrc_topmed_exome 0.3  0.001 411     1.368744e-52\n",
       "27 hrc_topmed_exome 0.8  0.001 393     9.347929e-53"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "result_df %>% \n",
    "    mutate(data = factor(data, levels = c(\"exome\", \"hrc\", \"topmed\", \"hrc_topmed\", \"hrc_topmed_exome\"))) %>% \n",
    "    arrange(desc(as.numeric(mafs)), data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "447e8f31-687a-4ce3-823f-907d496ac558",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "result_df <- data.frame(data.frame(matrix(ncol = 5, nrow = 0)))\n",
    "colnames(result_df) <- c(\"data\", \"rsqs\", \"mafs\", \"num_var\", \"pvals\")\n",
    "\n",
    "for(maf in c(\"1\", \"05\", \"01\")){\n",
    "    exome_fname <- sprintf(\"PCSK9_exome_maf00%s_cadd_extracted.raw\", maf)\n",
    "    if(file.exists(exome_fname)){\n",
    "        exome <- read.table(exome_fname, sep = \"\\t\", header = TRUE) %>% arrange(FID, IID)\n",
    "        exome_X <- exome %>% arrange(FID, IID) %>% filter(FID %in% ldl_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "        result_exome <- BRV(2 - exome_X, ldl_y, ldl_cov, \"gaussian\")\n",
    "    } else {\n",
    "        exome_X <- matrix()\n",
    "        result_exome$pval <- NA\n",
    "    }\n",
    "\n",
    "    for(rsq in c(3, 8)){\n",
    "        hrc_fname <- sprintf(\"PCSK9_hrc_rsq0%d_maf00%s_cadd_extracted.raw\", rsq, maf)\n",
    "        topmed_fname <- sprintf(\"PCSK9_topmed_v3_rsq0%d_maf00%s_cadd_extracted.raw\", rsq, maf)\n",
    "        hrc_topmed_fname <- sprintf(\"PCSK9_hrc_topmed_v3_rsq0%d_maf00%s_cadd_extracted.raw\", rsq, maf)\n",
    "        hrc_topmed_exome_fname <- sprintf(\"PCSK9_hrc_topmed_v3_exome_rsq0%d_maf00%s_cadd_extracted.raw\", rsq, maf)\n",
    "\n",
    "        if(file.exists(hrc_fname)){\n",
    "            hrc <- read.table(hrc_fname, sep = \"\\t\", header = TRUE)\n",
    "            \n",
    "            fid_iid <- hrc$IID %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "            hrc$FID <- as.integer(fid_iid[,1])\n",
    "            hrc$IID <- as.integer(fid_iid[,2])\n",
    "            hrc_X <- hrc %>% arrange(FID, IID) %>% filter(FID %in% ldl_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "\n",
    "            result_hrc <- BRV(2 - hrc_X, ldl_y, ldl_cov,\"gaussian\")\n",
    "\n",
    "        } else {\n",
    "            hrc_X <- matrix()\n",
    "            result_hrc$pval <- NA\n",
    "        }\n",
    "        \n",
    "        if(file.exists(topmed_fname)){\n",
    "            topmed <- read.table(topmed_fname, sep = \"\\t\", header = TRUE)\n",
    "            \n",
    "            fid_iid <- topmed$IID %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "            topmed$FID <- as.integer(fid_iid[,1])\n",
    "            topmed$IID <- as.integer(fid_iid[,2])\n",
    "\n",
    "            topmed_X <- topmed %>% arrange(FID, IID) %>% filter(FID %in% ldl_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "            result_topmed <- BRV(2 - topmed_X, ldl_y, ldl_cov,\"gaussian\")\n",
    "        } else {\n",
    "            topmed_X <- matrix()\n",
    "            result_topmed$pval <- NA\n",
    "        }\n",
    "\n",
    "        if(file.exists(hrc_topmed_fname)){\n",
    "            hrc_topmed <- read.table(hrc_topmed_fname, sep = \",\", header = TRUE)\n",
    "            hrc_topmed_X <- hrc_topmed %>% arrange(FID, IID) %>% filter(FID %in% ldl_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "            result_hrc_topmed <- BRV(2 - hrc_topmed_X, ldl_y, ldl_cov,\"gaussian\")\n",
    "        } else {\n",
    "            hrc_topmed_X <- matrix()\n",
    "            result_hrc_topmed$pval <- NA\n",
    "        }\n",
    "\n",
    "        if(file.exists(hrc_topmed_exome_fname)){\n",
    "            hrc_topmed_exome <- read.table(hrc_topmed_exome_fname, sep = \",\", header = TRUE)\n",
    "            hrc_topmed_exome_X <- hrc_topmed_exome %>% arrange(FID, IID) %>% filter(FID %in% ldl_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "            result_hrc_topmed_exome <- BRV(2 - hrc_topmed_exome_X, ldl_y, ldl_cov,\"gaussian\")\n",
    "        } else {\n",
    "            hrc_topmed_exome_X <- matrix()\n",
    "            result_hrc_topmed_exome$pval <- NA\n",
    "        }\n",
    "\n",
    "        data <- c(\"hrc\", \"topmed\", \"hrc_topmed\", \"hrc_topmed_exome\")\n",
    "        rsqs <- rep(rsq/10, 4)\n",
    "        mafs <- rep(paste0(\"0.0\", maf), 4)\n",
    "        pvals <- c(result_hrc$pval, result_topmed$pval, result_hrc_topmed$pval, result_hrc_topmed_exome$pval)\n",
    "        num_var <- c(ncol(hrc_X), ncol(topmed_X), ncol(hrc_topmed_X), ncol(hrc_topmed_exome_X))\n",
    "        sub_df <- data.frame(data, rsqs, mafs, num_var, pvals)\n",
    "        result_df <- rbind(result_df, sub_df)\n",
    "    }\n",
    "    result_df <- rbind(data.frame(data = \"exome\", rsqs = \"NA\", mafs = paste0(\"0.0\", maf), num_var = ncol(exome_X), pvals = result_exome$pval), result_df)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "63dbcb72-fc6e-4d98-8cf7-1fbb94ae1f9e",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 27 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>data</th><th scope=col>rsqs</th><th scope=col>mafs</th><th scope=col>num_var</th><th scope=col>pvals</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.01 </td><td>200</td><td>8.246876e-93</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.01 </td><td>  6</td><td>1.912681e-12</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.01 </td><td>  2</td><td>4.563676e-15</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.01 </td><td> 61</td><td>1.287643e-54</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.01 </td><td> 15</td><td>1.227744e-44</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.01 </td><td> 61</td><td>3.754704e-54</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.01 </td><td> 16</td><td>3.683660e-45</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.01 </td><td>204</td><td>2.554497e-93</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.01 </td><td>200</td><td>8.246876e-93</td></tr>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.005</td><td>200</td><td>8.246876e-93</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.005</td><td>  6</td><td>1.912681e-12</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.005</td><td>  2</td><td>4.563676e-15</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.005</td><td> 61</td><td>1.287643e-54</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.005</td><td> 15</td><td>1.227744e-44</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.005</td><td> 61</td><td>3.754704e-54</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.005</td><td> 16</td><td>3.683660e-45</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.005</td><td>204</td><td>2.554497e-93</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.005</td><td>200</td><td>8.246876e-93</td></tr>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.001</td><td>199</td><td>2.525363e-74</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.001</td><td>  5</td><td>5.727402e-12</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.001</td><td>  2</td><td>4.563676e-15</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.001</td><td> 60</td><td>9.894265e-42</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.001</td><td> 14</td><td>3.983099e-34</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.001</td><td> 60</td><td>3.143422e-41</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.001</td><td> 15</td><td>1.152055e-34</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.001</td><td>203</td><td>7.679194e-75</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.001</td><td>199</td><td>2.525363e-74</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 27 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " data & rsqs & mafs & num\\_var & pvals\\\\\n",
       " <fct> & <chr> & <chr> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t exome            & NA  & 0.01  & 200 & 8.246876e-93\\\\\n",
       "\t hrc              & 0.3 & 0.01  &   6 & 1.912681e-12\\\\\n",
       "\t hrc              & 0.8 & 0.01  &   2 & 4.563676e-15\\\\\n",
       "\t topmed           & 0.3 & 0.01  &  61 & 1.287643e-54\\\\\n",
       "\t topmed           & 0.8 & 0.01  &  15 & 1.227744e-44\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.01  &  61 & 3.754704e-54\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.01  &  16 & 3.683660e-45\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.01  & 204 & 2.554497e-93\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.01  & 200 & 8.246876e-93\\\\\n",
       "\t exome            & NA  & 0.005 & 200 & 8.246876e-93\\\\\n",
       "\t hrc              & 0.3 & 0.005 &   6 & 1.912681e-12\\\\\n",
       "\t hrc              & 0.8 & 0.005 &   2 & 4.563676e-15\\\\\n",
       "\t topmed           & 0.3 & 0.005 &  61 & 1.287643e-54\\\\\n",
       "\t topmed           & 0.8 & 0.005 &  15 & 1.227744e-44\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.005 &  61 & 3.754704e-54\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.005 &  16 & 3.683660e-45\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.005 & 204 & 2.554497e-93\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.005 & 200 & 8.246876e-93\\\\\n",
       "\t exome            & NA  & 0.001 & 199 & 2.525363e-74\\\\\n",
       "\t hrc              & 0.3 & 0.001 &   5 & 5.727402e-12\\\\\n",
       "\t hrc              & 0.8 & 0.001 &   2 & 4.563676e-15\\\\\n",
       "\t topmed           & 0.3 & 0.001 &  60 & 9.894265e-42\\\\\n",
       "\t topmed           & 0.8 & 0.001 &  14 & 3.983099e-34\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.001 &  60 & 3.143422e-41\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.001 &  15 & 1.152055e-34\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.001 & 203 & 7.679194e-75\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.001 & 199 & 2.525363e-74\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 27 × 5\n",
       "\n",
       "| data &lt;fct&gt; | rsqs &lt;chr&gt; | mafs &lt;chr&gt; | num_var &lt;int&gt; | pvals &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| exome            | NA  | 0.01  | 200 | 8.246876e-93 |\n",
       "| hrc              | 0.3 | 0.01  |   6 | 1.912681e-12 |\n",
       "| hrc              | 0.8 | 0.01  |   2 | 4.563676e-15 |\n",
       "| topmed           | 0.3 | 0.01  |  61 | 1.287643e-54 |\n",
       "| topmed           | 0.8 | 0.01  |  15 | 1.227744e-44 |\n",
       "| hrc_topmed       | 0.3 | 0.01  |  61 | 3.754704e-54 |\n",
       "| hrc_topmed       | 0.8 | 0.01  |  16 | 3.683660e-45 |\n",
       "| hrc_topmed_exome | 0.3 | 0.01  | 204 | 2.554497e-93 |\n",
       "| hrc_topmed_exome | 0.8 | 0.01  | 200 | 8.246876e-93 |\n",
       "| exome            | NA  | 0.005 | 200 | 8.246876e-93 |\n",
       "| hrc              | 0.3 | 0.005 |   6 | 1.912681e-12 |\n",
       "| hrc              | 0.8 | 0.005 |   2 | 4.563676e-15 |\n",
       "| topmed           | 0.3 | 0.005 |  61 | 1.287643e-54 |\n",
       "| topmed           | 0.8 | 0.005 |  15 | 1.227744e-44 |\n",
       "| hrc_topmed       | 0.3 | 0.005 |  61 | 3.754704e-54 |\n",
       "| hrc_topmed       | 0.8 | 0.005 |  16 | 3.683660e-45 |\n",
       "| hrc_topmed_exome | 0.3 | 0.005 | 204 | 2.554497e-93 |\n",
       "| hrc_topmed_exome | 0.8 | 0.005 | 200 | 8.246876e-93 |\n",
       "| exome            | NA  | 0.001 | 199 | 2.525363e-74 |\n",
       "| hrc              | 0.3 | 0.001 |   5 | 5.727402e-12 |\n",
       "| hrc              | 0.8 | 0.001 |   2 | 4.563676e-15 |\n",
       "| topmed           | 0.3 | 0.001 |  60 | 9.894265e-42 |\n",
       "| topmed           | 0.8 | 0.001 |  14 | 3.983099e-34 |\n",
       "| hrc_topmed       | 0.3 | 0.001 |  60 | 3.143422e-41 |\n",
       "| hrc_topmed       | 0.8 | 0.001 |  15 | 1.152055e-34 |\n",
       "| hrc_topmed_exome | 0.3 | 0.001 | 203 | 7.679194e-75 |\n",
       "| hrc_topmed_exome | 0.8 | 0.001 | 199 | 2.525363e-74 |\n",
       "\n"
      ],
      "text/plain": [
       "   data             rsqs mafs  num_var pvals       \n",
       "1  exome            NA   0.01  200     8.246876e-93\n",
       "2  hrc              0.3  0.01    6     1.912681e-12\n",
       "3  hrc              0.8  0.01    2     4.563676e-15\n",
       "4  topmed           0.3  0.01   61     1.287643e-54\n",
       "5  topmed           0.8  0.01   15     1.227744e-44\n",
       "6  hrc_topmed       0.3  0.01   61     3.754704e-54\n",
       "7  hrc_topmed       0.8  0.01   16     3.683660e-45\n",
       "8  hrc_topmed_exome 0.3  0.01  204     2.554497e-93\n",
       "9  hrc_topmed_exome 0.8  0.01  200     8.246876e-93\n",
       "10 exome            NA   0.005 200     8.246876e-93\n",
       "11 hrc              0.3  0.005   6     1.912681e-12\n",
       "12 hrc              0.8  0.005   2     4.563676e-15\n",
       "13 topmed           0.3  0.005  61     1.287643e-54\n",
       "14 topmed           0.8  0.005  15     1.227744e-44\n",
       "15 hrc_topmed       0.3  0.005  61     3.754704e-54\n",
       "16 hrc_topmed       0.8  0.005  16     3.683660e-45\n",
       "17 hrc_topmed_exome 0.3  0.005 204     2.554497e-93\n",
       "18 hrc_topmed_exome 0.8  0.005 200     8.246876e-93\n",
       "19 exome            NA   0.001 199     2.525363e-74\n",
       "20 hrc              0.3  0.001   5     5.727402e-12\n",
       "21 hrc              0.8  0.001   2     4.563676e-15\n",
       "22 topmed           0.3  0.001  60     9.894265e-42\n",
       "23 topmed           0.8  0.001  14     3.983099e-34\n",
       "24 hrc_topmed       0.3  0.001  60     3.143422e-41\n",
       "25 hrc_topmed       0.8  0.001  15     1.152055e-34\n",
       "26 hrc_topmed_exome 0.3  0.001 203     7.679194e-75\n",
       "27 hrc_topmed_exome 0.8  0.001 199     2.525363e-74"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "result_df %>% \n",
    "    mutate(data = factor(data, levels = c(\"exome\", \"hrc\", \"topmed\", \"hrc_topmed\", \"hrc_topmed_exome\"))) %>% \n",
    "    arrange(desc(as.numeric(mafs)), data)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1023be46-7d3a-4fad-b525-240481c70e80",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "#### PCSK9: chr1:55052701:C:T (hg38)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "b9d5206b-b654-4789-9335-b9d59fafe742",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "# pcsk9_id_maf001 <- read.csv(\"PCSK9_exome_maf001_annot.csv.gz\")$ID_hg38\n",
    "# pcsk9_id_maf0001 <- read.csv(\"PCSK9_exome_maf0001_annot.csv.gz\")$ID_hg38\n",
    "# setdiff(pcsk9_id_maf001, pcsk9_id_maf0001)\n",
    "\n",
    "# read.csv(\"PCSK9_exome_maf001_annot.csv.gz\") %>% filter(ID_hg38 == 'chr1:55052701:C:T')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4c87a3f8-b946-43b9-8ccc-1052f9713c2d",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "We are interested in this variant, because we see that after removal, the p-value has increased a lot. Therefore, we are going to conduct a single varaint association testing for this variants alone using exome and imputed data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "b48ea397-b076-40c1-b262-1b13c4d655d0",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "# ## exome data\n",
    "# exome_bim <- read_bim(\"PCSK9_exome_maf001_extracted.bim\")\n",
    "# exome_fam <- read_fam(\"PCSK9_exome_maf001_extracted.fam\")\n",
    "# exome <- read_bed(\"PCSK9_exome_maf001_extracted.bed\", exome_bim$id, exome_fam$id)\n",
    "\n",
    "# exome_X <- data.frame(ID = names(exome[\"chr1:55052701:C:T\",]), value=exome[\"chr1:55052701:C:T\",], row.names=NULL)\n",
    "# exome_X <- exome_X %>% arrange(as.numeric(ID)) %>% filter(ID %in% ldl_id) %>% select(-ID) %>% as.matrix()\n",
    "\n",
    "# exome.fit <- glm(ldl_y ~ exome_X + ldl_cov$age + ldl_cov$sex + ldl_cov$PC1 + ldl_cov$PC2, family = \"gaussian\")\n",
    "# coef(summary(exome.fit))[2, 4]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "7aa5fe18-a4f4-40a8-97a0-77f0d9fca848",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "# read.csv(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/hrc_topmed/hrc_topmed_v3_168206ids_rsq03_maf001_annot.csv.gz\") %>%\n",
    "#     filter(ID_hg38 == \"chr1:55052701:C:T\")\n",
    "\n",
    "# ## hrc\n",
    "# hrc <- read.csv(\"PCSK9_hrc_rsq03_maf001_extracted.traw\") %>% filter(SNP == \"chr1:55518374:C:T\")\n",
    "# hrc_X <- format_traw(hrc, ldl_id, \"hrc\")\n",
    "\n",
    "# hrc.fit <- glm(ldl_y ~ hrc_X + ldl_cov$age + ldl_cov$sex + ldl_cov$PC1 + ldl_cov$PC2, family = \"gaussian\")\n",
    "# coef(summary(hrc.fit))[2, 4]\n",
    "\n",
    "# ## topmed\n",
    "# topmed <- read.csv(\"PCSK9_topmed_rsq03_maf001_extracted.traw\") %>% filter(SNP == \"chr1:55052701:C:T\")\n",
    "# topmed_X <- format_traw(topmed, ldl_id, \"topmed\")\n",
    "\n",
    "# topmed.fit <- glm(ldl_y ~ topmed_X + ldl_cov$age + ldl_cov$sex + ldl_cov$PC1 + ldl_cov$PC2, family = \"gaussian\")\n",
    "# coef(summary(topmed.fit))[2, 4]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1a23cc7d-0b1c-4751-941c-9dcd48bf3ba2",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "### APOC3 v.s. TG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "1cf3e5d9-6067-4bee-8761-12a776222236",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "setwd('/mnt/vast/hpc/csg/tl3031/imputation-rvtest/analysis/imputation_aggregated_analysis/apoc3')\n",
    "\n",
    "result_df <- data.frame(data.frame(matrix(ncol = 5, nrow = 0)))\n",
    "colnames(result_df) <- c(\"data\", \"rsqs\", \"mafs\", \"num_var\", \"pvals\")\n",
    "\n",
    "for(maf in c(\"1\", \"05\", \"01\")){\n",
    "    exome_fname <- sprintf(\"APOC3_exome_maf00%s_extracted.raw\", maf)\n",
    "    if(file.exists(exome_fname)){\n",
    "        exome <- read.table(exome_fname, sep = \"\\t\", header = TRUE)\n",
    "        exome_X <- exome %>% arrange(FID, IID) %>% filter(FID %in% tg_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "        result_exome <- BRV(2 - exome_X, tg_y, tg_cov,\"gaussian\")\n",
    "    } else {\n",
    "        exome_X <- matrix()\n",
    "        result_exome$pval <- NA\n",
    "    }\n",
    "\n",
    "    for(rsq in c(3, 8)){\n",
    "        hrc_fname <- sprintf(\"APOC3_hrc_rsq0%d_maf00%s_extracted.raw\", rsq, maf)\n",
    "        topmed_fname <- sprintf(\"APOC3_topmed_v3_rsq0%d_maf00%s_extracted.raw\", rsq, maf)\n",
    "        hrc_topmed_fname <- sprintf(\"APOC3_hrc_topmed_v3_rsq0%d_maf00%s_extracted.raw\", rsq, maf)\n",
    "        hrc_topmed_exome_fname <- sprintf(\"APOC3_hrc_topmed_v3_exome_rsq0%d_maf00%s_extracted.raw\", rsq, maf)\n",
    "\n",
    "        if(file.exists(hrc_fname)){\n",
    "            hrc <- read.table(hrc_fname, sep = \"\\t\", header = TRUE)\n",
    "            \n",
    "            fid_iid <- hrc$IID %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "            hrc$FID <- as.integer(fid_iid[,1])\n",
    "            hrc$IID <- as.integer(fid_iid[,2])\n",
    "            hrc_X <- hrc %>% arrange(FID, IID) %>% filter(FID %in% tg_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "\n",
    "            result_hrc <- BRV(2 - hrc_X, tg_y, tg_cov,\"gaussian\")\n",
    "\n",
    "        } else {\n",
    "            hrc_X <- matrix()\n",
    "            result_hrc$pval <- NA\n",
    "        }\n",
    "        \n",
    "        if(file.exists(topmed_fname)){\n",
    "            topmed <- read.table(topmed_fname, sep = \"\\t\", header = TRUE)\n",
    "            \n",
    "            fid_iid <- topmed$IID %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "            topmed$FID <- as.integer(fid_iid[,1])\n",
    "            topmed$IID <- as.integer(fid_iid[,2])\n",
    "\n",
    "            topmed_X <- topmed %>% arrange(FID, IID) %>% filter(FID %in% tg_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "            result_topmed <- BRV(2 - topmed_X, tg_y, tg_cov,\"gaussian\")\n",
    "        } else {\n",
    "            topmed_X <- matrix()\n",
    "            result_topmed$pval <- NA\n",
    "        }\n",
    "\n",
    "        if(file.exists(hrc_topmed_fname)){\n",
    "            hrc_topmed <- read.table(hrc_topmed_fname, sep = \",\", header = TRUE)\n",
    "            hrc_topmed_X <- hrc_topmed %>% arrange(FID, IID) %>% filter(FID %in% tg_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "            result_hrc_topmed <- BRV(2 - hrc_topmed_X, tg_y, tg_cov,\"gaussian\")\n",
    "        } else {\n",
    "            hrc_topmed_X <- matrix()\n",
    "            result_hrc_topmed$pval <- NA\n",
    "        }\n",
    "\n",
    "        if(file.exists(hrc_topmed_exome_fname)){\n",
    "            hrc_topmed_exome <- read.csv(hrc_topmed_exome_fname, sep = \",\", header = TRUE)\n",
    "            hrc_topmed_exome_X <- hrc_topmed_exome %>% arrange(FID, IID) %>% filter(FID %in% tg_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "            result_hrc_topmed_exome <- BRV(2 - hrc_topmed_exome_X, tg_y, tg_cov,\"gaussian\")\n",
    "        } else {\n",
    "            hrc_topmed_exome_X <- matrix()\n",
    "            result_hrc_topmed_exome$pval <- NA\n",
    "        }\n",
    "\n",
    "        data <- c(\"hrc\", \"topmed\", \"hrc_topmed\", \"hrc_topmed_exome\")\n",
    "        rsqs <- rep(rsq/10, 4)\n",
    "        mafs <- rep(paste0(\"0.0\", maf), 4)\n",
    "        pvals <- c(result_hrc$pval, result_topmed$pval, result_hrc_topmed$pval, result_hrc_topmed_exome$pval)\n",
    "        num_var <- c(ncol(hrc_X), ncol(topmed_X), ncol(hrc_topmed_X), ncol(hrc_topmed_exome_X))\n",
    "        sub_df <- data.frame(data, rsqs, mafs, num_var, pvals)\n",
    "        result_df <- rbind(result_df, sub_df)\n",
    "    }\n",
    "    result_df <- rbind(data.frame(data = \"exome\", rsqs = \"NA\", mafs = paste0(\"0.0\", maf), num_var = ncol(exome_X), pvals = result_exome$pval), result_df)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "dfa7f085-7028-458f-985e-c1cab8ec9464",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 27 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>data</th><th scope=col>rsqs</th><th scope=col>mafs</th><th scope=col>num_var</th><th scope=col>pvals</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.01 </td><td>60</td><td>3.073400e-234</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.01 </td><td> 3</td><td>3.376964e-182</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.01 </td><td> 2</td><td>9.967896e-170</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.01 </td><td>18</td><td>1.342452e-190</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.01 </td><td> 4</td><td>1.761751e-183</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.01 </td><td>18</td><td>4.145103e-187</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.01 </td><td> 4</td><td>6.413400e-183</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.01 </td><td>65</td><td>3.666208e-231</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.01 </td><td>60</td><td>3.073400e-234</td></tr>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.005</td><td>60</td><td>3.073400e-234</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.005</td><td> 3</td><td>3.376964e-182</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.005</td><td> 2</td><td>9.967896e-170</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.005</td><td>18</td><td>1.342452e-190</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.005</td><td> 4</td><td>1.761751e-183</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.005</td><td>18</td><td>4.145103e-187</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.005</td><td> 4</td><td>6.413400e-183</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.005</td><td>65</td><td>3.666208e-231</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.005</td><td>60</td><td>3.073400e-234</td></tr>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.001</td><td>59</td><td> 5.468090e-32</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.001</td><td> 2</td><td> 3.385530e-24</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.001</td><td> 1</td><td>           NA</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.001</td><td>17</td><td> 1.393809e-25</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.001</td><td> 3</td><td> 1.488747e-13</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.001</td><td>17</td><td> 2.634573e-20</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.001</td><td> 3</td><td> 1.488747e-13</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.001</td><td>64</td><td> 1.188269e-30</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.001</td><td>59</td><td> 5.468090e-32</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 27 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " data & rsqs & mafs & num\\_var & pvals\\\\\n",
       " <fct> & <chr> & <chr> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t exome            & NA  & 0.01  & 60 & 3.073400e-234\\\\\n",
       "\t hrc              & 0.3 & 0.01  &  3 & 3.376964e-182\\\\\n",
       "\t hrc              & 0.8 & 0.01  &  2 & 9.967896e-170\\\\\n",
       "\t topmed           & 0.3 & 0.01  & 18 & 1.342452e-190\\\\\n",
       "\t topmed           & 0.8 & 0.01  &  4 & 1.761751e-183\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.01  & 18 & 4.145103e-187\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.01  &  4 & 6.413400e-183\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.01  & 65 & 3.666208e-231\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.01  & 60 & 3.073400e-234\\\\\n",
       "\t exome            & NA  & 0.005 & 60 & 3.073400e-234\\\\\n",
       "\t hrc              & 0.3 & 0.005 &  3 & 3.376964e-182\\\\\n",
       "\t hrc              & 0.8 & 0.005 &  2 & 9.967896e-170\\\\\n",
       "\t topmed           & 0.3 & 0.005 & 18 & 1.342452e-190\\\\\n",
       "\t topmed           & 0.8 & 0.005 &  4 & 1.761751e-183\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.005 & 18 & 4.145103e-187\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.005 &  4 & 6.413400e-183\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.005 & 65 & 3.666208e-231\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.005 & 60 & 3.073400e-234\\\\\n",
       "\t exome            & NA  & 0.001 & 59 &  5.468090e-32\\\\\n",
       "\t hrc              & 0.3 & 0.001 &  2 &  3.385530e-24\\\\\n",
       "\t hrc              & 0.8 & 0.001 &  1 &            NA\\\\\n",
       "\t topmed           & 0.3 & 0.001 & 17 &  1.393809e-25\\\\\n",
       "\t topmed           & 0.8 & 0.001 &  3 &  1.488747e-13\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.001 & 17 &  2.634573e-20\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.001 &  3 &  1.488747e-13\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.001 & 64 &  1.188269e-30\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.001 & 59 &  5.468090e-32\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 27 × 5\n",
       "\n",
       "| data &lt;fct&gt; | rsqs &lt;chr&gt; | mafs &lt;chr&gt; | num_var &lt;int&gt; | pvals &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| exome            | NA  | 0.01  | 60 | 3.073400e-234 |\n",
       "| hrc              | 0.3 | 0.01  |  3 | 3.376964e-182 |\n",
       "| hrc              | 0.8 | 0.01  |  2 | 9.967896e-170 |\n",
       "| topmed           | 0.3 | 0.01  | 18 | 1.342452e-190 |\n",
       "| topmed           | 0.8 | 0.01  |  4 | 1.761751e-183 |\n",
       "| hrc_topmed       | 0.3 | 0.01  | 18 | 4.145103e-187 |\n",
       "| hrc_topmed       | 0.8 | 0.01  |  4 | 6.413400e-183 |\n",
       "| hrc_topmed_exome | 0.3 | 0.01  | 65 | 3.666208e-231 |\n",
       "| hrc_topmed_exome | 0.8 | 0.01  | 60 | 3.073400e-234 |\n",
       "| exome            | NA  | 0.005 | 60 | 3.073400e-234 |\n",
       "| hrc              | 0.3 | 0.005 |  3 | 3.376964e-182 |\n",
       "| hrc              | 0.8 | 0.005 |  2 | 9.967896e-170 |\n",
       "| topmed           | 0.3 | 0.005 | 18 | 1.342452e-190 |\n",
       "| topmed           | 0.8 | 0.005 |  4 | 1.761751e-183 |\n",
       "| hrc_topmed       | 0.3 | 0.005 | 18 | 4.145103e-187 |\n",
       "| hrc_topmed       | 0.8 | 0.005 |  4 | 6.413400e-183 |\n",
       "| hrc_topmed_exome | 0.3 | 0.005 | 65 | 3.666208e-231 |\n",
       "| hrc_topmed_exome | 0.8 | 0.005 | 60 | 3.073400e-234 |\n",
       "| exome            | NA  | 0.001 | 59 |  5.468090e-32 |\n",
       "| hrc              | 0.3 | 0.001 |  2 |  3.385530e-24 |\n",
       "| hrc              | 0.8 | 0.001 |  1 |            NA |\n",
       "| topmed           | 0.3 | 0.001 | 17 |  1.393809e-25 |\n",
       "| topmed           | 0.8 | 0.001 |  3 |  1.488747e-13 |\n",
       "| hrc_topmed       | 0.3 | 0.001 | 17 |  2.634573e-20 |\n",
       "| hrc_topmed       | 0.8 | 0.001 |  3 |  1.488747e-13 |\n",
       "| hrc_topmed_exome | 0.3 | 0.001 | 64 |  1.188269e-30 |\n",
       "| hrc_topmed_exome | 0.8 | 0.001 | 59 |  5.468090e-32 |\n",
       "\n"
      ],
      "text/plain": [
       "   data             rsqs mafs  num_var pvals        \n",
       "1  exome            NA   0.01  60      3.073400e-234\n",
       "2  hrc              0.3  0.01   3      3.376964e-182\n",
       "3  hrc              0.8  0.01   2      9.967896e-170\n",
       "4  topmed           0.3  0.01  18      1.342452e-190\n",
       "5  topmed           0.8  0.01   4      1.761751e-183\n",
       "6  hrc_topmed       0.3  0.01  18      4.145103e-187\n",
       "7  hrc_topmed       0.8  0.01   4      6.413400e-183\n",
       "8  hrc_topmed_exome 0.3  0.01  65      3.666208e-231\n",
       "9  hrc_topmed_exome 0.8  0.01  60      3.073400e-234\n",
       "10 exome            NA   0.005 60      3.073400e-234\n",
       "11 hrc              0.3  0.005  3      3.376964e-182\n",
       "12 hrc              0.8  0.005  2      9.967896e-170\n",
       "13 topmed           0.3  0.005 18      1.342452e-190\n",
       "14 topmed           0.8  0.005  4      1.761751e-183\n",
       "15 hrc_topmed       0.3  0.005 18      4.145103e-187\n",
       "16 hrc_topmed       0.8  0.005  4      6.413400e-183\n",
       "17 hrc_topmed_exome 0.3  0.005 65      3.666208e-231\n",
       "18 hrc_topmed_exome 0.8  0.005 60      3.073400e-234\n",
       "19 exome            NA   0.001 59       5.468090e-32\n",
       "20 hrc              0.3  0.001  2       3.385530e-24\n",
       "21 hrc              0.8  0.001  1                 NA\n",
       "22 topmed           0.3  0.001 17       1.393809e-25\n",
       "23 topmed           0.8  0.001  3       1.488747e-13\n",
       "24 hrc_topmed       0.3  0.001 17       2.634573e-20\n",
       "25 hrc_topmed       0.8  0.001  3       1.488747e-13\n",
       "26 hrc_topmed_exome 0.3  0.001 64       1.188269e-30\n",
       "27 hrc_topmed_exome 0.8  0.001 59       5.468090e-32"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "result_df %>% \n",
    "    mutate(data = factor(data, levels = c(\"exome\", \"hrc\", \"topmed\", \"hrc_topmed\", \"hrc_topmed_exome\"))) %>% \n",
    "    arrange(desc(as.numeric(mafs)), data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "8356218a-8e06-44ba-a005-9055676f8111",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "result_df <- data.frame(data.frame(matrix(ncol = 5, nrow = 0)))\n",
    "colnames(result_df) <- c(\"data\", \"rsqs\", \"mafs\", \"num_var\", \"pvals\")\n",
    "\n",
    "for(maf in c(\"1\", \"05\", \"01\")){\n",
    "    exome_fname <- sprintf(\"APOC3_exome_maf00%s_cadd_extracted.raw\", maf)\n",
    "    if(file.exists(exome_fname)){\n",
    "        exome <- read.table(exome_fname, sep = \"\\t\", header = TRUE)\n",
    "        exome_X <- exome %>% arrange(FID, IID) %>% filter(FID %in% tg_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "        result_exome <- BRV(2 - exome_X, tg_y, tg_cov,\"gaussian\")\n",
    "    } else {\n",
    "        exome_X <- matrix()\n",
    "        result_exome$pval <- NA\n",
    "    }\n",
    "\n",
    "    for(rsq in c(3, 8)){\n",
    "        hrc_fname <- sprintf(\"APOC3_hrc_rsq0%d_maf00%s_cadd_extracted.raw\", rsq, maf)\n",
    "        topmed_fname <- sprintf(\"APOC3_topmed_v3_rsq0%d_maf00%s_cadd_extracted.raw\", rsq, maf)\n",
    "        hrc_topmed_fname <- sprintf(\"APOC3_hrc_topmed_v3_rsq0%d_maf00%s_cadd_extracted.raw\", rsq, maf)\n",
    "        hrc_topmed_exome_fname <- sprintf(\"APOC3_hrc_topmed_v3_exome_rsq0%d_maf00%s_cadd_extracted.raw\", rsq, maf)\n",
    "\n",
    "        if(file.exists(hrc_fname)){\n",
    "            hrc <- read.table(hrc_fname, sep = \"\\t\", header = TRUE)\n",
    "            hrc_names <- colnames(hrc)[7: dim(hrc)[2]] %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "            colnames(hrc)[7:dim(hrc)[2]] <- hrc_names[,1]\n",
    "\n",
    "            fid_iid <- hrc$IID %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "            hrc$FID <- as.integer(fid_iid[,1])\n",
    "            hrc$IID <- as.integer(fid_iid[,2])\n",
    "            hrc_X <- hrc %>% arrange(FID, IID) %>% filter(FID %in% tg_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "\n",
    "            result_hrc <- BRV(2 - hrc_X, tg_y, tg_cov,\"gaussian\")\n",
    "\n",
    "        } else {\n",
    "            hrc_X <- matrix()\n",
    "            result_hrc$pval <- NA\n",
    "        }\n",
    "        \n",
    "        if(file.exists(topmed_fname)){\n",
    "            topmed <- read.csv(topmed_fname, sep = \"\\t\", header = TRUE)\n",
    "            topmed_names <- colnames(topmed)[7: dim(topmed)[2]] %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "            colnames(topmed)[7:dim(topmed)[2]] <- topmed_names[,1]\n",
    "\n",
    "            fid_iid <- topmed$IID %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "            topmed$FID <- as.integer(fid_iid[,1])\n",
    "            topmed$IID <- as.integer(fid_iid[,2])\n",
    "\n",
    "            topmed_X <- topmed %>% arrange(FID, IID) %>% filter(FID %in% tg_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "            result_topmed <- BRV(2 - topmed_X, tg_y, tg_cov,\"gaussian\")\n",
    "        } else {\n",
    "            topmed_X <- matrix()\n",
    "            result_topmed$pval <- NA\n",
    "        }\n",
    "\n",
    "        if(file.exists(hrc_topmed_fname)){\n",
    "            hrc_topmed <- read.table(hrc_topmed_fname, sep = \",\", header = TRUE)\n",
    "            hrc_topmed_X <- hrc_topmed %>% arrange(FID, IID) %>% filter(FID %in% tg_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "            result_hrc_topmed <- BRV(2 - hrc_topmed_X, tg_y, tg_cov,\"gaussian\")\n",
    "        } else {\n",
    "            hrc_topmed_X <- matrix()\n",
    "            result_hrc_topmed$pval <- NA\n",
    "        }\n",
    "\n",
    "        if(file.exists(hrc_topmed_exome_fname)){\n",
    "            hrc_topmed_exome <- read.table(hrc_topmed_exome_fname, sep = \",\", header = TRUE)\n",
    "            hrc_topmed_exome_X <- hrc_topmed_exome %>% arrange(FID, IID) %>% filter(FID %in% tg_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "            result_hrc_topmed_exome <- BRV(2 - hrc_topmed_exome_X, tg_y, tg_cov,\"gaussian\")\n",
    "        } else {\n",
    "            hrc_topmed_exome_X <- matrix()\n",
    "            result_hrc_topmed_exome$pval <- NA\n",
    "        }\n",
    "\n",
    "        data <- c(\"hrc\", \"topmed\", \"hrc_topmed\", \"hrc_topmed_exome\")\n",
    "        rsqs <- rep(rsq/10, 4)\n",
    "        mafs <- rep(paste0(\"0.0\", maf), 4)\n",
    "        pvals <- c(result_hrc$pval, result_topmed$pval, result_hrc_topmed$pval, result_hrc_topmed_exome$pval)\n",
    "        num_var <- c(ncol(hrc_X), ncol(topmed_X), ncol(hrc_topmed_X), ncol(hrc_topmed_exome_X))\n",
    "        sub_df <- data.frame(data, rsqs, mafs, num_var, pvals)\n",
    "        result_df <- rbind(result_df, sub_df)\n",
    "    }\n",
    "    result_df <- rbind(data.frame(data = \"exome\", rsqs = \"NA\", mafs = paste0(\"0.0\", maf), num_var = ncol(exome_X), pvals = result_exome$pval), result_df)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "9abc5e21-0b9f-46d0-a721-6b14d25f1b23",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 27 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>data</th><th scope=col>rsqs</th><th scope=col>mafs</th><th scope=col>num_var</th><th scope=col>pvals</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.01 </td><td>26</td><td>1.050377e-262</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.01 </td><td> 3</td><td>3.376964e-182</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.01 </td><td> 2</td><td>9.967896e-170</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.01 </td><td>10</td><td>4.038258e-197</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.01 </td><td> 4</td><td>1.761751e-183</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.01 </td><td>10</td><td>4.597154e-194</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.01 </td><td> 4</td><td>6.413400e-183</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.01 </td><td>28</td><td>8.537625e-260</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.01 </td><td>26</td><td>1.050377e-262</td></tr>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.005</td><td>26</td><td>1.050377e-262</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.005</td><td> 3</td><td>3.376964e-182</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.005</td><td> 2</td><td>9.967896e-170</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.005</td><td>10</td><td>4.038258e-197</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.005</td><td> 4</td><td>1.761751e-183</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.005</td><td>10</td><td>4.597154e-194</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.005</td><td> 4</td><td>6.413400e-183</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.005</td><td>28</td><td>8.537625e-260</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.005</td><td>26</td><td>1.050377e-262</td></tr>\n",
       "\t<tr><td>exome           </td><td>NA </td><td>0.001</td><td>25</td><td> 9.616993e-48</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.3</td><td>0.001</td><td> 2</td><td> 3.385530e-24</td></tr>\n",
       "\t<tr><td>hrc             </td><td>0.8</td><td>0.001</td><td> 1</td><td>           NA</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.3</td><td>0.001</td><td> 9</td><td> 2.254842e-28</td></tr>\n",
       "\t<tr><td>topmed          </td><td>0.8</td><td>0.001</td><td> 3</td><td> 1.488747e-13</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.3</td><td>0.001</td><td> 9</td><td> 2.233109e-23</td></tr>\n",
       "\t<tr><td>hrc_topmed      </td><td>0.8</td><td>0.001</td><td> 3</td><td> 1.488747e-13</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.3</td><td>0.001</td><td>27</td><td> 1.482569e-45</td></tr>\n",
       "\t<tr><td>hrc_topmed_exome</td><td>0.8</td><td>0.001</td><td>25</td><td> 9.616993e-48</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 27 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " data & rsqs & mafs & num\\_var & pvals\\\\\n",
       " <fct> & <chr> & <chr> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t exome            & NA  & 0.01  & 26 & 1.050377e-262\\\\\n",
       "\t hrc              & 0.3 & 0.01  &  3 & 3.376964e-182\\\\\n",
       "\t hrc              & 0.8 & 0.01  &  2 & 9.967896e-170\\\\\n",
       "\t topmed           & 0.3 & 0.01  & 10 & 4.038258e-197\\\\\n",
       "\t topmed           & 0.8 & 0.01  &  4 & 1.761751e-183\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.01  & 10 & 4.597154e-194\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.01  &  4 & 6.413400e-183\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.01  & 28 & 8.537625e-260\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.01  & 26 & 1.050377e-262\\\\\n",
       "\t exome            & NA  & 0.005 & 26 & 1.050377e-262\\\\\n",
       "\t hrc              & 0.3 & 0.005 &  3 & 3.376964e-182\\\\\n",
       "\t hrc              & 0.8 & 0.005 &  2 & 9.967896e-170\\\\\n",
       "\t topmed           & 0.3 & 0.005 & 10 & 4.038258e-197\\\\\n",
       "\t topmed           & 0.8 & 0.005 &  4 & 1.761751e-183\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.005 & 10 & 4.597154e-194\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.005 &  4 & 6.413400e-183\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.005 & 28 & 8.537625e-260\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.005 & 26 & 1.050377e-262\\\\\n",
       "\t exome            & NA  & 0.001 & 25 &  9.616993e-48\\\\\n",
       "\t hrc              & 0.3 & 0.001 &  2 &  3.385530e-24\\\\\n",
       "\t hrc              & 0.8 & 0.001 &  1 &            NA\\\\\n",
       "\t topmed           & 0.3 & 0.001 &  9 &  2.254842e-28\\\\\n",
       "\t topmed           & 0.8 & 0.001 &  3 &  1.488747e-13\\\\\n",
       "\t hrc\\_topmed       & 0.3 & 0.001 &  9 &  2.233109e-23\\\\\n",
       "\t hrc\\_topmed       & 0.8 & 0.001 &  3 &  1.488747e-13\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.3 & 0.001 & 27 &  1.482569e-45\\\\\n",
       "\t hrc\\_topmed\\_exome & 0.8 & 0.001 & 25 &  9.616993e-48\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 27 × 5\n",
       "\n",
       "| data &lt;fct&gt; | rsqs &lt;chr&gt; | mafs &lt;chr&gt; | num_var &lt;int&gt; | pvals &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| exome            | NA  | 0.01  | 26 | 1.050377e-262 |\n",
       "| hrc              | 0.3 | 0.01  |  3 | 3.376964e-182 |\n",
       "| hrc              | 0.8 | 0.01  |  2 | 9.967896e-170 |\n",
       "| topmed           | 0.3 | 0.01  | 10 | 4.038258e-197 |\n",
       "| topmed           | 0.8 | 0.01  |  4 | 1.761751e-183 |\n",
       "| hrc_topmed       | 0.3 | 0.01  | 10 | 4.597154e-194 |\n",
       "| hrc_topmed       | 0.8 | 0.01  |  4 | 6.413400e-183 |\n",
       "| hrc_topmed_exome | 0.3 | 0.01  | 28 | 8.537625e-260 |\n",
       "| hrc_topmed_exome | 0.8 | 0.01  | 26 | 1.050377e-262 |\n",
       "| exome            | NA  | 0.005 | 26 | 1.050377e-262 |\n",
       "| hrc              | 0.3 | 0.005 |  3 | 3.376964e-182 |\n",
       "| hrc              | 0.8 | 0.005 |  2 | 9.967896e-170 |\n",
       "| topmed           | 0.3 | 0.005 | 10 | 4.038258e-197 |\n",
       "| topmed           | 0.8 | 0.005 |  4 | 1.761751e-183 |\n",
       "| hrc_topmed       | 0.3 | 0.005 | 10 | 4.597154e-194 |\n",
       "| hrc_topmed       | 0.8 | 0.005 |  4 | 6.413400e-183 |\n",
       "| hrc_topmed_exome | 0.3 | 0.005 | 28 | 8.537625e-260 |\n",
       "| hrc_topmed_exome | 0.8 | 0.005 | 26 | 1.050377e-262 |\n",
       "| exome            | NA  | 0.001 | 25 |  9.616993e-48 |\n",
       "| hrc              | 0.3 | 0.001 |  2 |  3.385530e-24 |\n",
       "| hrc              | 0.8 | 0.001 |  1 |            NA |\n",
       "| topmed           | 0.3 | 0.001 |  9 |  2.254842e-28 |\n",
       "| topmed           | 0.8 | 0.001 |  3 |  1.488747e-13 |\n",
       "| hrc_topmed       | 0.3 | 0.001 |  9 |  2.233109e-23 |\n",
       "| hrc_topmed       | 0.8 | 0.001 |  3 |  1.488747e-13 |\n",
       "| hrc_topmed_exome | 0.3 | 0.001 | 27 |  1.482569e-45 |\n",
       "| hrc_topmed_exome | 0.8 | 0.001 | 25 |  9.616993e-48 |\n",
       "\n"
      ],
      "text/plain": [
       "   data             rsqs mafs  num_var pvals        \n",
       "1  exome            NA   0.01  26      1.050377e-262\n",
       "2  hrc              0.3  0.01   3      3.376964e-182\n",
       "3  hrc              0.8  0.01   2      9.967896e-170\n",
       "4  topmed           0.3  0.01  10      4.038258e-197\n",
       "5  topmed           0.8  0.01   4      1.761751e-183\n",
       "6  hrc_topmed       0.3  0.01  10      4.597154e-194\n",
       "7  hrc_topmed       0.8  0.01   4      6.413400e-183\n",
       "8  hrc_topmed_exome 0.3  0.01  28      8.537625e-260\n",
       "9  hrc_topmed_exome 0.8  0.01  26      1.050377e-262\n",
       "10 exome            NA   0.005 26      1.050377e-262\n",
       "11 hrc              0.3  0.005  3      3.376964e-182\n",
       "12 hrc              0.8  0.005  2      9.967896e-170\n",
       "13 topmed           0.3  0.005 10      4.038258e-197\n",
       "14 topmed           0.8  0.005  4      1.761751e-183\n",
       "15 hrc_topmed       0.3  0.005 10      4.597154e-194\n",
       "16 hrc_topmed       0.8  0.005  4      6.413400e-183\n",
       "17 hrc_topmed_exome 0.3  0.005 28      8.537625e-260\n",
       "18 hrc_topmed_exome 0.8  0.005 26      1.050377e-262\n",
       "19 exome            NA   0.001 25       9.616993e-48\n",
       "20 hrc              0.3  0.001  2       3.385530e-24\n",
       "21 hrc              0.8  0.001  1                 NA\n",
       "22 topmed           0.3  0.001  9       2.254842e-28\n",
       "23 topmed           0.8  0.001  3       1.488747e-13\n",
       "24 hrc_topmed       0.3  0.001  9       2.233109e-23\n",
       "25 hrc_topmed       0.8  0.001  3       1.488747e-13\n",
       "26 hrc_topmed_exome 0.3  0.001 27       1.482569e-45\n",
       "27 hrc_topmed_exome 0.8  0.001 25       9.616993e-48"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "result_df %>% \n",
    "    mutate(data = factor(data, levels = c(\"exome\", \"hrc\", \"topmed\", \"hrc_topmed\", \"hrc_topmed_exome\"))) %>% \n",
    "    arrange(desc(as.numeric(mafs)), data)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c758871c-4b96-41dc-b9ff-cd3d9932e91b",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "source": [
    "#### APOC3: chr11:116830638\n",
    "\n",
    "We specifically looked into this variant, for the same reason we did for the PCSK9 variant"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 279,
   "id": "63cb5603-f425-486e-8a3f-4d09fb259520",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "'chr11:116830638:G:A'"
      ],
      "text/latex": [
       "'chr11:116830638:G:A'"
      ],
      "text/markdown": [
       "'chr11:116830638:G:A'"
      ],
      "text/plain": [
       "[1] \"chr11:116830638:G:A\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Chr</th><th scope=col>Start</th><th scope=col>End</th><th scope=col>Ref</th><th scope=col>Alt</th><th scope=col>ID_hg38</th><th scope=col>ID_hg19</th><th scope=col>R2_hrc</th><th scope=col>R2_topmed</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>11</td><td>116830638</td><td>116830638</td><td>G</td><td>A</td><td>chr11:116830638:G:A</td><td>chr11:116701354:G:A</td><td>0.836202</td><td>0.853641</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 9\n",
       "\\begin{tabular}{lllllllll}\n",
       " Chr & Start & End & Ref & Alt & ID\\_hg38 & ID\\_hg19 & R2\\_hrc & R2\\_topmed\\\\\n",
       " <int> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t 11 & 116830638 & 116830638 & G & A & chr11:116830638:G:A & chr11:116701354:G:A & 0.836202 & 0.853641\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 9\n",
       "\n",
       "| Chr &lt;int&gt; | Start &lt;int&gt; | End &lt;int&gt; | Ref &lt;chr&gt; | Alt &lt;chr&gt; | ID_hg38 &lt;chr&gt; | ID_hg19 &lt;chr&gt; | R2_hrc &lt;dbl&gt; | R2_topmed &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| 11 | 116830638 | 116830638 | G | A | chr11:116830638:G:A | chr11:116701354:G:A | 0.836202 | 0.853641 |\n",
       "\n"
      ],
      "text/plain": [
       "  Chr Start     End       Ref Alt ID_hg38             ID_hg19            \n",
       "1 11  116830638 116830638 G   A   chr11:116830638:G:A chr11:116701354:G:A\n",
       "  R2_hrc   R2_topmed\n",
       "1 0.836202 0.853641 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "annot_001 <- read.csv(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/hrc_topmed_exome/hrc_topmed_v3_exome_168206ids_chr11_rsq03_maf001_annot.csv.gz\") %>%\n",
    "    filter(Gene.refGene == \"APOC3\") %>% pull(ID)\n",
    "annot_0001 <- read.csv(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/hrc_topmed_exome/hrc_topmed_v3_exome_168206ids_chr11_rsq03_maf0001_annot.csv.gz\")  %>%\n",
    "    filter(Gene.refGene == \"APOC3\") %>% pull(ID)\n",
    "\n",
    "setdiff(annot_001, annot_0001)\n",
    "\n",
    "read.csv(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/hrc_topmed_exome/hrc_topmed_v3_exome_168206ids_chr11_rsq03_maf001_annot.csv.gz\") %>%\n",
    "    filter(ID == 'chr11:116830638:G:A') %>% select(c(1:7, R2_hrc, R2_topmed))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 258,
   "id": "c089b21b-6670-40d7-8ac7-5bda55134c22",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$zstat</dt>\n",
       "\t\t<dd>-31.6578684474724</dd>\n",
       "\t<dt>$pval</dt>\n",
       "\t\t<dd>2.83630024304138e-219</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$zstat] -31.6578684474724\n",
       "\\item[\\$pval] 2.83630024304138e-219\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$zstat\n",
       ":   -31.6578684474724\n",
       "$pval\n",
       ":   2.83630024304138e-219\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$zstat\n",
       "[1] -31.65787\n",
       "\n",
       "$pval\n",
       "[1] 2.8363e-219\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "exome <- read.csv(\"APOC3_exome_maf001_extracted.raw\")\n",
    "exome_names <- colnames(exome)[7: dim(exome)[2]] %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "colnames(exome)[7:dim(exome)[2]] <- exome_names[,1]\n",
    "\n",
    "exome_X <- exome %>% arrange(FID, IID) %>% filter(FID %in% tg_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "\n",
    "result_exome <- BRV(2 - exome_X[,'chr11:116830638:G:A'], tg_y, tg_cov,\"gaussian\")\n",
    "result_exome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 271,
   "id": "0b50c929-924f-4f0d-882d-1594c5a6f2a2",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$zstat</dt>\n",
       "\t\t<dd>-27.1246994882253</dd>\n",
       "\t<dt>$pval</dt>\n",
       "\t\t<dd>1.17443048801482e-161</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$zstat] -27.1246994882253\n",
       "\\item[\\$pval] 1.17443048801482e-161\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$zstat\n",
       ":   -27.1246994882253\n",
       "$pval\n",
       ":   1.17443048801482e-161\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$zstat\n",
       "[1] -27.1247\n",
       "\n",
       "$pval\n",
       "[1] 1.17443e-161\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "hrc <- read.csv(\"APOC3_hrc_rsq03_maf001_extracted.raw\")\n",
    "hrc_names <- colnames(hrc)[7: dim(hrc)[2]] %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "colnames(hrc)[7:dim(hrc)[2]] <- hrc_names[,1]\n",
    "\n",
    "fid_iid <- hrc$IID %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "hrc$FID <- as.integer(fid_iid[,1])\n",
    "hrc$IID <- as.integer(fid_iid[,2])\n",
    "\n",
    "hrc_X <- hrc %>% arrange(FID, IID) %>% filter(FID %in% tg_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "\n",
    "result_hrc <- BRV(2 - hrc_X[,'chr11:116701354:G:A'], tg_y, tg_cov,\"gaussian\")\n",
    "result_hrc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 274,
   "id": "c481aab3-a1bb-4ebe-a765-1e7203e42392",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$zstat</dt>\n",
       "\t\t<dd>-28.2447050272768</dd>\n",
       "\t<dt>$pval</dt>\n",
       "\t\t<dd>4.47229986689203e-175</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$zstat] -28.2447050272768\n",
       "\\item[\\$pval] 4.47229986689203e-175\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$zstat\n",
       ":   -28.2447050272768\n",
       "$pval\n",
       ":   4.47229986689203e-175\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$zstat\n",
       "[1] -28.24471\n",
       "\n",
       "$pval\n",
       "[1] 4.4723e-175\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "topmed <- read.csv(\"APOC3_topmed_v3_rsq03_maf001_extracted.raw\")\n",
    "topmed_names <- colnames(topmed)[7: dim(topmed)[2]] %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "colnames(topmed)[7:dim(topmed)[2]] <- topmed_names[,1]\n",
    "\n",
    "fid_iid <- topmed$IID %>% stringr::str_split(pattern = \"_\", simplify = TRUE)\n",
    "topmed$FID <- as.integer(fid_iid[,1])\n",
    "topmed$IID <- as.integer(fid_iid[,2])\n",
    "\n",
    "topmed_X <- topmed %>% arrange(FID, IID) %>% filter(FID %in% tg_id) %>% select(-c(1:6)) %>% as.matrix()\n",
    "\n",
    "result_topmed <- BRV(2 - topmed_X[,'chr11:116830638:G:A'], tg_y, tg_cov,\"gaussian\")\n",
    "result_topmed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8aa498dc-4d4a-4f35-b696-d6a1620d2a68",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [],
   "source": [
    "annot_001 <- read.csv(\"~/project/imputation-rvtest/analysis/imputation_aggregated_analysis/hrc_topmed/hrc_topmed_v3_168206ids_chr11_rsq03_maf001_annot.csv.gz\") %>%\n",
    "    filter(Gene.refGene == \"APOC3\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a0e55034-eca9-4dc4-be71-5d45ef8e7bbd",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 18 × 20</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Chr</th><th scope=col>Start</th><th scope=col>End</th><th scope=col>Ref</th><th scope=col>Alt</th><th scope=col>Func.refGene</th><th scope=col>Gene.refGene</th><th scope=col>ExonicFunc.refGene</th><th scope=col>MAF_nfe_exome</th><th scope=col>REVEL_score</th><th scope=col>ID_hg19</th><th scope=col>ID</th><th scope=col>Function</th><th scope=col>R2_hrc</th><th scope=col>R2_topmed</th><th scope=col>R2</th><th scope=col>source</th><th scope=col>RawScore</th><th scope=col>PHRED</th><th scope=col>ID_hg38</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>11</td><td>116830637</td><td>116830637</td><td>C </td><td>T</td><td>exonic;splicing</td><td>APOC3</td><td>stopgain         </td><td>6.000e-04</td><td>   NA</td><td>chr11:116701353:C:T</td><td>chr11:116830637:C:T </td><td>LoF     </td><td>0.542862</td><td>0.458809</td><td>0.542862</td><td>hrc   </td><td> 5.613917</td><td>34.000</td><td>chr11:116830637:C:T </td></tr>\n",
       "\t<tr><td>11</td><td>116830638</td><td>116830638</td><td>G </td><td>A</td><td>splicing       </td><td>APOC3</td><td>.                </td><td>2.300e-03</td><td>   NA</td><td>chr11:116701354:G:A</td><td>chr11:116830638:G:A </td><td>splicing</td><td>0.836202</td><td>0.853641</td><td>0.853641</td><td>topmed</td><td> 4.947405</td><td>33.000</td><td>chr11:116830638:G:A </td></tr>\n",
       "\t<tr><td>11</td><td>116830844</td><td>116830844</td><td>G </td><td>A</td><td>exonic         </td><td>APOC3</td><td>nonsynonymous SNV</td><td>2.000e-04</td><td>0.653</td><td>chr11:116701560:G:A</td><td>chr11:116830844:G:A </td><td>missense</td><td>0.973659</td><td>0.963205</td><td>0.973659</td><td>hrc   </td><td> 2.818346</td><td>23.200</td><td>chr11:116830844:G:A </td></tr>\n",
       "\t<tr><td>11</td><td>116830568</td><td>116830568</td><td>A </td><td>G</td><td>splicing       </td><td>APOC3</td><td>.                </td><td>6.169e-05</td><td>0.277</td><td>                   </td><td>chr11:116830568:A:G </td><td>splicing</td><td>0.000000</td><td>0.383569</td><td>0.383569</td><td>topmed</td><td> 4.960331</td><td>33.000</td><td>chr11:116830568:A:G </td></tr>\n",
       "\t<tr><td>11</td><td>116830592</td><td>116830592</td><td>C </td><td>T</td><td>exonic         </td><td>APOC3</td><td>nonsynonymous SNV</td><td>8.807e-06</td><td>0.380</td><td>                   </td><td>chr11:116830592:C:T </td><td>missense</td><td>0.000000</td><td>0.591272</td><td>0.591272</td><td>topmed</td><td> 2.265048</td><td>21.300</td><td>chr11:116830592:C:T </td></tr>\n",
       "\t<tr><td>11</td><td>116830593</td><td>116830593</td><td>G </td><td>A</td><td>exonic         </td><td>APOC3</td><td>nonsynonymous SNV</td><td>1.761e-05</td><td>0.342</td><td>                   </td><td>chr11:116830593:G:A </td><td>missense</td><td>0.000000</td><td>0.881843</td><td>0.881843</td><td>topmed</td><td> 2.287065</td><td>21.500</td><td>chr11:116830593:G:A </td></tr>\n",
       "\t<tr><td>11</td><td>116830620</td><td>116830620</td><td>C </td><td>T</td><td>exonic         </td><td>APOC3</td><td>nonsynonymous SNV</td><td>3.525e-05</td><td>0.062</td><td>                   </td><td>chr11:116830620:C:T </td><td>missense</td><td>0.000000</td><td>0.656993</td><td>0.656993</td><td>topmed</td><td> 0.402428</td><td> 5.454</td><td>chr11:116830620:C:T </td></tr>\n",
       "\t<tr><td>11</td><td>116830622</td><td>116830622</td><td>C </td><td>T</td><td>exonic         </td><td>APOC3</td><td>nonsynonymous SNV</td><td>0.000e+00</td><td>0.189</td><td>                   </td><td>chr11:116830622:C:T </td><td>missense</td><td>0.000000</td><td>0.597868</td><td>0.597868</td><td>topmed</td><td> 2.078035</td><td>19.850</td><td>chr11:116830622:C:T </td></tr>\n",
       "\t<tr><td>11</td><td>116830808</td><td>116830808</td><td>T </td><td>G</td><td>exonic         </td><td>APOC3</td><td>nonsynonymous SNV</td><td>       NA</td><td>0.157</td><td>                   </td><td>chr11:116830808:T:G </td><td>missense</td><td>0.000000</td><td>0.570624</td><td>0.570624</td><td>topmed</td><td> 0.188174</td><td> 2.990</td><td>chr11:116830808:T:G </td></tr>\n",
       "\t<tr><td>11</td><td>116830823</td><td>116830823</td><td>A </td><td>G</td><td>exonic         </td><td>APOC3</td><td>nonsynonymous SNV</td><td>8.860e-06</td><td>0.260</td><td>                   </td><td>chr11:116830823:A:G </td><td>missense</td><td>0.000000</td><td>0.749498</td><td>0.749498</td><td>topmed</td><td>-1.343152</td><td> 0.001</td><td>chr11:116830823:A:G </td></tr>\n",
       "\t<tr><td>11</td><td>116830832</td><td>116830832</td><td>G </td><td>A</td><td>exonic         </td><td>APOC3</td><td>nonsynonymous SNV</td><td>8.866e-06</td><td>0.508</td><td>                   </td><td>chr11:116830832:G:A </td><td>missense</td><td>0.000000</td><td>0.695691</td><td>0.695691</td><td>topmed</td><td> 3.295387</td><td>24.100</td><td>chr11:116830832:G:A </td></tr>\n",
       "\t<tr><td>11</td><td>116830833</td><td>116830833</td><td>C </td><td>A</td><td>exonic         </td><td>APOC3</td><td>nonsynonymous SNV</td><td>5.319e-05</td><td>0.589</td><td>                   </td><td>chr11:116830833:C:A </td><td>missense</td><td>0.000000</td><td>0.758045</td><td>0.758045</td><td>topmed</td><td> 2.985550</td><td>23.400</td><td>chr11:116830833:C:A </td></tr>\n",
       "\t<tr><td>11</td><td>116830838</td><td>116830838</td><td>A </td><td>G</td><td>exonic         </td><td>APOC3</td><td>nonsynonymous SNV</td><td>       NA</td><td>0.433</td><td>                   </td><td>chr11:116830838:A:G </td><td>missense</td><td>0.000000</td><td>0.321759</td><td>0.321759</td><td>topmed</td><td> 2.212210</td><td>21.000</td><td>chr11:116830838:A:G </td></tr>\n",
       "\t<tr><td>11</td><td>116830897</td><td>116830897</td><td>G </td><td>T</td><td>splicing       </td><td>APOC3</td><td>.                </td><td>1.799e-05</td><td>   NA</td><td>                   </td><td>chr11:116830897:G:T </td><td>splicing</td><td>0.000000</td><td>0.918159</td><td>0.918159</td><td>topmed</td><td> 6.211482</td><td>35.000</td><td>chr11:116830897:G:T </td></tr>\n",
       "\t<tr><td>11</td><td>116830901</td><td>116830902</td><td>AC</td><td>A</td><td>splicing       </td><td>APOC3</td><td>.                </td><td>4.539e-05</td><td>   NA</td><td>                   </td><td>chr11:116830901:AC:A</td><td>splicing</td><td>0.000000</td><td>0.448906</td><td>0.448906</td><td>topmed</td><td> 0.813508</td><td> 9.527</td><td>chr11:116830901:AC:A</td></tr>\n",
       "\t<tr><td>11</td><td>116832777</td><td>116832777</td><td>G </td><td>A</td><td>exonic         </td><td>APOC3</td><td>nonsynonymous SNV</td><td>8.823e-06</td><td>0.152</td><td>                   </td><td>chr11:116832777:G:A </td><td>missense</td><td>0.000000</td><td>0.481611</td><td>0.481611</td><td>topmed</td><td> 1.570958</td><td>16.190</td><td>chr11:116832777:G:A </td></tr>\n",
       "\t<tr><td>11</td><td>116832858</td><td>116832858</td><td>A </td><td>G</td><td>exonic         </td><td>APOC3</td><td>nonsynonymous SNV</td><td>1.759e-05</td><td>0.212</td><td>                   </td><td>chr11:116832858:A:G </td><td>missense</td><td>0.000000</td><td>0.782377</td><td>0.782377</td><td>topmed</td><td>-0.753573</td><td> 0.033</td><td>chr11:116832858:A:G </td></tr>\n",
       "\t<tr><td>11</td><td>116832873</td><td>116832873</td><td>G </td><td>A</td><td>exonic         </td><td>APOC3</td><td>nonsynonymous SNV</td><td>8.800e-06</td><td>0.165</td><td>                   </td><td>chr11:116832873:G:A </td><td>missense</td><td>0.000000</td><td>0.462782</td><td>0.462782</td><td>topmed</td><td> 0.639350</td><td> 7.890</td><td>chr11:116832873:G:A </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 18 × 20\n",
       "\\begin{tabular}{llllllllllllllllllll}\n",
       " Chr & Start & End & Ref & Alt & Func.refGene & Gene.refGene & ExonicFunc.refGene & MAF\\_nfe\\_exome & REVEL\\_score & ID\\_hg19 & ID & Function & R2\\_hrc & R2\\_topmed & R2 & source & RawScore & PHRED & ID\\_hg38\\\\\n",
       " <int> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t 11 & 116830637 & 116830637 & C  & T & exonic;splicing & APOC3 & stopgain          & 6.000e-04 &    NA & chr11:116701353:C:T & chr11:116830637:C:T  & LoF      & 0.542862 & 0.458809 & 0.542862 & hrc    &  5.613917 & 34.000 & chr11:116830637:C:T \\\\\n",
       "\t 11 & 116830638 & 116830638 & G  & A & splicing        & APOC3 & .                 & 2.300e-03 &    NA & chr11:116701354:G:A & chr11:116830638:G:A  & splicing & 0.836202 & 0.853641 & 0.853641 & topmed &  4.947405 & 33.000 & chr11:116830638:G:A \\\\\n",
       "\t 11 & 116830844 & 116830844 & G  & A & exonic          & APOC3 & nonsynonymous SNV & 2.000e-04 & 0.653 & chr11:116701560:G:A & chr11:116830844:G:A  & missense & 0.973659 & 0.963205 & 0.973659 & hrc    &  2.818346 & 23.200 & chr11:116830844:G:A \\\\\n",
       "\t 11 & 116830568 & 116830568 & A  & G & splicing        & APOC3 & .                 & 6.169e-05 & 0.277 &                     & chr11:116830568:A:G  & splicing & 0.000000 & 0.383569 & 0.383569 & topmed &  4.960331 & 33.000 & chr11:116830568:A:G \\\\\n",
       "\t 11 & 116830592 & 116830592 & C  & T & exonic          & APOC3 & nonsynonymous SNV & 8.807e-06 & 0.380 &                     & chr11:116830592:C:T  & missense & 0.000000 & 0.591272 & 0.591272 & topmed &  2.265048 & 21.300 & chr11:116830592:C:T \\\\\n",
       "\t 11 & 116830593 & 116830593 & G  & A & exonic          & APOC3 & nonsynonymous SNV & 1.761e-05 & 0.342 &                     & chr11:116830593:G:A  & missense & 0.000000 & 0.881843 & 0.881843 & topmed &  2.287065 & 21.500 & chr11:116830593:G:A \\\\\n",
       "\t 11 & 116830620 & 116830620 & C  & T & exonic          & APOC3 & nonsynonymous SNV & 3.525e-05 & 0.062 &                     & chr11:116830620:C:T  & missense & 0.000000 & 0.656993 & 0.656993 & topmed &  0.402428 &  5.454 & chr11:116830620:C:T \\\\\n",
       "\t 11 & 116830622 & 116830622 & C  & T & exonic          & APOC3 & nonsynonymous SNV & 0.000e+00 & 0.189 &                     & chr11:116830622:C:T  & missense & 0.000000 & 0.597868 & 0.597868 & topmed &  2.078035 & 19.850 & chr11:116830622:C:T \\\\\n",
       "\t 11 & 116830808 & 116830808 & T  & G & exonic          & APOC3 & nonsynonymous SNV &        NA & 0.157 &                     & chr11:116830808:T:G  & missense & 0.000000 & 0.570624 & 0.570624 & topmed &  0.188174 &  2.990 & chr11:116830808:T:G \\\\\n",
       "\t 11 & 116830823 & 116830823 & A  & G & exonic          & APOC3 & nonsynonymous SNV & 8.860e-06 & 0.260 &                     & chr11:116830823:A:G  & missense & 0.000000 & 0.749498 & 0.749498 & topmed & -1.343152 &  0.001 & chr11:116830823:A:G \\\\\n",
       "\t 11 & 116830832 & 116830832 & G  & A & exonic          & APOC3 & nonsynonymous SNV & 8.866e-06 & 0.508 &                     & chr11:116830832:G:A  & missense & 0.000000 & 0.695691 & 0.695691 & topmed &  3.295387 & 24.100 & chr11:116830832:G:A \\\\\n",
       "\t 11 & 116830833 & 116830833 & C  & A & exonic          & APOC3 & nonsynonymous SNV & 5.319e-05 & 0.589 &                     & chr11:116830833:C:A  & missense & 0.000000 & 0.758045 & 0.758045 & topmed &  2.985550 & 23.400 & chr11:116830833:C:A \\\\\n",
       "\t 11 & 116830838 & 116830838 & A  & G & exonic          & APOC3 & nonsynonymous SNV &        NA & 0.433 &                     & chr11:116830838:A:G  & missense & 0.000000 & 0.321759 & 0.321759 & topmed &  2.212210 & 21.000 & chr11:116830838:A:G \\\\\n",
       "\t 11 & 116830897 & 116830897 & G  & T & splicing        & APOC3 & .                 & 1.799e-05 &    NA &                     & chr11:116830897:G:T  & splicing & 0.000000 & 0.918159 & 0.918159 & topmed &  6.211482 & 35.000 & chr11:116830897:G:T \\\\\n",
       "\t 11 & 116830901 & 116830902 & AC & A & splicing        & APOC3 & .                 & 4.539e-05 &    NA &                     & chr11:116830901:AC:A & splicing & 0.000000 & 0.448906 & 0.448906 & topmed &  0.813508 &  9.527 & chr11:116830901:AC:A\\\\\n",
       "\t 11 & 116832777 & 116832777 & G  & A & exonic          & APOC3 & nonsynonymous SNV & 8.823e-06 & 0.152 &                     & chr11:116832777:G:A  & missense & 0.000000 & 0.481611 & 0.481611 & topmed &  1.570958 & 16.190 & chr11:116832777:G:A \\\\\n",
       "\t 11 & 116832858 & 116832858 & A  & G & exonic          & APOC3 & nonsynonymous SNV & 1.759e-05 & 0.212 &                     & chr11:116832858:A:G  & missense & 0.000000 & 0.782377 & 0.782377 & topmed & -0.753573 &  0.033 & chr11:116832858:A:G \\\\\n",
       "\t 11 & 116832873 & 116832873 & G  & A & exonic          & APOC3 & nonsynonymous SNV & 8.800e-06 & 0.165 &                     & chr11:116832873:G:A  & missense & 0.000000 & 0.462782 & 0.462782 & topmed &  0.639350 &  7.890 & chr11:116832873:G:A \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 18 × 20\n",
       "\n",
       "| Chr &lt;int&gt; | Start &lt;int&gt; | End &lt;int&gt; | Ref &lt;chr&gt; | Alt &lt;chr&gt; | Func.refGene &lt;chr&gt; | Gene.refGene &lt;chr&gt; | ExonicFunc.refGene &lt;chr&gt; | MAF_nfe_exome &lt;dbl&gt; | REVEL_score &lt;dbl&gt; | ID_hg19 &lt;chr&gt; | ID &lt;chr&gt; | Function &lt;chr&gt; | R2_hrc &lt;dbl&gt; | R2_topmed &lt;dbl&gt; | R2 &lt;dbl&gt; | source &lt;chr&gt; | RawScore &lt;dbl&gt; | PHRED &lt;dbl&gt; | ID_hg38 &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 11 | 116830637 | 116830637 | C  | T | exonic;splicing | APOC3 | stopgain          | 6.000e-04 |    NA | chr11:116701353:C:T | chr11:116830637:C:T  | LoF      | 0.542862 | 0.458809 | 0.542862 | hrc    |  5.613917 | 34.000 | chr11:116830637:C:T  |\n",
       "| 11 | 116830638 | 116830638 | G  | A | splicing        | APOC3 | .                 | 2.300e-03 |    NA | chr11:116701354:G:A | chr11:116830638:G:A  | splicing | 0.836202 | 0.853641 | 0.853641 | topmed |  4.947405 | 33.000 | chr11:116830638:G:A  |\n",
       "| 11 | 116830844 | 116830844 | G  | A | exonic          | APOC3 | nonsynonymous SNV | 2.000e-04 | 0.653 | chr11:116701560:G:A | chr11:116830844:G:A  | missense | 0.973659 | 0.963205 | 0.973659 | hrc    |  2.818346 | 23.200 | chr11:116830844:G:A  |\n",
       "| 11 | 116830568 | 116830568 | A  | G | splicing        | APOC3 | .                 | 6.169e-05 | 0.277 | <!----> | chr11:116830568:A:G  | splicing | 0.000000 | 0.383569 | 0.383569 | topmed |  4.960331 | 33.000 | chr11:116830568:A:G  |\n",
       "| 11 | 116830592 | 116830592 | C  | T | exonic          | APOC3 | nonsynonymous SNV | 8.807e-06 | 0.380 | <!----> | chr11:116830592:C:T  | missense | 0.000000 | 0.591272 | 0.591272 | topmed |  2.265048 | 21.300 | chr11:116830592:C:T  |\n",
       "| 11 | 116830593 | 116830593 | G  | A | exonic          | APOC3 | nonsynonymous SNV | 1.761e-05 | 0.342 | <!----> | chr11:116830593:G:A  | missense | 0.000000 | 0.881843 | 0.881843 | topmed |  2.287065 | 21.500 | chr11:116830593:G:A  |\n",
       "| 11 | 116830620 | 116830620 | C  | T | exonic          | APOC3 | nonsynonymous SNV | 3.525e-05 | 0.062 | <!----> | chr11:116830620:C:T  | missense | 0.000000 | 0.656993 | 0.656993 | topmed |  0.402428 |  5.454 | chr11:116830620:C:T  |\n",
       "| 11 | 116830622 | 116830622 | C  | T | exonic          | APOC3 | nonsynonymous SNV | 0.000e+00 | 0.189 | <!----> | chr11:116830622:C:T  | missense | 0.000000 | 0.597868 | 0.597868 | topmed |  2.078035 | 19.850 | chr11:116830622:C:T  |\n",
       "| 11 | 116830808 | 116830808 | T  | G | exonic          | APOC3 | nonsynonymous SNV |        NA | 0.157 | <!----> | chr11:116830808:T:G  | missense | 0.000000 | 0.570624 | 0.570624 | topmed |  0.188174 |  2.990 | chr11:116830808:T:G  |\n",
       "| 11 | 116830823 | 116830823 | A  | G | exonic          | APOC3 | nonsynonymous SNV | 8.860e-06 | 0.260 | <!----> | chr11:116830823:A:G  | missense | 0.000000 | 0.749498 | 0.749498 | topmed | -1.343152 |  0.001 | chr11:116830823:A:G  |\n",
       "| 11 | 116830832 | 116830832 | G  | A | exonic          | APOC3 | nonsynonymous SNV | 8.866e-06 | 0.508 | <!----> | chr11:116830832:G:A  | missense | 0.000000 | 0.695691 | 0.695691 | topmed |  3.295387 | 24.100 | chr11:116830832:G:A  |\n",
       "| 11 | 116830833 | 116830833 | C  | A | exonic          | APOC3 | nonsynonymous SNV | 5.319e-05 | 0.589 | <!----> | chr11:116830833:C:A  | missense | 0.000000 | 0.758045 | 0.758045 | topmed |  2.985550 | 23.400 | chr11:116830833:C:A  |\n",
       "| 11 | 116830838 | 116830838 | A  | G | exonic          | APOC3 | nonsynonymous SNV |        NA | 0.433 | <!----> | chr11:116830838:A:G  | missense | 0.000000 | 0.321759 | 0.321759 | topmed |  2.212210 | 21.000 | chr11:116830838:A:G  |\n",
       "| 11 | 116830897 | 116830897 | G  | T | splicing        | APOC3 | .                 | 1.799e-05 |    NA | <!----> | chr11:116830897:G:T  | splicing | 0.000000 | 0.918159 | 0.918159 | topmed |  6.211482 | 35.000 | chr11:116830897:G:T  |\n",
       "| 11 | 116830901 | 116830902 | AC | A | splicing        | APOC3 | .                 | 4.539e-05 |    NA | <!----> | chr11:116830901:AC:A | splicing | 0.000000 | 0.448906 | 0.448906 | topmed |  0.813508 |  9.527 | chr11:116830901:AC:A |\n",
       "| 11 | 116832777 | 116832777 | G  | A | exonic          | APOC3 | nonsynonymous SNV | 8.823e-06 | 0.152 | <!----> | chr11:116832777:G:A  | missense | 0.000000 | 0.481611 | 0.481611 | topmed |  1.570958 | 16.190 | chr11:116832777:G:A  |\n",
       "| 11 | 116832858 | 116832858 | A  | G | exonic          | APOC3 | nonsynonymous SNV | 1.759e-05 | 0.212 | <!----> | chr11:116832858:A:G  | missense | 0.000000 | 0.782377 | 0.782377 | topmed | -0.753573 |  0.033 | chr11:116832858:A:G  |\n",
       "| 11 | 116832873 | 116832873 | G  | A | exonic          | APOC3 | nonsynonymous SNV | 8.800e-06 | 0.165 | <!----> | chr11:116832873:G:A  | missense | 0.000000 | 0.462782 | 0.462782 | topmed |  0.639350 |  7.890 | chr11:116832873:G:A  |\n",
       "\n"
      ],
      "text/plain": [
       "   Chr Start     End       Ref Alt Func.refGene    Gene.refGene\n",
       "1  11  116830637 116830637 C   T   exonic;splicing APOC3       \n",
       "2  11  116830638 116830638 G   A   splicing        APOC3       \n",
       "3  11  116830844 116830844 G   A   exonic          APOC3       \n",
       "4  11  116830568 116830568 A   G   splicing        APOC3       \n",
       "5  11  116830592 116830592 C   T   exonic          APOC3       \n",
       "6  11  116830593 116830593 G   A   exonic          APOC3       \n",
       "7  11  116830620 116830620 C   T   exonic          APOC3       \n",
       "8  11  116830622 116830622 C   T   exonic          APOC3       \n",
       "9  11  116830808 116830808 T   G   exonic          APOC3       \n",
       "10 11  116830823 116830823 A   G   exonic          APOC3       \n",
       "11 11  116830832 116830832 G   A   exonic          APOC3       \n",
       "12 11  116830833 116830833 C   A   exonic          APOC3       \n",
       "13 11  116830838 116830838 A   G   exonic          APOC3       \n",
       "14 11  116830897 116830897 G   T   splicing        APOC3       \n",
       "15 11  116830901 116830902 AC  A   splicing        APOC3       \n",
       "16 11  116832777 116832777 G   A   exonic          APOC3       \n",
       "17 11  116832858 116832858 A   G   exonic          APOC3       \n",
       "18 11  116832873 116832873 G   A   exonic          APOC3       \n",
       "   ExonicFunc.refGene MAF_nfe_exome REVEL_score ID_hg19            \n",
       "1  stopgain           6.000e-04        NA       chr11:116701353:C:T\n",
       "2  .                  2.300e-03        NA       chr11:116701354:G:A\n",
       "3  nonsynonymous SNV  2.000e-04     0.653       chr11:116701560:G:A\n",
       "4  .                  6.169e-05     0.277                          \n",
       "5  nonsynonymous SNV  8.807e-06     0.380                          \n",
       "6  nonsynonymous SNV  1.761e-05     0.342                          \n",
       "7  nonsynonymous SNV  3.525e-05     0.062                          \n",
       "8  nonsynonymous SNV  0.000e+00     0.189                          \n",
       "9  nonsynonymous SNV         NA     0.157                          \n",
       "10 nonsynonymous SNV  8.860e-06     0.260                          \n",
       "11 nonsynonymous SNV  8.866e-06     0.508                          \n",
       "12 nonsynonymous SNV  5.319e-05     0.589                          \n",
       "13 nonsynonymous SNV         NA     0.433                          \n",
       "14 .                  1.799e-05        NA                          \n",
       "15 .                  4.539e-05        NA                          \n",
       "16 nonsynonymous SNV  8.823e-06     0.152                          \n",
       "17 nonsynonymous SNV  1.759e-05     0.212                          \n",
       "18 nonsynonymous SNV  8.800e-06     0.165                          \n",
       "   ID                   Function R2_hrc   R2_topmed R2       source RawScore \n",
       "1  chr11:116830637:C:T  LoF      0.542862 0.458809  0.542862 hrc     5.613917\n",
       "2  chr11:116830638:G:A  splicing 0.836202 0.853641  0.853641 topmed  4.947405\n",
       "3  chr11:116830844:G:A  missense 0.973659 0.963205  0.973659 hrc     2.818346\n",
       "4  chr11:116830568:A:G  splicing 0.000000 0.383569  0.383569 topmed  4.960331\n",
       "5  chr11:116830592:C:T  missense 0.000000 0.591272  0.591272 topmed  2.265048\n",
       "6  chr11:116830593:G:A  missense 0.000000 0.881843  0.881843 topmed  2.287065\n",
       "7  chr11:116830620:C:T  missense 0.000000 0.656993  0.656993 topmed  0.402428\n",
       "8  chr11:116830622:C:T  missense 0.000000 0.597868  0.597868 topmed  2.078035\n",
       "9  chr11:116830808:T:G  missense 0.000000 0.570624  0.570624 topmed  0.188174\n",
       "10 chr11:116830823:A:G  missense 0.000000 0.749498  0.749498 topmed -1.343152\n",
       "11 chr11:116830832:G:A  missense 0.000000 0.695691  0.695691 topmed  3.295387\n",
       "12 chr11:116830833:C:A  missense 0.000000 0.758045  0.758045 topmed  2.985550\n",
       "13 chr11:116830838:A:G  missense 0.000000 0.321759  0.321759 topmed  2.212210\n",
       "14 chr11:116830897:G:T  splicing 0.000000 0.918159  0.918159 topmed  6.211482\n",
       "15 chr11:116830901:AC:A splicing 0.000000 0.448906  0.448906 topmed  0.813508\n",
       "16 chr11:116832777:G:A  missense 0.000000 0.481611  0.481611 topmed  1.570958\n",
       "17 chr11:116832858:A:G  missense 0.000000 0.782377  0.782377 topmed -0.753573\n",
       "18 chr11:116832873:G:A  missense 0.000000 0.462782  0.462782 topmed  0.639350\n",
       "   PHRED  ID_hg38             \n",
       "1  34.000 chr11:116830637:C:T \n",
       "2  33.000 chr11:116830638:G:A \n",
       "3  23.200 chr11:116830844:G:A \n",
       "4  33.000 chr11:116830568:A:G \n",
       "5  21.300 chr11:116830592:C:T \n",
       "6  21.500 chr11:116830593:G:A \n",
       "7   5.454 chr11:116830620:C:T \n",
       "8  19.850 chr11:116830622:C:T \n",
       "9   2.990 chr11:116830808:T:G \n",
       "10  0.001 chr11:116830823:A:G \n",
       "11 24.100 chr11:116830832:G:A \n",
       "12 23.400 chr11:116830833:C:A \n",
       "13 21.000 chr11:116830838:A:G \n",
       "14 35.000 chr11:116830897:G:T \n",
       "15  9.527 chr11:116830901:AC:A\n",
       "16 16.190 chr11:116832777:G:A \n",
       "17  0.033 chr11:116832858:A:G \n",
       "18  7.890 chr11:116832873:G:A "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "annot_001"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "049fb1a8-a706-45d3-924f-307db6ab7c6d",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "SoS",
   "language": "sos",
   "name": "sos"
  },
  "language_info": {
   "codemirror_mode": "sos",
   "file_extension": ".sos",
   "mimetype": "text/x-sos",
   "name": "sos",
   "nbconvert_exporter": "sos_notebook.converter.SoS_Exporter",
   "pygments_lexer": "sos"
  },
  "sos": {
   "kernels": [
    [
     "Bash",
     "calysto_bash",
     "Bash",
     "#E6EEFF",
     "shell"
    ],
    [
     "R",
     "ir",
     "R",
     "#DCDCDA",
     "r"
    ]
   ],
   "version": "0.24.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
